

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>gMCS &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../index.html"/>
        <link rel="up" title="Analysis" href="../index.html"/>
        <link rel="next" title="KLdistance" href="../KLdistance/index.html"/>
        <link rel="prev" title="geometricFBA" href="../geometricFBA/index.html"/> 

  
  <script src="../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../index.html">
          

          
            
            <img src="http://gibbs.unal.edu.co/cobradoc/cobratoolbox/img/logo.png" class="logo" alt="logo"/>
          
          </a>
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../_static/../" + item.website_url + "' ><img src='../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../dynamicFBA/index.html">Dynamic FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FBA/index.html">FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../findMIIS/index.html">findMIIS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fluxEnrichmentAnalysis/index.html">fluxEnrichmentAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FVA/index.html">FVA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../geometricFBA/index.html">geometricFBA</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">gMCS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../KLdistance/index.html">KLdistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../mdFBA/index.html">mdFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../MOMA/index.html">MOMA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../multiSpecies/index.html">MultiSpecies</a></li>
<li class="toctree-l3"><a class="reference internal" href="../parsimoniousFBA/index.html">parsimoniousFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../relaxedFBA/index.html">relaxedFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rFBA/index.html">rFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rMTA/index.html">rMTA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../robustnessAnalysis/index.html">robustnessAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../ROOM/index.html">ROOM</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rumba/index.html">rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sparseFBA/index.html">sparseFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../thermo/index.html">Thermo</a></li>
<li class="toctree-l3"><a class="reference internal" href="../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../uFBA/index.html">uFBA</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../dataIntegration/index.html">Data integration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../funding.html">Funding</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../contents.html">Docs</a> &raquo;</li>
        
          <li><a href="../../index.html">Functions</a> &raquo;</li>
        
          <li><a href="../index.html">Analysis</a> &raquo;</li>
        
      <li>gMCS</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="gmcs">
<span id="id1"></span><h1>gMCS<a class="headerlink" href="#gmcs" title="Permalink to this headline">¶</a></h1>
<span class="target" id="module-src.analysis.gMCS"></span><dl class="function">
<dt id="src.analysis.gMCS.GPR2models">
<code class="descname">GPR2models</code><span class="sig-paren">(</span><em>metabolic_model</em>, <em>selected_rxns</em>, <em>separate_transcript</em>, <em>numWorkers</em>, <em>printLevel</em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/efb2c06be702aaed38349581ba73ed420251992b/src/analysis/gMCS/GPR2models.m"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#src.analysis.gMCS.GPR2models" title="Permalink to this definition">¶</a></dt>
<dd><p>Each GPR rule is converted into a network where the reaction and genes
involved are interconnected.</p>
<p><strong>Usage</strong></p>
<blockquote>
<div>[networks, rxnNumGenes] = GPR2models(metabolic_model, selected_rxns, separate_transcript, numWorkers, printLevel)</div></blockquote>
<p><strong>Inputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>metabolic_model</strong> &#8211;
Metabolic model structure (COBRA Toolbox format)</li>
<li><strong>selected_rxns</strong> &#8211;
Index array which indicates selected reactions to
calculate the network.</li>
<li><strong>separate_transcript</strong> &#8211;
Character used to discriminate
different isoforms of a gene. Default &#8216;&#8217;.</li>
</ul>
</div></blockquote>
<p><strong>Optional inputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>numWorkers</strong> &#8211;
Maximum number of workers
* 0 - maximum provided by the system (automatic)
(default). If parallel pool is active, numWorkers
is defined by the system, otherwise is 1.
* 1 - sequential
* 2+ - parallel</li>
<li><strong>printLevel</strong> &#8211;
show the reactions created in models.
* 0 - shows nothing
* 1 - shows progress by reactions (default)
* 2+ - shows everything (reaction and network generation)</li>
</ul>
</div></blockquote>
<p><strong>Outputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>networks</strong> &#8211;
Cell array which contains all the networks.</li>
<li><strong>rxnNumGenes</strong> &#8211;
Array with the number of genes present in each GPR
rule for each reaction.</li>
</ul>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="src.analysis.gMCS.buildGmatrix">
<code class="descname">buildGmatrix</code><span class="sig-paren">(</span><em>model_name</em>, <em>model</em>, <em>separate_isoform</em>, <em>numWorkers</em>, <em>printLevel</em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/efb2c06be702aaed38349581ba73ed420251992b/src/analysis/gMCS/buildGmatrix.m"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#src.analysis.gMCS.buildGmatrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Build the G matrix required for the calculation of genetic Minimal Cut
Sets (gMCSs).</p>
<p><strong>Usage</strong></p>
<blockquote>
<div>[G, G_ind, related, n_genes_KO, G_time] = buildGmatrix(model_name, model_struct, separate_isoform, numWorkers, printLevel)</div></blockquote>
<p><strong>Inputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>model_name</strong> &#8211;
Name of the metabolic model under study.</li>
<li><strong>model_struct</strong> &#8211;
Metabolic model structure (COBRA Toolbox format).</li>
<li><strong>separate_isoform</strong> &#8211;
Character used to discriminate different isoforms of a gene.</li>
</ul>
</div></blockquote>
<p><strong>Optional inputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>numWorkers</strong> &#8211;
Maximum number of workers
* 0 - maximum provided by the system (automatic)
* 1 - sequential
* 2+ - parallel</li>
<li><strong>printLevel</strong> &#8211;
show the reactions created in models.
* 0 - shows nothing
* 1 - shows progress by reactions (default)
* 2+ - shows everything (reaction and network generation)</li>
</ul>
</div></blockquote>
<p><strong>Outputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>G</strong> &#8211;
G matrix.</li>
<li><strong>G_ind</strong> &#8211;
Gene knockouts associated with each row in the G matrix.</li>
<li><strong>related</strong> &#8211;
Relationships among rows of the G matrix.</li>
<li><strong>n_genes_KO</strong> &#8211;
Number of genes whose knockout is added by each row of the G matrix.</li>
<li><strong>G_time</strong> &#8211;
Calculation times for each of the steps.</li>
</ul>
</div></blockquote>
<p><strong>Example</strong></p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">[</span><span class="n">G</span><span class="p">,</span> <span class="n">G_ind</span><span class="p">,</span> <span class="n">related</span><span class="p">,</span> <span class="n">n_genes_KO</span><span class="p">,</span> <span class="n">G_time</span><span class="p">]</span> <span class="o">=</span> <span class="n">buildGmatrix</span><span class="p">(</span><span class="s1">&#39;Recon2.v04&#39;</span><span class="p">,</span> <span class="n">modelR204</span><span class="p">,</span> <span class="s1">&#39;.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="src.analysis.gMCS.calculateGeneMCS">
<code class="descname">calculateGeneMCS</code><span class="sig-paren">(</span><em>model_name</em>, <em>model_struct</em>, <em>n_gmcs</em>, <em>max_len_gmcs</em>, <em>options</em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/efb2c06be702aaed38349581ba73ed420251992b/src/analysis/gMCS/calculateGeneMCS.m"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#src.analysis.gMCS.calculateGeneMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate genetic Minimal Cut Sets (gMCSs) using the warm-start strategy
available in CPLEX, namely cplex.populate(), with or without selecting a
given knockout, among all the genes included in the model or a given
subset of them. Apaolaza et al., 2017 (Nature Communications).</p>
<p><strong>Usage</strong></p>
<blockquote>
<div>[gmcs, gmcs_time] = calculateGeneMCS(model_name, model_struct, n_gmcs, max_len_gmcs, options)</div></blockquote>
<p><strong>Inputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>model_name</strong> &#8211;
Name of the metabolic model under study (in order to
identify the G matrix).</li>
<li><strong>model_struct</strong> &#8211;
Metabolic model structure (COBRA Toolbox format).</li>
<li><strong>n_gmcs</strong> &#8211;
Number of gMCSs to calculate.</li>
<li><strong>max_len_gmcs</strong> &#8211;
Number of genes in the largest gMCS to be calculated.</li>
</ul>
</div></blockquote>
<p><strong>Optional input</strong></p>
<blockquote>
<div><ul>
<li><p class="first"><strong>options</strong> &#8211;
Structure with fields:</p>
<ul>
<li><p class="first">.KO - Selected gene knockout. Default: [].</p>
</li>
<li><p class="first">.gene_set - Cell array containing the set of
genes among which the gMCSs are wanted to be calculated.
Default: [] (all genes are included).</p>
</li>
<li><p class="first">.timelimit - Time limit for the calculation of gMCSs
each time the solver is called. Default: 1e75.</p>
</li>
<li><p class="first">.target_b - Desired activity level of the metabolic
task to be disrupted. Default: 1e-3;</p>
</li>
<li><p class="first">.separate_transcript - Character used to discriminate
different transcripts of a gene. Default: &#8216;&#8217;.
Example: separate_transcript = &#8216;&#8217;</p>
<blockquote>
<div><blockquote>
<div><p>gene 10005.1    ==&gt;    gene 10005.1
gene 10005.2    ==&gt;    gene 10005.2
gene 10005.3    ==&gt;    gene 10005.3</p>
</div></blockquote>
<dl class="docutils">
<dt>separate_transcript = &#8216;.&#8217;</dt>
<dd><p class="first last">gene 10005.1
gene 10005.2    ==&gt;    gene 10005
gene 10005.3</p>
</dd>
</dl>
</div></blockquote>
</li>
<li><p class="first">.forceLength - 1 if the constraint limiting the
length of the gMCSs is to be active (recommended for
enumerating low order gMCSs), 0 otherwise.
Default: 1.</p>
</li>
<li><p class="first">.numWorkers  - is the maximun number of workers</p>
</li>
</ul>
<p>used by Cplex and GPR2models. 0 = automatic, 1 =
sequential, &gt;1 = parallel. Default = 0;
* .printLevel - 1 if the process is wanted to be</p>
<blockquote>
<div><p>shown on the screen, 0 otherwise. Default: 1.</p>
</div></blockquote>
</li>
</ul>
</div></blockquote>
<p><strong>Outputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>gmcs</strong> &#8211;
Cell array containing the calculated gMCSs.</li>
<li><strong>gmcs_time</strong> &#8211;
Calculation times of the different processes in
the algorithm.</li>
</ul>
</div></blockquote>
<p><strong>Example</strong></p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="o">%</span><span class="n">With</span> <span class="n">optional</span> <span class="n">values</span>
<span class="p">[</span><span class="n">gmcs</span><span class="p">,</span> <span class="n">gmcs_time</span><span class="p">]</span> <span class="o">=</span> <span class="n">calculateGeneMCS</span><span class="p">(</span><span class="s1">&#39;Recon2.v04&#39;</span><span class="p">,</span> <span class="n">modelR204</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="n">options</span><span class="p">)</span>
<span class="o">%</span><span class="n">Being</span><span class="p">:</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">KO</span> <span class="o">=</span> <span class="s1">&#39;6240&#39;</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">gene_set</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;2987&#39;</span><span class="p">;</span> <span class="s1">&#39;6241&#39;</span><span class="p">}</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">timelimit</span> <span class="o">=</span> <span class="mi">300</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">target_b</span> <span class="o">=</span> <span class="mf">1e-4</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">separate_transcript</span> <span class="o">=</span> <span class="s1">&#39;.&#39;</span><span class="p">;</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">forceLength</span> <span class="o">=</span> <span class="mi">0</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">printLevel</span> <span class="o">=</span> <span class="mi">0</span>

<span class="o">%</span><span class="n">Without</span> <span class="n">optional</span> <span class="n">values</span>
<span class="p">[</span><span class="n">gmcs</span><span class="p">,</span> <span class="n">gmcs_time</span><span class="p">]</span> <span class="o">=</span> <span class="n">calculateGeneMCS</span><span class="p">(</span><span class="s1">&#39;ecoli_core_model&#39;</span><span class="p">,</span> <span class="n">model</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="mi">10</span><span class="p">)</span>
</pre></div>
</div>
<p>Check the installation of cplex</p>
</dd></dl>

<dl class="function">
<dt id="src.analysis.gMCS.calculateMCS">
<code class="descname">calculateMCS</code><span class="sig-paren">(</span><em>model_struct</em>, <em>n_mcs</em>, <em>max_len_mcs</em>, <em>options</em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/efb2c06be702aaed38349581ba73ed420251992b/src/analysis/gMCS/calculateMCS.m"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#src.analysis.gMCS.calculateMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate Minimal Cut Sets (MCSs) using the warm-start strategy available
in CPLEX, namely cplex.populate(), with or without selecting a given
knockout, among all the reactions included in the model or a given subset
of them. Tobalina et al., 2016 (Bioinformatics); von Kamp and Klamt, 2014
(PLoS Computational Biology).</p>
<p><strong>Usage</strong></p>
<blockquote>
<div>[mcs, mcs_time] = populateMCS(model_struct, n_mcs, max_len_mcs, options)</div></blockquote>
<p><strong>Inputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>model_struct</strong> &#8211;
Metabolic model structure (COBRA Toolbox format).</li>
<li><strong>n_mcs</strong> &#8211;
Number of MCSs to calculate.</li>
<li><strong>max_len_mcs</strong> &#8211;
Number of reactions in the largest MCS to be calculated.</li>
</ul>
</div></blockquote>
<p><strong>Optional input</strong></p>
<blockquote>
<div><ul>
<li><p class="first"><strong>options</strong> &#8211;
Structure with fields:</p>
<ul class="simple">
<li>.KO - Selected reaction knockout. Default: [].</li>
<li>.rxn_set - Cell array containing the set of
reactions (&#8216;rxns&#8217;) among which the MCSs are
wanted to be calculated. Default: [] (all reactions
are included).</li>
<li>.timelimit - Time limit for the calculation of MCSs
each time the solver is called. Default: 1e75.</li>
<li>.target_b - Desired activity level of the metabolic
task to be disrupted. Default: 1e-3;</li>
<li>.forceLength - 1 if the constraint limiting the
length of the MCSs is to be active (recommended for
enumerating low order MCSs), 0 otherwise.
Default: 1.</li>
<li>.numWorkers  - is the maximun number of workers</li>
</ul>
<p>used by Cplex and GPR2models. 0 = automatic, 1 =
sequential, &gt;1 = parallel. Default = 0;
* .printLevel - 1 if the process is wanted to be</p>
<blockquote>
<div><p>shown on the screen, 0 otherwise. Default: 1.</p>
</div></blockquote>
</li>
</ul>
</div></blockquote>
<p><strong>Outputs</strong></p>
<blockquote>
<div><ul class="simple">
<li><strong>mcs</strong> &#8211;
Cell array containing the calculated MCSs.</li>
<li><strong>mcs_time</strong> &#8211;
Calculation times of the different processes in
the algorithm.</li>
</ul>
</div></blockquote>
<p><strong>Example</strong></p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="o">%</span><span class="n">With</span> <span class="n">optional</span> <span class="n">values</span>
<span class="p">[</span><span class="n">mcs</span><span class="p">,</span> <span class="n">mcs_time</span><span class="p">]</span> <span class="o">=</span> <span class="n">calculateMCS</span><span class="p">(</span><span class="n">modelR204</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="n">options</span><span class="p">)</span>
<span class="o">%</span><span class="n">Being</span><span class="p">:</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">KO</span> <span class="o">=</span> <span class="s1">&#39;r1651&#39;</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">rxn_set</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;r1652&#39;</span><span class="p">;</span> <span class="s1">&#39;r1653&#39;</span><span class="p">;</span> <span class="s1">&#39;r1654&#39;</span><span class="p">}</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">timelimit</span> <span class="o">=</span> <span class="mi">300</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">target_b</span> <span class="o">=</span> <span class="mf">1e-4</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">forceLength</span> <span class="o">=</span> <span class="mi">0</span>
<span class="o">%</span><span class="n">options</span><span class="o">.</span><span class="n">printLevel</span> <span class="o">=</span> <span class="mi">0</span>

<span class="o">%</span><span class="n">Without</span> <span class="n">optional</span> <span class="n">values</span>
<span class="p">[</span><span class="n">mcs</span><span class="p">,</span> <span class="n">mcs_time</span><span class="p">]</span> <span class="o">=</span> <span class="n">calculateMCS</span><span class="p">(</span><span class="n">model</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="mi">10</span><span class="p">)</span>
</pre></div>
</div>
<p>Check the installation of cplex</p>
</dd></dl>

</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../KLdistance/index.html" class="btn btn-default float-right" title="KLdistance" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../geometricFBA/index.html" class="btn btn-default" title="geometricFBA" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2018, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/gh/mathjax/MathJax@2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  


  
  
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>
