<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="generator" content="MATLAB R2016a"><meta http-equiv="X-UA-Compatible" content="IE=edge,IE=9,chrome=1"><title>OptGene Tutorial</title><style type="text/css">
* {margin: 0; padding: 0;}
body {text-align: start; line-height: 17.2339992523193px; min-height: 0px; white-space: normal; color: rgb(0, 0, 0); font-family: Consolas, Inconsolata, Menlo, monospace; font-style: normal; font-size: 14px; font-weight: normal; text-decoration: none; white-space: normal; }
h1, h2 {font-weight: normal;}
.content { padding: 30px; }

.S0 { margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S1 { line-height: 26.3999996185303px; min-height: 24px; white-space: pre-wrap; color: rgb(213, 80, 0); font-family: Helvetica, Arial, sans-serif; font-size: 22px; white-space: pre-wrap; margin-left: 4px; margin-top: 3px; margin-bottom: 15px; margin-right: 10px;  }
.S2 { min-height: 0px; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S3 { line-height: 20.576000213623px; min-height: 20px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-size: 16px; font-weight: bold; white-space: pre-wrap; margin-left: 4px; margin-top: 15px; margin-bottom: 9px; margin-right: 10px;  }
.S4 { line-height: 21px; min-height: 17px; white-space: pre-wrap; font-family: Helvetica, Arial, sans-serif; white-space: pre-wrap; margin-left: 4px; margin-top: 2px; margin-bottom: 9px; margin-right: 10px;  }
.S5 { line-height: 20.576000213623px; min-height: 20px; white-space: pre-wrap; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-size: 16px; font-weight: bold; white-space: pre-wrap; margin-left: 4px; margin-top: 3px; margin-bottom: 9px; margin-right: 10px;  }
.S6 { font-family: Helvetica, Arial, sans-serif; margin-left: 0px; margin-top: 10px; margin-bottom: 20px; margin-right: 0px;  }
.S7 { text-align: left; line-height: 21px; white-space: pre-wrap; white-space: pre-wrap; margin-left: 56px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S8 { min-height: 0px; font-weight: bold; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S9 { margin-left: 3px; margin-top: 10px; margin-bottom: 4px; margin-right: 3px;  }
.S10 { line-height: 15.5926666259766px; min-height: 18px; white-space: nowrap; font-size: 12.6666669845581px; white-space: nowrap; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S11 { line-height: 15.5926675796509px; min-height: 0px; white-space: pre; color: rgb(0, 0, 255); font-size: 12.6666679382324px; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S12 { line-height: 15.5926675796509px; min-height: 0px; white-space: pre; font-size: 12.6666679382324px; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S13 { line-height: 15.5926675796509px; min-height: 0px; white-space: pre; font-size: 12.6666679382324px; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S14 { line-height: 15.5926675796509px; min-height: 0px; white-space: pre; color: rgb(34, 139, 34); font-size: 12.6666679382324px; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S15 { line-height: 15.5926675796509px; min-height: 0px; white-space: pre; color: rgb(160, 32, 240); font-size: 12.6666679382324px; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S16 { line-height: 15.5926675796509px; min-height: 0px; white-space: pre; color: rgb(0, 0, 255); font-size: 12.6666679382324px; white-space: pre; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 45px;  }
.S17 { color: rgb(64, 64, 64); margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }
.S18 { margin-left: 3px; margin-top: 10px; margin-bottom: 10px; margin-right: 3px;  }
.S19 { line-height: 21px; min-height: 17px; white-space: pre-wrap; font-family: Helvetica, Arial, sans-serif; white-space: pre-wrap; margin-left: 4px; margin-top: 10px; margin-bottom: 9px; margin-right: 10px;  }
.S20 { min-height: 0px; font-style: italic; margin-left: 0px; margin-top: 0px; margin-bottom: 0px; margin-right: 0px;  }

.LineNodeBlock {margin: 10px 0 10px 0; background-color: #F7F7F7;}
.LineNodeBlock+.paragraphNode {margin-top: 10px;}
.lineNode {padding-left: 10px; border-left: 1px solid #E9E9E9; border-right: 1px solid #E9E9E9;}
.inlineWrapper:first-child .lineNode,.inlineWrapper.outputs+.inlineWrapper .lineNode {padding-top: 5px; border-top: 1px solid #E9E9E9;}
.inlineWrapper:last-child .lineNode,.inlineWrapper.outputs .lineNode {padding-bottom: 5px; border-bottom: 1px solid #E9E9E9;}
.lineNode .textBox {white-space: pre;}
.outputGroup {margin: 2px 0 2px 0; padding: 2px 2px 2px 4px;}
.outputRegion {}
.outputParagraph {color: rgba(64, 64, 64, 1); padding: 10px 0 6px 17px; background: white; overflow-x: hidden;}
.inlineWrapper:last-child .outputParagraph {border-bottom-left-radius: 4px; border-bottom-right-radius: 4px;}
.outputParagraph:empty {margin: 0;}
.inlineElement .symbolicElement {margin-top: 1px; margin-bottom: 1px;}
.embeddedOutputsSymbolicElement .MathEquation {margin-top: 4px; margin-bottom: 4px;}
.embeddedOutputsSymbolicElement .MathEquation.displaySymbolicElement {margin-left: 15px;}
.embeddedOutputsSymbolicElement .MathEquation.inlineSymbolicElement {}
.embeddedOutputsSymbolicElement {overflow-x: auto; overflow-y: hidden;}
.embeddedOutputsSymbolicElement { overflow: initial !important;}
.embeddedOutputsTextElement,.embeddedOutputsVariableStringElement {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px; white-space: pre; word-wrap: initial; min-height: 18px; max-height: 250px; overflow: auto;}
.textElement,.rtcDataTipElement .textElement {padding-top: 3px;}
.embeddedOutputsTextElement.inlineElement,.embeddedOutputsVariableStringElement.inlineElement {}
.inlineElement .textElement {}
.embeddedOutputsTextElement,.embeddedOutputsVariableStringElement { max-height: none !important; overflow: initial !important;}
.veSpecifier {}
.veContainer {}
.veSpecifierBox {height: 400px; width: 500px;}
.veSpecifier .veTable {padding-top: 3px; padding-bottom: 4px;}
.veSpecifierBox .veSpecifier .veContainer {position: relative; width: 100%; height: 370px;}
.veSpecifierBox .dijitDialogPaneContent {width: 97% !important; height: 88% !important;}
.veSpecifier .veTable .rowHeadersWrapper {padding-bottom: 0;}
.veSpecifier .veTable .scroller .variableEditorRenderers {padding-right: 3px; -webkit-user-select: none; -moz-user-select: none; -ms-user-select: none;}
.veSpecifier .veTable .topHeaderWrapper,.veSpecifier .veTable .bottomRowHeaderWrapper {visibility: hidden; z-index: 0;}
.veMetaSummary {font-style: italic;}
.veSpecifier .veTable .scroller {overflow: hidden;}
.veSpecifier .veTable:hover .scroller {overflow: auto;}
.veSpecifier .veVariableName,.veSpecifier .veDimensionFont {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px;}
.veSpecifier .veVariableName {padding-top: 2px;}
.veSpecifier .veDimensionFont {font-style: italic; color: #9A9A9A;}
.veSpecifier .scroller::-webkit-scrollbar-track {background-color: white;}
.veSpecifier .scroller::-webkit-scrollbar-corner {background-color: white;}
.veSpecifier .veTable .topRowHeaderWrapper {border: none; background-color: #F8F9FA;}
.veSpecifier .mw_type_ListBase.showCellBorders,.veSpecifier .veTable .topHeaderWrapper,.veSpecifier .veTable .bottomRowHeaderWrapper,.veSpecifier .veTable .verticalScrollSpacer,.veSpecifier .veTable .horizontalScrollSpacer {border: none;}
.veSpecifier .veTable .dataScrollerNode {border: 1px solid #BFBFBF;}
.veSpecifier .veTable .columnHeaderNode,.veSpecifier .veTable .rowHeaderNode,.veSpecifier .veTable .dataBody {font-family: Arial; font-size: 13px;}
.veSpecifier .veTable .columnHeaderNode,.veSpecifier .veTable .rowHeaderNode {color: #7F7F7F;}
.veSpecifier .veTable .dataBody {color: #000000;}
.veSpecifier .veTable .columnHeaderNode .cell .drag,.veSpecifier .veTable .columnHeaderNode .cell .header,.veSpecifier .veTable .topHeaderWrapper,.veSpecifier .veTable .bottomRowHeaderWrapper {background-color: #F8F9FA;}
.veSpecifier .veTable .columnHeaderNode .cell .dragBorder {border-right: 1px solid #F8F9FA;}
.veSpecifier .veTable .rowHeaderNode .cell {padding-top: 3px; text-align: center; border-bottom: 1px solid #F8F9FA;}
.veSpecifier .veTable .dataBody .cell .plainText {text-align: right;}
.veSpecifier .veTable .dataBody .row .cell {border-bottom: 1px solid #E9E9E9; border-right: 1px solid #E9E9E9;}
.embeddedOutputsVariableElement {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px; white-space: pre-wrap; word-wrap: break-word; min-height: 18px; max-height: 250px; overflow: auto;}
.variableElement {}
.embeddedOutputsVariableElement.inlineElement {}
.inlineElement .variableElement {}
.variableNameElement {margin-bottom: 3px; display: inline-block;}
.variableValue { width: 100% !important; }
.embeddedOutputsMatrixElement {min-height: 18px; box-sizing: border-box; font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px;}
.embeddedOutputsMatrixElement .matrixElement,.rtcDataTipElement .matrixElement {position: relative;}
.matrixElement .variableValue,.rtcDataTipElement .matrixElement .variableValue {white-space: pre; display: inline-block; vertical-align: top; overflow: hidden;}
.embeddedOutputsMatrixElement.inlineElement {}
.embeddedOutputsMatrixElement.inlineElement .topHeaderWrapper {display: none;}
.embeddedOutputsMatrixElement.inlineElement .veTable .body {padding-top: 0 !important; max-height: 100px;}
.inlineElement .matrixElement {max-height: 300px;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer,.rtcDataTipElement .matrixElement .valueContainer {white-space: nowrap; margin-bottom: 3px;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer .horizontalEllipsis.hide,.embeddedOutputsMatrixElement .matrixElement .verticalEllipsis.hide,.rtcDataTipElement .matrixElement .valueContainer .horizontalEllipsis.hide,.rtcDataTipElement .matrixElement .verticalEllipsis.hide {display: none;}
.embeddedOutputsMatrixElement .matrixElement .valueContainer .horizontalEllipsis {margin-bottom: -3px;}
.matrixElement { max-height: none !important; }
.dijitTooltipDialog .dijitTooltipContainer .dijitTooltipContents .alertPlugin-alertMessage {min-width: 12px; max-width: 400px; max-height: 300px; overflow: auto;}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar {width: 11px; height: 11px;}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-track {background-color: rgba(0, 0, 0, 0);}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-corner {background-color: rgba(0, 0, 0, 0);}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-thumb {border-radius: 7px; background-color: rgba(0, 0, 0, 0.1); border: 2px solid rgba(0, 0, 0, 0); background-clip: padding-box;}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-thumb:hover {background-color: rgba(0, 0, 0, 0.2);}
.dijitTooltipDialog .alertPlugin-alertMessage::-webkit-scrollbar-thumb {background-color: rgba(0, 0, 0, 0);}
.dijitTooltipDialog .alertPlugin-alertMessage:hover::-webkit-scrollbar-thumb {background-color: rgba(0, 0, 0, 0.1);}
.dijitTooltipDialog .alertPlugin-alertMessage:hover::-webkit-scrollbar-thumb:hover {background-color: rgba(0, 0, 0, 0.2);}
.alertPlugin-alertLine {position: absolute; display: initial; width: 40px; text-align: right; cursor: text;}
.alertPlugin-onTextLine {visibility: hidden;}
.alertPlugin-hasTooltip .alertPlugin-warningImg,.alertPlugin-hasTooltip .alertPlugin-errorImg {cursor: pointer;}
.alertPlugin-alertLine .alertPlugin-warningElement,.alertPlugin-alertLine .alertPlugin-errorElement {display: inline-block; margin-right: 4px;}
.alertPlugin-isStale {-webkit-filter: opacity(0.4) grayscale(80%); filter: opacity(0.4) grayscale(80%);}
.diagnosticMessage-wrapper {font-family: Consolas, Inconsolata, Menlo, monospace; font-size: 12px;}
.diagnosticMessage-wrapper.diagnosticMessage-warningType {color: rgb(255,100,0);}
.diagnosticMessage-wrapper.diagnosticMessage-warningType a {color: rgb(255,100,0); text-decoration: underline;}
.diagnosticMessage-wrapper.diagnosticMessage-errorType {color: rgb(230,0,0);}
.diagnosticMessage-wrapper.diagnosticMessage-errorType a {color: rgb(230,0,0); text-decoration: underline;}
.diagnosticMessage-wrapper .diagnosticMessage-messagePart {white-space: pre-wrap;}
.diagnosticMessage-wrapper .diagnosticMessage-stackPart {white-space: pre;}
.embeddedOutputsWarningElement{min-height: 18px; max-height: 250px; overflow: auto;}
.embeddedOutputsWarningElement.inlineElement {}
.embeddedOutputsErrorElement {min-height: 18px; max-height: 250px; overflow: auto;}
.embeddedOutputsErrorElement.inlineElement {}
.variableNameElement .headerElement {color: rgb(179, 179, 179); font-style: italic;}
.variableNameElement .headerElement .headerDataType {color: rgb(147, 176, 230);}
</style></head><body><div class = "content"><div class = 'SectionBlock containment active'><h1 class = "S1"><span class = "S2">OptGene Tutorial</span></h1><h2 class = "S3"><span class = "S2">Author: Sebastián N. Mendoza,  Center for Mathematical Modeling, University of Chile. snmendoz@uc.cl</span></h2><h2 class = "S3"><span class = "S2">Reviewer(s): Sylvian Arreckx</span></h2><h2 class = "S3"><span class = "S2">INTRODUCTION:</span></h2><div class = "S4"><span class = "S2">In this tutorial we will run optGene For a detailed description of the procedure, please see [1]. Briefly, the problem is to find a set of reactions of size "K" such that when these reactions are deleted from the model, the mutant created will produce a particular target of interest in a higher rate than the wild-type strain. </span></div><div class = "S4"><span class = "S2">For example, imagine that we would like to increase the production of succinate in Escherichia coli. Which are the knockouts needed to increase the production of succinate? We will approach this problem in this tutorial.</span></div></div><div class = "S0"></div><div class = 'SectionBlock containment'><h2 class = "S5"><span class = "S2">MATERIALS</span></h2><h2 class = "S3"><span class = "S2">EQUIPMENT</span></h2><ol class = "S6"><li class = "S7"><span class = "S0">MATLAB</span></li><li class = "S7"><span class = "S0">A solver for QP problems. For example, Gurobi. I encourage the users to use Gurobi since I've not obtained good results using glpk.</span></li></ol><h2 class = "S3"><span class = "S2">EQUIPMENT SETUP</span></h2><div class = "S4"><span class = "S2">Use changeCobraSolver to choose the solver for QP problems. </span></div></div><div class = "S0"></div><div class = 'SectionBlock containment'><h2 class = "S5"><span class = "S2">PROCEDURE</span></h2><div class = "S4"><span class = "S2">The proceduce consists on the following steps</span></div><div class = "S4"><span class = "S2">1) Define contraints (manual task)</span></div><div class = "S4"><span class = "S2">2) Select a list of reactions or genes (manual task). Reactions or genes in this list could be deleted. Elements that are not in the list will no be deleted. </span></div><div class = "S4"><span class = "S2">3) Define some (manual task)</span></div><div class = "S4"><span class = "S2">4) Run optGene. </span><span class = "S8">TIMING</span><span class = "S2">: This task should take from a few minutes to a few days, depending on the size of your reconstruction and the criterion for stoping optGene</span></div><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S11">global </span><span class = "S12">TUTORIAL_INIT_CB;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S11">if </span><span class = "S12">~isempty(TUTORIAL_INIT_CB) &amp;&amp; TUTORIAL_INIT_CB  ==  1</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    initCobraToolbox(false) </span><span class = "S14">% false, as we don't want to update</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    changeCobraSolver(</span><span class = "S15">'gurobi'</span><span class = "S13">, </span><span class = "S15">'all'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">fullPath = which (</span><span class = "S15">'tutorial_optGene.mlx'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">folder = fileparts(fullPath);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">cd(folder);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">threshold = 3; </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper outputs'><div class = "S10 lineNode"><span class = "S13">model = readCbModel(</span><span class = "S15">'iJO1366.mat'</span><span class = "S12">);</span></div><div class="outputParagraph" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 14px;"><div class="inlineElement embeddedOutputsTextElement" uid="0B0D8B0F" data-width="984" data-testid="output_0" style="width: 1014px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">Each model.subSystems{x} is a character array, and this format is retained.</div></div></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">biomass = </span><span class = "S15">'BIOMASS_Ec_iJO1366_core_53p95M'</span><span class = "S12">;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">%SETTING SPECIFIC CONSTRAINTS</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% prespecified amount of glucose uptake 10 mmol/grDW*hr</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_glc__D_e'</span><span class = "S13">, -10, </span><span class = "S15">'b'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% Unconstrained uptake routes for inorganic phosphate, sulfate and</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% ammonia</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_o2_e'</span><span class = "S13">, 0, </span><span class = "S15">'l'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_pi_e'</span><span class = "S13">, -1000, </span><span class = "S15">'l'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_so4_e'</span><span class = "S13">, -1000, </span><span class = "S15">'l'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_nh4_e'</span><span class = "S13">, -1000, </span><span class = "S15">'l'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% The optimization step could opt for or against the phosphotransferase</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% system, glucokinase, or both mechanisms for the uptake of glucose</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'GLCabcpp'</span><span class = "S13">, -1000, </span><span class = "S15">'l'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'GLCptspp'</span><span class = "S13">, -1000, </span><span class = "S15">'l'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'GLCabcpp'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'GLCptspp'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'GLCt2pp'</span><span class = "S13">, 0, </span><span class = "S15">'b'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% Secretion routes  for acetate, carbon dioxide, ethanol, formate, lactate</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% and succinate are enabled</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_ac_e'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_co2_e'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_etoh_e'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_for_e'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_lac__D_e'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeRxnBounds(model, </span><span class = "S15">'EX_succ_e'</span><span class = "S13">, 1000, </span><span class = "S15">'u'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% FINDING RATES IN WILD-TYPE</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% The follling rates are those calculated in the wild-type without any</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% mutation.</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% determine succinate production and growth rate before optimizacion</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">fbaWT = optimizeCbModel(model);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">growthRateWT = fbaWT.f;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">model = changeObjective(model, </span><span class = "S15">'EX_succ_e'</span><span class = "S12">); </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">fbaWTMin = optimizeCbModel(model, </span><span class = "S15">'min'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">fbaWTMax = optimizeCbModel(model, </span><span class = "S15">'max'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">minSuccFluxWT = fbaWTMin.f;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">maxSuccFluxWT = fbaWTMax.f;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">model = changeObjective(model, biomass);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper outputs'><div class = "S10 lineNode"><span class = "S13">fprintf(</span><span class = "S15">'The minimum and maximum production of succinate before optimization is %.1f and %.1f respectively\n'</span><span class = "S12">, minSuccFluxWT, maxSuccFluxWT);</span></div><div class="outputParagraph" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 14px;"><div class="inlineElement embeddedOutputsTextElement" uid="B3157698" data-width="984" data-testid="output_1" style="width: 1014px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">The minimum and maximum production of succinate before optimization is 0.0 and 15.8 respectively</div></div></div></div><div class = 'inlineWrapper outputs'><div class = "S10 lineNode"><span class = "S13">fprintf(</span><span class = "S15">'The growth rate before optimization is %.2f \n'</span><span class = "S12">, growthRateWT);</span></div><div class="outputParagraph" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 14px;"><div class="inlineElement embeddedOutputsTextElement" uid="3BFEAE8C" data-width="984" data-testid="output_2" style="width: 1014px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">The growth rate before optimization is 0.24 </div></div></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12"></span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% OPTGENE SETTING</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">selectedGeneList = {};</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S14">% use prespecified reactions. Faster option</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">selectedRxnList = {</span><span class = "S15">'GLCabcpp'</span><span class = "S13">; </span><span class = "S15">'GLCptspp'</span><span class = "S13">; </span><span class = "S15">'HEX1'</span><span class = "S13">; </span><span class = "S15">'PGI'</span><span class = "S13">; </span><span class = "S15">'PFK'</span><span class = "S13">; </span><span class = "S15">'FBA'</span><span class = "S13">; </span><span class = "S15">'TPI'</span><span class = "S13">; </span><span class = "S15">'GAPD'</span><span class = "S13">; </span><span class = "S15">'PGK'</span><span class = "S13">; </span><span class = "S15">'PGM'</span><span class = "S13">; </span><span class = "S15">'ENO'</span><span class = "S13">; </span><span class = "S15">'PYK'</span><span class = "S13">; </span><span class = "S15">'LDH_D'</span><span class = "S13">; </span><span class = "S15">'PFL'</span><span class = "S13">; </span><span class = "S15">'ALCD2x'</span><span class = "S13">; </span><span class = "S15">'PTAr'</span><span class = "S13">; </span><span class = "S15">'ACKr'</span><span class = "S13">; </span><span class = "S15">'G6PDH2r'</span><span class = "S13">; </span><span class = "S15">'PGL'</span><span class = "S13">; </span><span class = "S15">'GND'</span><span class = "S13">; </span><span class = "S15">'RPI'</span><span class = "S13">; </span><span class = "S15">'RPE'</span><span class = "S13">; </span><span class = "S15">'TKT1'</span><span class = "S13">; </span><span class = "S15">'TALA'</span><span class = "S13">; </span><span class = "S15">'TKT2'</span><span class = "S13">; </span><span class = "S15">'FUM'</span><span class = "S13">; </span><span class = "S15">'FRD2'</span><span class = "S13">; </span><span class = "S15">'SUCOAS'</span><span class = "S13">; </span><span class = "S15">'AKGDH'</span><span class = "S13">; </span><span class = "S15">'ACONTa'</span><span class = "S13">; </span><span class = "S15">'ACONTb'</span><span class = "S13">; </span><span class = "S15">'ICDHyr'</span><span class = "S13">; </span><span class = "S15">'CS'</span><span class = "S13">; </span><span class = "S15">'MDH'</span><span class = "S13">; </span><span class = "S15">'MDH2'</span><span class = "S13">; </span><span class = "S15">'MDH3'</span><span class = "S13">; </span><span class = "S15">'ACALD'</span><span class = "S12">};</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">genesByReaction = regexp(regexprep(model.grRules(ismember(model.rxns, selectedRxnList)), </span><span class = "S15">'\or|and|\(|\)'</span><span class = "S13">, </span><span class = "S15">''</span><span class = "S13">), </span><span class = "S15">'\  '</span><span class = "S13">, </span><span class = "S15">'split'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S11">for </span><span class = "S12">i = 1:length(genesByReaction)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    selectedGeneList = union(selectedGeneList, genesByReaction{i});</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S16">end</span></div></div></div></div><div class = "S0"></div><div class = 'SectionBlock containment'><div class = "S4"><span class = "S8">I) SUCCINATE OVERPRODUCTION</span></div><div class = "S4"><span class = "S8">EXAMPLE 1: finding reaction knockouts sets of large 2 or less, using a limit of time to stop optGene</span></div><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper outputs'><div class = "S10 lineNode"><span class = "S13">fprintf(</span><span class = "S15">'\n...EXAMPLE 1: Finding optGene sets\n\n'</span><span class = "S12">)</span></div><div class="outputParagraph" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 14px;"><div class="inlineElement embeddedOutputsTextElement" uid="82D30DEC" data-width="984" data-testid="output_3" style="width: 1014px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">...EXAMPLE 1: Finding optGene sets</div></div></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">previousSolutions = cell(10, 1);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">contPreviousSolutions = 1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">nIter = 0;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S11">while </span><span class = "S12">nIter &lt; threshold</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    fprintf(</span><span class = "S15">'...Performing optGene analysis...\n'</span><span class = "S12">)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S14">%optGene algorithm is run with the following options: target: 'EX_lac__D_e'</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    [~, ~, ~, optGeneSol] = optGene(model, </span><span class = "S15">'EX_succ_e'</span><span class = "S13">, </span><span class = "S15">'EX_glc__D_e'</span><span class = "S13">, selectedGeneList, </span><span class = "S15">'MaxKOs'</span><span class = "S13">, 2, </span><span class = "S15">'TimeLimit'</span><span class = "S12">, 120);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    SET_M1 = optGeneSol.geneList;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S11">if </span><span class = "S12">~isempty(SET_M1)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">        previousSolutions{contPreviousSolutions} = SET_M1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">        contPreviousSolutions = contPreviousSolutions + 1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S14">%printing results</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'optGene found a knockout set of large %d composed by '</span><span class = "S12">, length(SET_M1));</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S11">for </span><span class = "S12">j = 1:length(SET_M1)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S11">if </span><span class = "S12">j == 1</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">                fprintf(</span><span class = "S15">'%s '</span><span class = "S12">,SET_M1{j});</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S11">elseif </span><span class = "S12">j == length(SET_M1)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">                fprintf(</span><span class = "S15">'and %s'</span><span class = "S12">,SET_M1{j});</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S16">else</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">                fprintf(</span><span class = "S15">', %s '</span><span class = "S12">,SET_M1{j});</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'...Performing coupling analysis...\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        [type, maxGrowth, maxProd, minProd] = analyzeOptKnock(model, optGeneSol.geneList, </span><span class = "S15">'EX_succ_e'</span><span class = "S12">, biomass, 1);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'The solution is of type: %s\n'</span><span class = "S12">,type);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'The maximun growth rate after optimizacion is %.2f\n'</span><span class = "S12">, maxGrowth);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'The maximun and minimun production of succinate after optimization is %.2f and %.2f, respectively \n\n'</span><span class = "S12">, minProd, maxProd);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">        </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S16">else</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S11">if </span><span class = "S12">nIter  ==  1</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            fprintf(</span><span class = "S15">'optGene was not able to found an optGene set\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S16">else</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            fprintf(</span><span class = "S15">'optGene was not able to found additional optGene sets\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S11">break</span><span class = "S12">;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    nIter = nIter + 1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    </span></div></div><div class = 'inlineWrapper outputs'><div class = "S10 lineNode"><span class = "S16">end</span></div><div class="outputParagraph" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 14px;"><div class="inlineElement embeddedOutputsTextElement" uid="02674712" data-width="984" data-testid="output_4" style="width: 1014px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">...Performing optGene analysis...</div></div><div class="inlineElement embeddedOutputsTextElement" uid="2410438A" data-width="984" data-testid="output_5" style="width: 1014px; white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: pre; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">assuming list is genes</div></div><div class="inlineElement embeddedOutputsErrorElement" uid="F94E46BC" data-width="984" data-testid="output_6" style="width: 1014px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 14px;"><div class="diagnosticMessage-wrapper diagnosticMessage-errorType" style="white-space: normal; font-style: normal; color: rgb(230, 0, 0); font-size: 12px;"><div class="diagnosticMessage-messagePart" style="white-space: pre-wrap; font-style: normal; color: rgb(230, 0, 0); font-size: 12px;">'gaoptimset' requires Global Optimization Toolbox.</div><div class="diagnosticMessage-stackPart" style="white-space: pre; font-style: normal; color: rgb(230, 0, 0); font-size: 12px;"><br style="white-space: pre; font-style: normal; color: rgb(230, 0, 0); font-size: 12px;">Error in optGene (line 143)<br style="white-space: pre; font-style: normal; color: rgb(230, 0, 0); font-size: 12px;">options = gaoptimset(...</div></div></div></div></div></div></div><div class = "S0"></div><div class = 'SectionBlock containment'><div class = "S4"><span class = "S8">EXAMPLE 2: finding reaction knockouts sets of large 2 or less, using the number of generations to stop optGene</span></div><div class = 'LineNodeBlock contiguous'><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">fprintf(</span><span class = "S15">'\n...EXAMPLE 2: Finding optGene sets\n\n'</span><span class = "S12">)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">previousSolutions = cell(10, 1);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">contPreviousSolutions = 1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">nIter = 0;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S11">while </span><span class = "S12">nIter &lt; threshold</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    fprintf(</span><span class = "S15">'...Performing optGene analysis...\n'</span><span class = "S12">)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S14">%optGene algorithm is run with the following options: target: 'EX_lac__D_e'</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    [~, ~, ~, optGeneSol] = optGene(model, </span><span class = "S15">'EX_succ_e'</span><span class = "S13">, </span><span class = "S15">'EX_glc__D_e'</span><span class = "S13">, selectedGeneList, </span><span class = "S15">'MaxKOs'</span><span class = "S13">, 2, </span><span class = "S15">'Generations'</span><span class = "S12">, 20);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    SET_M1 = optGeneSol.geneList;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S11">if </span><span class = "S12">~isempty(SET_M1)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">        previousSolutions{contPreviousSolutions} = SET_M1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">        contPreviousSolutions = contPreviousSolutions + 1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S14">%printing results</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'optGene found a knockout set of large %d composed by '</span><span class = "S12">, length(SET_M1));</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S11">for </span><span class = "S12">j = 1:length(SET_M1)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S11">if </span><span class = "S12">j == 1</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">                fprintf(</span><span class = "S15">'%s '</span><span class = "S12">,SET_M1{j});</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S11">elseif </span><span class = "S12">j == length(SET_M1)</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">                fprintf(</span><span class = "S15">'and %s'</span><span class = "S12">,SET_M1{j});</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S16">else</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">                fprintf(</span><span class = "S15">', %s '</span><span class = "S12">,SET_M1{j});</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'...Performing coupling analysis...\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        [type, maxGrowth, maxProd, minProd] = analyzeOptKnock(model, optGeneSol.geneList, </span><span class = "S15">'EX_succ_e'</span><span class = "S12">, biomass, 1);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'The solution is of type: %s\n'</span><span class = "S12">,type);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'The maximun growth rate after optimizacion is %.2f\n'</span><span class = "S12">, maxGrowth);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        fprintf(</span><span class = "S15">'The maximun and minimun production of succinate after optimization is %.2f and %.2f, respectively \n\n'</span><span class = "S12">, minProd, maxProd);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">        </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S16">else</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S11">if </span><span class = "S12">nIter  ==  1</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            fprintf(</span><span class = "S15">'optGene was not able to found an optGene set\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S16">else</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">            fprintf(</span><span class = "S15">'optGene was not able to found additional optGene sets\n'</span><span class = "S12">);</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">        </span><span class = "S11">break</span><span class = "S12">;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S13">    </span><span class = "S16">end</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    nIter = nIter + 1;</span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S12">    </span></div></div><div class = 'inlineWrapper'><div class = "S10 lineNode"><span class = "S16">end</span></div></div></div><div class = "S19"><span class = "S2"> </span></div></div><div class = "S0"></div><div class = 'SectionBlock containment'><h2 class = "S5"><span class = "S2">TIMING</span></h2><ol class = "S6"><li class = "S7"><span class = "S0">EXAMPLE 1: ~ 6 minutes (2 minutes per iteration)</span></li><li class = "S7"><span class = "S0">EXAMPLE 2: ~ 7 minutes (2-3 minutes per iteration)</span></li></ol></div><div class = "S0"></div><div class = 'SectionBlock containment'><h2 class = "S5"><span class = "S2">TROUBLESHOOTING</span></h2><div class = "S4"><span class = "S2">1) problem: "optGene didn't find any set"</span></div><div class = "S4"><span class = "S2">possible reason: probably, the limit of time or the number of generations has not been enough. Another explination is that the solver is not suited for solving optGene</span></div><div class = "S4"><span class = "S2">solution: Try with a higher number for inputs "TimeLimit" of "Generations" or using another solver. </span></div><div class = "S4"><span class = "S2">2) problem: "I got an error when running optGene"</span></div><div class = "S4"><span class = "S2">possible reason: the solver is not suited for solving optGene</span></div><div class = "S4"><span class = "S2">solution: Try with another solver</span></div></div><div class = "S0"></div><div class = 'SectionBlock containment'><h2 class = "S5"><span class = "S2">ANTICIPATED RESULTS</span></h2><div class = "S4"><span class = "S2">The optGene algorithm will find sets of reactions that should increase the production of your target when they are deleted from the network. Since optGene is based on a genetic algorithm, the solutions found could vary between different runnings, even though the algorithm has been executed with the same input parameters. It is possible that optGene don't find a set of knockouts because the runtime is too short or because the number of generations is too small. In those cases try to increases those input variables. </span></div></div><div class = "S0"></div><div class = 'SectionBlock containment'><h2 class = "S5"><span class = "S2">References</span></h2><div class = "S4"><span class = "S2">[1] Patil, K. R., Rocha, I., Förster, J., &amp; Nielsen, J. (2005). Evolutionary programming as a platform for in silico metabolic engineering.</span><span class = "S2"> </span><span class = "S20">BMC bioinformatics</span><span class = "S2">,</span><span class = "S2"> </span><span class = "S20">6</span><span class = "S2">(1), 308.</span></div><div class = "S4"><span class = "S2">[2] Orth, J. D., Conrad, T. M., Na, J., Lerman, J. A., Nam, H., Feist, A. M., &amp; Palsson, B. Ø. (2011). A comprehensive genome‐scale reconstruction of Escherichia coli metabolism—2011.</span><span class = "S2"> </span><span class = "S20">Molecular systems biology</span><span class = "S2">,</span><span class = "S2"> </span><span class = "S20">7</span><span class = "S2">(1), 535.</span></div></div></div>
<!-- 
##### SOURCE BEGIN #####
%% OptGene Tutorial
%% Author: Sebastián N. Mendoza,  Center for Mathematical Modeling, University of Chile. snmendoz@uc.cl
%% *Reviewer(s): Sylvian Arreckx*
%% *INTRODUCTION:*
% In this tutorial we will run optGene For a detailed description of the procedure, 
% please see [1]. Briefly, the problem is to find a set of reactions of size "K" 
% such that when these reactions are deleted from the model, the mutant created 
% will produce a particular target of interest in a higher rate than the wild-type 
% strain. 
% 
% For example, imagine that we would like to increase the production of succinate 
% in Escherichia coli. Which are the knockouts needed to increase the production 
% of succinate? We will approach this problem in this tutorial.
%% MATERIALS
%% EQUIPMENT
% # MATLAB
% # A solver for QP problems. For example, Gurobi. I encourage the users to 
% use Gurobi since I've not obtained good results using glpk.
%% *EQUIPMENT SETUP*
% Use changeCobraSolver to choose the solver for QP problems. 
%% PROCEDURE
% The proceduce consists on the following steps
% 
% 1) Define contraints (manual task)
% 
% 2) Select a list of reactions or genes (manual task). Reactions or genes 
% in this list could be deleted. Elements that are not in the list will no be 
% deleted. 
% 
% 3) Define some (manual task)
% 
% 4) Run optGene. *TIMING*: This task should take from a few minutes to a 
% few days, depending on the size of your reconstruction and the criterion for 
% stoping optGene
%%
global TUTORIAL_INIT_CB;
if ~isempty(TUTORIAL_INIT_CB) && TUTORIAL_INIT_CB  ==  1
    initCobraToolbox(false) % false, as we don't want to update
    changeCobraSolver('gurobi', 'all');
end

fullPath = which ('tutorial_optGene.mlx');
folder = fileparts(fullPath);
cd(folder);

threshold = 3; 

model = readCbModel('iJO1366.mat');
biomass = 'BIOMASS_Ec_iJO1366_core_53p95M';

%SETTING SPECIFIC CONSTRAINTS
% prespecified amount of glucose uptake 10 mmol/grDW*hr
model = changeRxnBounds(model, 'EX_glc__D_e', -10, 'b');

% Unconstrained uptake routes for inorganic phosphate, sulfate and
% ammonia
model = changeRxnBounds(model, 'EX_o2_e', 0, 'l');
model = changeRxnBounds(model, 'EX_pi_e', -1000, 'l');
model = changeRxnBounds(model, 'EX_so4_e', -1000, 'l');
model = changeRxnBounds(model, 'EX_nh4_e', -1000, 'l');

% The optimization step could opt for or against the phosphotransferase
% system, glucokinase, or both mechanisms for the uptake of glucose
model = changeRxnBounds(model, 'GLCabcpp', -1000, 'l');
model = changeRxnBounds(model, 'GLCptspp', -1000, 'l');
model = changeRxnBounds(model, 'GLCabcpp', 1000, 'u');
model = changeRxnBounds(model, 'GLCptspp', 1000, 'u');
model = changeRxnBounds(model, 'GLCt2pp', 0, 'b');

% Secretion routes  for acetate, carbon dioxide, ethanol, formate, lactate
% and succinate are enabled
model = changeRxnBounds(model, 'EX_ac_e', 1000, 'u');
model = changeRxnBounds(model, 'EX_co2_e', 1000, 'u');
model = changeRxnBounds(model, 'EX_etoh_e', 1000, 'u');
model = changeRxnBounds(model, 'EX_for_e', 1000, 'u');
model = changeRxnBounds(model, 'EX_lac__D_e', 1000, 'u');
model = changeRxnBounds(model, 'EX_succ_e', 1000, 'u');

% FINDING RATES IN WILD-TYPE
% The follling rates are those calculated in the wild-type without any
% mutation.

% determine succinate production and growth rate before optimizacion
fbaWT = optimizeCbModel(model);
growthRateWT = fbaWT.f;

model = changeObjective(model, 'EX_succ_e'); 
fbaWTMin = optimizeCbModel(model, 'min');
fbaWTMax = optimizeCbModel(model, 'max');
minSuccFluxWT = fbaWTMin.f;
maxSuccFluxWT = fbaWTMax.f;

model = changeObjective(model, biomass);

fprintf('The minimum and maximum production of succinate before optimization is %.1f and %.1f respectively\n', minSuccFluxWT, maxSuccFluxWT);
fprintf('The growth rate before optimization is %.2f \n', growthRateWT);

% OPTGENE SETTING
selectedGeneList = {};
% use prespecified reactions. Faster option
selectedRxnList = {'GLCabcpp'; 'GLCptspp'; 'HEX1'; 'PGI'; 'PFK'; 'FBA'; 'TPI'; 'GAPD'; 'PGK'; 'PGM'; 'ENO'; 'PYK'; 'LDH_D'; 'PFL'; 'ALCD2x'; 'PTAr'; 'ACKr'; 'G6PDH2r'; 'PGL'; 'GND'; 'RPI'; 'RPE'; 'TKT1'; 'TALA'; 'TKT2'; 'FUM'; 'FRD2'; 'SUCOAS'; 'AKGDH'; 'ACONTa'; 'ACONTb'; 'ICDHyr'; 'CS'; 'MDH'; 'MDH2'; 'MDH3'; 'ACALD'};
genesByReaction = regexp(regexprep(model.grRules(ismember(model.rxns, selectedRxnList)), '\or|and|\(|\)', ''), '\  ', 'split');
for i = 1:length(genesByReaction)
    selectedGeneList = union(selectedGeneList, genesByReaction{i});
end
%% 
% *I) SUCCINATE OVERPRODUCTION*
% 
% *EXAMPLE 1: finding reaction knockouts sets of large 2 or less, using a 
% limit of time to stop optGene*
%%
fprintf('\n...EXAMPLE 1: Finding optGene sets\n\n')
previousSolutions = cell(10, 1);
contPreviousSolutions = 1;
nIter = 0;
while nIter < threshold
    fprintf('...Performing optGene analysis...\n')
    %optGene algorithm is run with the following options: target: 'EX_lac__D_e'
    [~, ~, ~, optGeneSol] = optGene(model, 'EX_succ_e', 'EX_glc__D_e', selectedGeneList, 'MaxKOs', 2, 'TimeLimit', 120);
    
    SET_M1 = optGeneSol.geneList;
    
    if ~isempty(SET_M1)
        previousSolutions{contPreviousSolutions} = SET_M1;
        contPreviousSolutions = contPreviousSolutions + 1;
        %printing results
        fprintf('optGene found a knockout set of large %d composed by ', length(SET_M1));
        for j = 1:length(SET_M1)
            if j == 1
                fprintf('%s ',SET_M1{j});
            elseif j == length(SET_M1)
                fprintf('and %s',SET_M1{j});
            else
                fprintf(', %s ',SET_M1{j});
            end
        end
        fprintf('\n');
        fprintf('...Performing coupling analysis...\n');
        [type, maxGrowth, maxProd, minProd] = analyzeOptKnock(model, optGeneSol.geneList, 'EX_succ_e', biomass, 1);
        fprintf('The solution is of type: %s\n',type);
        fprintf('The maximun growth rate after optimizacion is %.2f\n', maxGrowth);
        fprintf('The maximun and minimun production of succinate after optimization is %.2f and %.2f, respectively \n\n', minProd, maxProd);
        
    else
        if nIter  ==  1
            fprintf('optGene was not able to found an optGene set\n');
        else
            fprintf('optGene was not able to found additional optGene sets\n');
        end
        break;
    end
    nIter = nIter + 1;
    
end
%% 
% *EXAMPLE 2: finding reaction knockouts sets of large 2 or less, using 
% the number of generations to stop optGene*
%%
fprintf('\n...EXAMPLE 2: Finding optGene sets\n\n')
previousSolutions = cell(10, 1);
contPreviousSolutions = 1;
nIter = 0;
while nIter < threshold
    fprintf('...Performing optGene analysis...\n')
    %optGene algorithm is run with the following options: target: 'EX_lac__D_e'
    [~, ~, ~, optGeneSol] = optGene(model, 'EX_succ_e', 'EX_glc__D_e', selectedGeneList, 'MaxKOs', 2, 'Generations', 20);
    
    SET_M1 = optGeneSol.geneList;
    
    if ~isempty(SET_M1)
        previousSolutions{contPreviousSolutions} = SET_M1;
        contPreviousSolutions = contPreviousSolutions + 1;
        %printing results
        fprintf('optGene found a knockout set of large %d composed by ', length(SET_M1));
        for j = 1:length(SET_M1)
            if j == 1
                fprintf('%s ',SET_M1{j});
            elseif j == length(SET_M1)
                fprintf('and %s',SET_M1{j});
            else
                fprintf(', %s ',SET_M1{j});
            end
        end
        fprintf('\n');
        fprintf('...Performing coupling analysis...\n');
        [type, maxGrowth, maxProd, minProd] = analyzeOptKnock(model, optGeneSol.geneList, 'EX_succ_e', biomass, 1);
        fprintf('The solution is of type: %s\n',type);
        fprintf('The maximun growth rate after optimizacion is %.2f\n', maxGrowth);
        fprintf('The maximun and minimun production of succinate after optimization is %.2f and %.2f, respectively \n\n', minProd, maxProd);
        
    else
        if nIter  ==  1
            fprintf('optGene was not able to found an optGene set\n');
        else
            fprintf('optGene was not able to found additional optGene sets\n');
        end
        break;
    end
    nIter = nIter + 1;
    
end
%% 
%  
%% TIMING
% # EXAMPLE 1: ~ 6 minutes (2 minutes per iteration)
% # EXAMPLE 2: ~ 7 minutes (2-3 minutes per iteration)
%% TROUBLESHOOTING
% 1) problem: "optGene didn't find any set"
% 
% possible reason: probably, the limit of time or the number of generations 
% has not been enough. Another explination is that the solver is not suited for 
% solving optGene
% 
% solution: Try with a higher number for inputs "TimeLimit" of "Generations" 
% or using another solver. 
% 
% 2) problem: "I got an error when running optGene"
% 
% possible reason: the solver is not suited for solving optGene
% 
% solution: Try with another solver
%% ANTICIPATED RESULTS
% The optGene algorithm will find sets of reactions that should increase the 
% production of your target when they are deleted from the network. Since optGene 
% is based on a genetic algorithm, the solutions found could vary between different 
% runnings, even though the algorithm has been executed with the same input parameters. 
% It is possible that optGene don't find a set of knockouts because the runtime 
% is too short or because the number of generations is too small. In those cases 
% try to increases those input variables. 
%% References
% [1] Patil, K. R., Rocha, I., Förster, J., & Nielsen, J. (2005). Evolutionary 
% programming as a platform for in silico metabolic engineering. _BMC bioinformatics_, 
% _6_(1), 308.
% 
% [2] Orth, J. D., Conrad, T. M., Na, J., Lerman, J. A., Nam, H., Feist, 
% A. M., & Palsson, B. Ø. (2011). A comprehensive genome‐scale reconstruction 
% of Escherichia coli metabolism—2011. _Molecular systems biology_, _7_(1), 535.
##### SOURCE END #####
--></body></html>