

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Metabotools &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../index.html"/>
        <link rel="up" title="Data integration" href="../index.html"/>
        <link rel="next" title="Transcriptomics" href="../transcriptomics/index.html"/>
        <link rel="prev" title="c13solver" href="../fluxomics/c13solver/index.html"/> 

  
  <script src="../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="https://king.nuigalway.ie/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../_static/../" + item.website_url + "' ><img src='../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../index.html">Functions</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../analysis/index.html">Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../base/index.html">Base</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../index.html">Data integration</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../chemoInformatics/index.html">ChemoInformatics</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fluxomics/index.html">Fluxomics</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Metabotools</a></li>
<li class="toctree-l3"><a class="reference internal" href="../transcriptomics/index.html">Transcriptomics</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contact.html">Contact</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../contents.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Functions</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Data integration</a></li>
      <li class="breadcrumb-item active">Metabotools</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="metabotools">
<span id="id1"></span><h1>Metabotools<a class="headerlink" href="#metabotools" title="Permalink to this heading">¶</a></h1>
<span class="target" id="module-src.dataIntegration.metabotools"></span><dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.analyzeSingleGeneDeletion">
<span class="sig-name descname"><span class="pre">analyzeSingleGeneDeletion</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">ResultsAllCellLines</span></em>, <em class="sig-param"><span class="pre">path</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">cutoff</span></em>, <em class="sig-param"><span class="pre">OverViewResults</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.analyzeSingleGeneDeletion" title="Permalink to this definition">¶</a></dt>
<dd><p>This function conducts a single gene deletion analysis and provides
statistcs about the frequency of essential genes, and partial effects and
no effects across the unique set of genes and across the models in
ResultsAllCellLines defined by samples.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[genes, ResultsAllCellLines, OverViewResults] = analyzeSingleGeneDeletion(ResultsAllCellLines, path, samples, cutoff, OverViewResults)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>ResultsAllCellLines:  Structure containing the models (the two models can be the same or pruned and starting model) of the format</p>
<blockquote>
<div><ul class="simple">
<li><p>ResultsAllCellLines.samples.modelMin</p></li>
<li><p>ResultsAllCellLines.samples.modelPruned</p></li>
</ul>
</div></blockquote>
<p>path:                 Path where picture of heatmap is to be saved
samples:              Name of samples
cutoff:               Tolerance for small deviation from the optimal value whic his used to distinguish partial effects</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>OverViewResults:      Overview of results of the model building and analysis (generated by the function setQuantConstraints)</p>
</dd>
<dt>OUTPUTS:</dt><dd><dl class="simple">
<dt>genes:                Table that lists unique gene, category, number</dt><dd><p>of ‘no effect’, number of ‘KO’, number of ‘partial effect’, and
number of models where the gene is absent ‘gene absent’</p>
</dd>
</dl>
<p>ResultsAllCellLines:  Updated structure ResultsAllCellLines.sample.singleGeneDeletion</p>
</dd>
<dt>OPTIONAL OUTPUTS:</dt><dd><p>OverViewResults:      Updated overview</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.calculateLODs">
<span class="sig-name descname"><span class="pre">calculateLODs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">theo_mass</span></em>, <em class="sig-param"><span class="pre">lod_ngmL</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.calculateLODs" title="Permalink to this definition">¶</a></dt>
<dd><p>This function converts detection limits of the unit ng/mL to mM using the
theoretical mass (g/mol) for the metabolites</p>
<p>USAGE:</p>
<blockquote>
<div><p>[lod_mM] = calculateLODs(theo_mass, lod_ngmL)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>theo_mass:         Vector that specifies the theoretical mass (g/mol) of each metabolite
lod_ngmL:          Instrumental limit of detection (ng/mL)</p>
</dd>
<dt>OUTPUT:</dt><dd><p>lod_mM:            Detection limits in mM</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.calculateQuantitativeDiffs">
<span class="sig-name descname"><span class="pre">calculateQuantitativeDiffs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">data_RXNS</span></em>, <em class="sig-param"><span class="pre">slope_Ratio</span></em>, <em class="sig-param"><span class="pre">ex_RXNS</span></em>, <em class="sig-param"><span class="pre">lod_mM</span></em>, <em class="sig-param"><span class="pre">cond1_uptake</span></em>, <em class="sig-param"><span class="pre">cond2_uptake</span></em>, <em class="sig-param"><span class="pre">cond1_secretion</span></em>, <em class="sig-param"><span class="pre">cond2_secretion</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.calculateQuantitativeDiffs" title="Permalink to this definition">¶</a></dt>
<dd><p>This function provides sets of exchange reactions with higher uptake and
secretion in condition 1 and condition 2.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[cond1_upt_higher, cond2_upt_higher, cond2_secr_higher, cond1_secr_higher, cond1_uptake_LODs, cond2_uptake_LODs, cond1_secretion_LODs, cond2_secretion_LODs] = calculateQuantitativeDiffs(data_RXNS, slope_Ratio, ex_RXNS, lod_mM, cond1_uptake, cond2_uptake, cond1_secretion, cond2_secretion);</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>data_RXNS:              Exchange reactions same order as Input_A and Input_B
slope_Ratio:            For length of inputA/B, relative difference
ex_RXNS:                Exchange reactions in the same order as LODs(lod_mM)
lod_mM:                 Detection limits in mM
cond1_uptake:           List of exchanges that specify consumed metabolites in condition 1
cond2_uptake:           List of exchanges that specify consumed metabolites in condition 2
cond1_secretion:        List of exchanges that specify released metabolites in condition 1
cond2_secretion:        List of exchanges that specify released metabolites in condition 2</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>cond1_upt_higher:       Exchange reactions and relative differences with higher uptake in condition 1 compared to condition 2
cond2_upt_higher:       Exchange reactions and relative differences with higher uptake in condition 2 compared to condition 1
cond2_secr_higher:      Exchange reactions and relative differences with higher secretion in condition 2 compared to condition 1
cond1_secr_higher:      Exchange reactions and relative differences with higher secretion in condition 1 compared to condition 2
cond1_uptake_LODs:      Detection limits for metabolites with higher uptake in condition 1
cond2_uptake_LODs:      Detection limits for metabolites with higher uptake in condition 2
cond1_secretion_LODs:   Detection limits for metabolites with higher secretion in condition 1
cond2_secretion_LODs:   Detection limits for metabolites with higher secretion in condition 2</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.checkEffectRxnKO">
<span class="sig-name descname"><span class="pre">checkEffectRxnKO</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">samples_to_test</span></em>, <em class="sig-param"><span class="pre">fill</span></em>, <em class="sig-param"><span class="pre">Genes_to_test</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">ResultsAllCellLines</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.checkEffectRxnKO" title="Permalink to this definition">¶</a></dt>
<dd><p>This function checks the effect of constraining reactions associated
with a single or set of genes on the ability of the model to satisfy
an objective.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[FBA_Rxns_KO, ListResults] = checkEffectRxnKO(samples_to_test, fill, Genes_to_test, samples, ResultsAllCellLines)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>ResultsAllCellLines:   uses <cite>modelMin</cite>
samples:               Name of samples
samples_to_test:       Name of samples that should be tested (can be samples if all should be tested)
fill:                  Identifier if the <cite>rxns</cite> is not in the model (e.g.,100, num(‘NAN’))
Genes_to_test:         Set of genes to be tested</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>FBA_Rxns_KO:           FBA results for constraining one reaction at a time to zero.
ListResults:           Reactions associated with <cite>Genes_to_test</cite>, same order as <cite>FBA_Rxns_KO</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.checkExchangeProfiles">
<span class="sig-name descname"><span class="pre">checkExchangeProfiles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">path</span></em>, <em class="sig-param"><span class="pre">nmets</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.checkExchangeProfiles" title="Permalink to this definition">¶</a></dt>
<dd><p>The Function generates a summary of the number of uptake exchanges
and secretion exchanges per samples.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[mapped_exchanges, minMax, mapped_uptake, mapped_secretion] = checkExchangeProfiles(samples, path, nmets)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>samples:
path:     path to output of make exchangeprofiles
nmets:    number of metabolites in data set (max number of uptake or secreted metabolites)</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>mapped_exchanges:    table listing for each sample the number of uptake and secretion that were mapped to the model
minMax:              lists minimal and maximal number of uptakes, secretions, and total number of exchanges
mapped_uptake:       table summarizing for each sample the uptake exchange reactions
mapped_secretion:    table summarizing for each sample the secretion exchange reactions</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.computeFluxSplits">
<span class="sig-name descname"><span class="pre">computeFluxSplits</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">mets</span></em>, <em class="sig-param"><span class="pre">V</span></em>, <em class="sig-param"><span class="pre">coeffSign</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.computeFluxSplits" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute relative contributions of fluxes (<cite>V</cite>) to the net production (<cite>P</cite>)
and consumption (<cite>C</cite>) of a set of metabolites (<cite>mets</cite>)</p>
<p>USAGE:</p>
<blockquote>
<div><p>[P, C, vP, vC, s] = computeFluxSplits(model, mets, V, 0);</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:     a COBRA model structure with required fields</p>
<blockquote>
<div><ul class="simple">
<li><p>.S:        the <cite>m</cite> x <cite>n</cite> stoichiometric matrix</p></li>
<li><p>.mets:     an <cite>m</cite> x <cite>1</cite> cell array of metabolite identifiers</p></li>
</ul>
</div></blockquote>
<p>mets:      a list of metabolite identifiers from model.mets
V:         an <cite>n</cite> x <cite>k</cite> matrix of <cite>k</cite> flux distributions</p>
</dd>
<dt>OPTIONAL INPUT:</dt><dd><dl class="simple">
<dt>coeffSign: only use the sign of the stoichiometric coefficient</dt><dd><p>(i.e. +1 or -1, Default = false)</p>
</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt><dd><p>P:     an <cite>n</cite> x <cite>k</cite> matrix of relative contributions to the production of mets
C:     an <cite>n</cite> x <cite>k</cite> matrix of relative contributions to the consumption of mets
vP:    an <cite>n</cite> x <cite>k</cite> matrix of net producing fluxes
vC:    an <cite>n</cite> x <cite>k</cite> matrix of net consuming fluxes
s:     net stoichiometry of reactions containing metabolites</p>
</dd>
</dl>
<p>Example: [P,C,vP,vC,s] = computeFluxSplits(model,mets,V,1);</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.conc2Rate">
<span class="sig-name descname"><span class="pre">conc2Rate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">metConc</span></em>, <em class="sig-param"><span class="pre">cellConc</span></em>, <em class="sig-param"><span class="pre">t</span></em>, <em class="sig-param"><span class="pre">cellWeight</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.conc2Rate" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts metabolite concentration and (viable) cell concentration into
uptake rate.
CellConc consumed MetConc in t.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[flux] = conc2Rate(metConc, cellConc, t, cellWeight)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>metConc:       Change in metabolite concentration (mM)
cellConc:      Cell concentration (cells per 1 ml)
t:             Time in hours
cellWeight:    gDW per cell</p>
</dd>
<dt>OUTPUT:</dt><dd><p>flux:          mmol/gDW/hr</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.defineUptakeSecretionProfiles">
<span class="sig-name descname"><span class="pre">defineUptakeSecretionProfiles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">input_A</span></em>, <em class="sig-param"><span class="pre">input_B</span></em>, <em class="sig-param"><span class="pre">data_RXNS</span></em>, <em class="sig-param"><span class="pre">tol</span></em>, <em class="sig-param"><span class="pre">essAA_excl</span></em>, <em class="sig-param"><span class="pre">exclude_upt</span></em>, <em class="sig-param"><span class="pre">exclude_secr</span></em>, <em class="sig-param"><span class="pre">add_secr</span></em>, <em class="sig-param"><span class="pre">add_upt</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.defineUptakeSecretionProfiles" title="Permalink to this definition">¶</a></dt>
<dd><p>This function calculated the slope ratios and gives out uptake&amp;secretion profiles for condition 1 and condition 2</p>
<p>USAGE:</p>
<blockquote>
<div><p>[cond1_uptake, cond2_uptake, cond1_secretion, cond2_secretion, slope_Ratio] = defineUptakeSecretionProfiles(input_A, input_B, data_RXNS, tol, essAA_excl, exclude_upt, exclude_secr, add_secr, add_upt)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>input_A:             Matrix, 4 columns</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Control TP1,</p></li>
<li><p>Control TP2,</p></li>
<li><p>Condition 1 TP1,</p></li>
<li><p>Condition 1 TP 2</p></li>
</ol>
</div></blockquote>
<p>input_B:             Matrix, 4 columns</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Control2 TP1,</p></li>
<li><p>Control2 TP2,</p></li>
<li><p>Condition 2 TP1,</p></li>
<li><p>Condition 2 TP 2</p></li>
</ol>
</div></blockquote>
<p>data_RXNS:           Exchange reactions same order as <cite>input_A</cite> and <cite>input_B</cite>
tol:                 minimal threshold to call uptake or secretion (Default 0.05)=5%</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl class="simple">
<dt>essAA_excl:          If essential amino acids should be excluded from the secretion profile 1(yes), or 0(no) (Default = 0);</dt><dd><p><cite>essAAs</cite> = {<cite>EX_his_L(e)</cite>; <cite>EX_ile_L(e)</cite>; <cite>EX_leu_L(e)</cite>; <cite>EX_lys_L(e)</cite>; <cite>EX_met_L(e)</cite>; <cite>EX_phe_L(e)</cite>; <cite>EX_thr_L(e)</cite>; <cite>EX_trp_L(e)</cite>; <cite>EX_val_L(e)</cite>}</p>
</dd>
</dl>
<p>exclude_upt:         Exclude uncertain metabolites from uptake (e.g., metabolites from GlutaMax, e.g., <cite>EX_gln_L(e)</cite>, <cite>EX_cys_L(e)</cite>, <cite>EX_ala_L(e)</cite>)
exclude_secr:        Exclude uncertain metabolites from secretion (e.g., metabolites from GlutaMax, e.g., <cite>EX_gln_L(e)</cite>, <cite>EX_cys_L(e)</cite>, <cite>EX_ala_L(e)</cite>)
add_secr:            Due to mising data points automatic analysis might do wrong assignment of metabolites to secretion
add_upt:             Due to mising data points automatic analysis might do wrong assignment of metabolites to uptake</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>cond1_uptake:        List of exchanges that specify consumed metabolites in condition 1
cond2_uptake:        List of exchanges that specify consumed metabolites in condition 2
cond1_secretion:     List of exchanges that specify released metabolites in condition 1
cond2_secretion:     List of exchanges that specify released metabolites in condition 2
slope_Ratio:         For length of inputA/B, relative difference</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.extractConditionSpecificModel">
<span class="sig-name descname"><span class="pre">extractConditionSpecificModel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">threshold</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.extractConditionSpecificModel" title="Permalink to this definition">¶</a></dt>
<dd><p>The function prunes a subnetwork based on a user-defined threshold. The subnetwork does not contain blocked reactions. Please note that Recon has blocked reactions which will always be removed. Thus, not all reactions are removed as a consequence of the data integration.
Depends on fastFVA, fluxVariability analysis</p>
<p>USAGE:</p>
<blockquote>
<div><p>[modelPruned] = extractConditionSpecificModel(model, threshold)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:           Global metabolic model (Recon) - constrained
threshold:        Fluxes below the threshold will be considered zero and respective reactions as blocked, e.g., 10e-6.</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>modelPruned:     submodel without blocked reactions</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.findMinCardModel">
<span class="sig-name descname"><span class="pre">findMinCardModel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">Ex_Rxns</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.findMinCardModel" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to find the minimal cardinality model.</p>
<dl class="simple">
<dt>USAGE:</dt><dd><p>[modelMin, AddedExchange] = findMinCardModel(model, Ex_Rxns)</p>
</dd>
<dt>INPUTS:</dt><dd><p>model:         Metabolic model
Ex_Rxns:       Vector of exchange reactions for FVA</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>modelMin:       Updated model with new reaction bounds
AddedExchange:  Vector of exchanged reactions</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.findOptExchRxns">
<span class="sig-name descname"><span class="pre">findOptExchRxns</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">Ex_Rxns</span></em>, <em class="sig-param"><span class="pre">parFVA</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.findOptExchRxns" title="Permalink to this definition">¶</a></dt>
<dd><p>USAGE:</p>
<blockquote>
<div><p>[OptExchRxns] = findOptExchRxns(model, Ex_Rxns, parFVA)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:         Metabolic model
Ex_Rxns:       Vector of exchange reactions for FVA
fastFVA:       use FastFVA (default=0)</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>OptExchRxns:</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.generateCompactExchModel">
<span class="sig-name descname"><span class="pre">generateCompactExchModel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">minGrowth</span></em>, <em class="sig-param"><span class="pre">biomassRxn</span></em>, <em class="sig-param"><span class="pre">prune</span></em>, <em class="sig-param"><span class="pre">fastFVA</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.generateCompactExchModel" title="Permalink to this definition">¶</a></dt>
<dd><p>This function identifies a subnetwork with the least number of possible exchange
reactions given the model and the applied constraints. It returns the resulting pruned model.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[modelMin, modelPruned, Ex_Rxns] = generateCompactExchModel(model, minGrowth, biomassRxn, prune, fastFVA)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:         model structure
minGrowth:     minimal Growth rate to be set on biomass reaction
biomassRxn:    biomass reaction name (default: ‘biomass_reaction2’)
prune:         optional: to prune the model based on exchange reactions</p>
<blockquote>
<div><p>(default: 1)</p>
</div></blockquote>
<dl class="simple">
<dt>fastFVA:       optional: to use fastFVA instead of fluxvariability for</dt><dd><p>computing FVA results (default: 0)</p>
</dd>
</dl>
<p>medium:        (default: {})</p>
</dd>
<dt>OUTPUTS:</dt><dd><dl class="simple">
<dt>modelUpdated:  same as input model but constraints on blocked reactions</dt><dd><p>are set to be 0</p>
</dd>
<dt>modelPruned:   pruned model, where all blocked reactions are removed</dt><dd><p>(attention this seems to cause issues with GPRs)</p>
</dd>
</dl>
<p>Ex_Rxns:       List of exchange reactions in pruned model</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.illustrate_ppp">
<span class="sig-name descname"><span class="pre">illustrate_ppp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">ResultsAllCellLines</span></em>, <em class="sig-param"><span class="pre">mets</span></em>, <em class="sig-param"><span class="pre">path</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">label</span></em>, <em class="sig-param"><span class="pre">fonts</span></em>, <em class="sig-param"><span class="pre">tol</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.illustrate_ppp" title="Permalink to this definition">¶</a></dt>
<dd><p>This function generates and saves heatmaps for the results of the function
performPPP for all sample models.</p>
<p>USAGE:</p>
<blockquote>
<div><p>illustrate_ppp(ResultsAllCellLines, mets, path, samples, label, fonts, tol)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>ResultsAllCellLines:     Result structure
mets:                    Metabolites that were tested in the phase plane analysis
step_size:               Step size of each metabololite tested
path:                    Path where output is saved
samples:                 Names of conditions
label:                   Defining label of X-axis, y-axis and z-axis, e.g., {<cite>Glucose uptake (fmol/cell/hr)</cite>; <a href="#id2"><span class="problematic" id="id3">`</span></a>Oxygen uptake</p>
<blockquote>
<div><p>(fmol/cell/hr)`; <cite>Growth rate (hr-1)</cite>} The z-axis is the growth rate which is color coded</p>
</div></blockquote>
<p>fonts:                   Font size for labels on heatmap
tol:                     Fluxes are considered zero if they are below the tol</p>
</dd>
</dl>
<p>Individual pdf files showing the result of the PPP are being saved automatically for each condition.</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.integrateGeneExpressionData">
<span class="sig-name descname"><span class="pre">integrateGeneExpressionData</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">dataGenes</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.integrateGeneExpressionData" title="Permalink to this definition">¶</a></dt>
<dd><p>This function sets constrains based on sets of absent genes. It does not test for the functionality of the model, to maintain full control over
the model generation process. Alternatively, the COBRA function <cite>createTissueSpecificModel</cite> can be used, and which can produce a functional model.
See the supplemental tutorial of <em>DOI:10.1371/journal.pone.0049978 (Aurich and Thiele, PloS One, 2012)</em> for description on potentially necessary curation when using this crude method of data integration.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[modelGE] = integrateGeneExpressionData(model, dataGenes)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:           Metabolic model (e.g., Recon)
dataGenes:       Vector of absent genes. Follow the supplemental tutorial of <em>DOI: 10.1371/journal.pone.0049978 (Aurich and Thiele, PloS One, 2012)</em> for the generation of P/A calls, e.g., <cite>Absent_genes</cite> = [535;1548];</p>
</dd>
<dt>OUTPUT:</dt><dd><dl class="simple">
<dt>modelGE:         Model, where all genes in <cite>DataGenes</cite> have been constrained to zero. Follow the supplemental tutorial of</dt><dd><p><em>DOI:10.1371/journal.pone.0049978 (Aurich and Thiele, PloS One, 2012)</em> for description of potentially necessary curation.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.make3Dplot">
<span class="sig-name descname"><span class="pre">make3Dplot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">PHs</span></em>, <em class="sig-param"><span class="pre">maximum_contributing_flux</span></em>, <em class="sig-param"><span class="pre">fonts</span></em>, <em class="sig-param"><span class="pre">path</span></em>, <em class="sig-param"><span class="pre">diff_view</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.make3Dplot" title="Permalink to this definition">¶</a></dt>
<dd><p>The function allows illustration on different phenotypes predicted in
MetTB onto the 3D illustration plot. The phenotypes are defined as
colors, the location of the points is based on the ATP flux predicted by the
function <cite>predictFluxSplits</cite>.</p>
<p>USAGE:</p>
<blockquote>
<div><p>make3Dplot(PHs, maximum_contributing_flux, fonts, path, diff_view)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>PHs:                          Phenotypes = [samples PHENOTYPE], is generated from different functions (<cite>predictFluxSplits</cite>,…)
maximum_contributing_flux:    Output of the function <cite>predictFluxSplits</cite>, option ATP.
fonts:                        Font size
path:                         Path to save output
diff_view:                    Next to the standard view, print graphic from 3 different(2D) viewpoints (default = 0/off)</p>
</dd>
</dl>
<p>PDFs are automatically saved to the user-defined path</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.makeSummaryModels">
<span class="sig-name descname"><span class="pre">makeSummaryModels</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">ResultsAllCellLines</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">mk_union</span></em>, <em class="sig-param"><span class="pre">mk_intersect</span></em>, <em class="sig-param"><span class="pre">mk_reactionDiff</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.makeSummaryModels" title="Permalink to this definition">¶</a></dt>
<dd><p>This function generates the union and the intersect model from the <cite>modelPruned</cite> in the <cite>ResultsAllCellLines</cite> structure.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[unionModel, intersectModel, diffRxns, diffExRxns] = makeSummaryModels(ResultsAllCellLines, samples, model, mk_union, mk_intersect, mk_reactionDiff)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>ResultsAllCellLines:                structure containing samples and models for the samples, e.g.,  <cite>ResultsAllCellLines.UACC_257.modelPruned</cite>
samples:                            conditions or cell lines, e.g., UACC_257</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>mk_union:                           make union model, yes=1, no=0 (Default = 1)
mk_intersect:                       make intersect model, yes=1, no=0 (Default = 1)
mk_reactionDiff:                    make reactionDiff, yes=1, no=0 (can only be 1 if union and intersect are 1 or []) (Default = 1).</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>unionModel:                         model containing all reactions appearing at least once in the models in the <cite>ResultsAllCellLines.sample.modelPruned</cite>
intersectModel:                     model containing all reactions shared by all models in the <cite>ResultsAllCellLines.sample.modelPruned</cite>
diffRxns:                           all differential reactions that distinguish <cite>unionModel</cite> and <cite>intersectModel</cite>
diffExRxns:                         all differential exchange reactions that distinguish <cite>unionModel</cite> and <cite>intersectModel</cite></p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.mkTableOfAddedExchanges">
<span class="sig-name descname"><span class="pre">mkTableOfAddedExchanges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">ResultsAllCellLines</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">Ex_added_all_unique</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.mkTableOfAddedExchanges" title="Permalink to this definition">¶</a></dt>
<dd><p>The function generates a table of the added exchanges defined by the
function <cite>generateCompactExchModel</cite> called by the function <cite>setQuantConstraints</cite>.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[Added_all] = mkTableOfAddedExchanges(ResultsAllCellLines, samples, Ex_added_all_unique)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>ResultsAllCellLines:
samples:
Ex_added_all_unique:    Output of the function <cite>statisticsAddedExchanges</cite></p>
</dd>
<dt>OUTPUT:</dt><dd><p>Added_all:              Table overview of all added Exchanges for each sample</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.networkTopology">
<span class="sig-name descname"><span class="pre">networkTopology</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.networkTopology" title="Permalink to this definition">¶</a></dt>
<dd><p>Analysis of the metabolite connectivity of a metabolic model</p>
<p>USAGE:</p>
<blockquote>
<div><p>[MetConn, RxnLength] = NetworkTopology(model)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:       Model structure</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>MetConn:     Vector of metabolite connectivity (how many reactions a metabolite participates in (in same order as model.mets)
RxnLength:   Vector of reaction participation, i.e., how many metabolites per reaction (in same order as <cite>mode.rxns</cite>)</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.performPPP">
<span class="sig-name descname"><span class="pre">performPPP</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">ResultsAllCellLines</span></em>, <em class="sig-param"><span class="pre">mets</span></em>, <em class="sig-param"><span class="pre">step_size</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">step_num</span></em>, <em class="sig-param"><span class="pre">direct</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.performPPP" title="Permalink to this definition">¶</a></dt>
<dd><p>This function performs the a phase plane analysis. The analysis starts from zero and proceed <cite>step_num*step_size</cite> in the specified direction and for two exchanges.
the results of the analysis will be saved into <cite>ResultsAllCellLines</cite>.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[ResultsAllCellLines] = performPPP(ResultsAllCellLines, mets, step_size, samples, step_num, direct)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>ResultsAllCellLines:
samples:                 Conditions
mets:                    Set of two metabolites to test e.g.  mets ={<cite>EX_glc(e)</cite>, <cite>EX_gln(e)</cite>; <cite>EX_o2(e)</cite>, <cite>EX_o2(e)</cite>}
step_size:               Step size for metabolites specified in mets e.g., <cite>step_size</cite> = 100
direct:                  Direction for metabolites specified in mets: uptake (-1) or secretion (1), e.g., direct = [-1,-1,-1,-1];
step_num:                Number of steps (1000/20 = 50), step_num = [5,5;5,5]</p>
</dd>
<dt>OUTPUT:</dt><dd><p>ResultsAllCellLines:     The matrix of growth rates are added to the <cite>ResultsAllCellLines</cite> structure alond with the values or the bounds using the same of the manipulated exchanges</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.performSampling">
<span class="sig-name descname"><span class="pre">performSampling</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">warmupn</span></em>, <em class="sig-param"><span class="pre">fileName</span></em>, <em class="sig-param"><span class="pre">nFiles</span></em>, <em class="sig-param"><span class="pre">pointsPerFile</span></em>, <em class="sig-param"><span class="pre">stepsPerPoint</span></em>, <em class="sig-param"><span class="pre">fileBaseNo</span></em>, <em class="sig-param"><span class="pre">maxTime</span></em>, <em class="sig-param"><span class="pre">path</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.performSampling" title="Permalink to this definition">¶</a></dt>
<dd><p>The function performs the sampling analysis for one model, combines
multiple COBRA toolbox functions</p>
<p>USAGE:</p>
<blockquote>
<div><p>performSampling(model, warmupn, fileName, nFiles, pointsPerFile, stepsPerPoint, fileBaseNo, maxTime, path)</p>
</div></blockquote>
<p>INPUTS:</p>
<blockquote>
<div><p>model:                      Model to be sampled</p>
<p>warmupn:                    Number of warm-up points
fileName:                   Name for output files
nFiles:                     Number of files saved
pointsPerFilePoints:        Points saved per file
stepsPerPoint:              Steps skipped between two saved points, in order to increase mixing
fileBaseNo:                 Counter for the numbering of the output files, e.g., 0 to start with 1;
maxTime:                    Maximal running time after which the analysis should be terminated, e.g., 3600000;
path:                       Path to file</p>
</div></blockquote>
<p>Automatically saved to path using specified file name</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.predictFluxSplits">
<span class="sig-name descname"><span class="pre">predictFluxSplits</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">obj</span></em>, <em class="sig-param"><span class="pre">met2test</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">ResultsAllCellLines</span></em>, <em class="sig-param"><span class="pre">dir</span></em>, <em class="sig-param"><span class="pre">transportRxns</span></em>, <em class="sig-param"><span class="pre">ATPprod</span></em>, <em class="sig-param"><span class="pre">carbon_source</span></em>, <em class="sig-param"><span class="pre">eucNorm</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.predictFluxSplits" title="Permalink to this definition">¶</a></dt>
<dd><p>This function performs the flux splits analysis for the metabolites of
interest, meaning it predicts the fraction of metabolite produced (or
consumed) based all reactions producing (or consuming) the metabolite</p>
<p>USAGE:</p>
<blockquote>
<div><p>[BMall, ResultsAllCellLines, metRsall, maximum_contributing_rxn, maximum_contributing_flux, ATPyield] = predictFluxSplits(model, obj, met2test, samples, ResultsAllCellLines, dir, transportRxns, ATPprod, carbon_source, eucNorm)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:                       Generic model, e.g., <cite>modelMedium</cite>
obj:                         objective function, e.g., biomass or ATPM
met2test:                    e.g., <cite>atp[c]</cite>. Mind that the metabolites are produced in multiple compartments.
samples:                     Name of conditions as in <cite>ResultsAllCellLines</cite>
ResultsAllCellLines:         Structure containing the pruned submodels of the samples</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>dir:                         Production = 1 (default = production), else consumption = 0
eucNorm:                     Default: 1e-6
transportRxns:               Vector of reactions that do not really produce or consume the metabolite, e.g., reactions that transport ATP from one compartment to the other. This input is optional only to allow the initial prediction of all producing reactions to define the exclude reaction set.
carbon_source:               Reference uptake for calculation of ATP yield, e.g.,{<cite>EX_glc(e)</cite>}.
ATPprod:</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>BMall:                       Matrix of flux vectors used for calculations
ResultsAllCellLines:         Structure containing results of run analysis
metRsall:                    Matrix of flux (producing or consuming) a defined metabolite
maximum_contributing_rxn:    Reactions with highest flux (producing or</p>
<blockquote>
<div><p>consuming) a defined metabolite across analyzed samples (not
necessarily &gt;50%), if multiple reactions have the same contribution, all will be reported seperated by a back slash.</p>
</div></blockquote>
<p>maximum_contributing_flux:   Matrix containing:</p>
<blockquote>
<div><ul class="simple">
<li><p>highest flux (column 1),</p></li>
<li><p>sum of flux (producing or consuming) a defined metabolite (column 2),</p></li>
<li><p>percentage (column 3),</p></li>
<li><p>contribution of glycolysis (column 4),</p></li>
<li><p>contribution of ETC (column 5),</p></li>
<li><p>combined contribution of glycolysis and ETC (column 6),</p></li>
<li><p>contribution of TCA (column 7),</p></li>
<li><p>combined contribution of glycolysis, ETC, and TCA (column 8).</p></li>
</ul>
</div></blockquote>
<dl class="simple">
<dt>ATPyield:                    ATP yield calculated from the sum of ATP production divided by the predicted uptake flux of the metabolite specified as carbon_source.</dt><dd><p>No extra constraints are applied, thus not only production  flux from the specified carbon source is considered.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Example</p>
<p>% if met2test is not atp to reduce number of useless outputs
[BMall, ResultsAllCellLines, metRsall] = predictFluxSplits(model, obj, met2test,samples,ResultsAllCellLines, dir, eucNorm, transportRxns, ATPprod, carbon_source)</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.prepIntegrationQuant">
<span class="sig-name descname"><span class="pre">prepIntegrationQuant</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">metData</span></em>, <em class="sig-param"><span class="pre">exchanges</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">test_max</span></em>, <em class="sig-param"><span class="pre">test_min</span></em>, <em class="sig-param"><span class="pre">outputPath</span></em>, <em class="sig-param"><span class="pre">tol</span></em>, <em class="sig-param"><span class="pre">variation</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.prepIntegrationQuant" title="Permalink to this definition">¶</a></dt>
<dd><p>This function generates individual uptake and secretion profiles from a
data matrix (fluxes) with samples as columns and metabolites as rows.
Negative values are interpreted as uptake and positive values are
interpreted as secretion. The function tests based on the input model if
uptake and secretion of all ‘Exchanges’ is possible. Subsequently, it
removes from each sample the metabolite uptakes or secretions that cannot
be realized by the model due to missing production or degradation
pathways, or blocked reactions. If only secretion is not possible, only
secretion is eliminated from the sample profile whereas uptake will still
be mapped.
The individual uptake and secretion profile for each sample is saved to
the location specified in outputPath using the unique sample name.</p>
<p>USAGE:</p>
<blockquote>
<div><p>prepIntegrationQuant(model, metData, exchanges, samples, test_max, test_min, outputPath, tol, variation)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:                   Prepared global model (e.g., <cite>model_for_CORE</cite> from <cite>prepModel</cite>)
exchanges:               Vector containing exchange reactions
metData:                 Fluxes of uptake (negative) and secretion(positive) flux values. The columns are the samples and the rows are the metabolites (Unit matching remaining model constraints!).
samples:                 Vector of sample names (no dublicate names)
test_max:                Maximal uptake/secretion set while testing if model can perform uptake and secretion of a metabolite (e.g., 500)
test_min:                Minimal uptake/secretion set while testing if model can perform uptake and secretion of a metabolite (e.g., 0.00001)
outputPath:              Path where output files should be saved (e.g. ‘Y:StudiesData_integrationCOREusingRecon1models')</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>tol:                     All fluxes below this value are considered to be zero, (e.g., 1e-6)
variation:               Lower and upper bound are established with this value as error range in % (e.g.,20)</p>
</dd>
</dl>
<p>For every sample a file is automatically saved and contains the following variables:</p>
<blockquote>
<div><ul class="simple">
<li><p>Secretion_not_possible:  Vector of metabolite (exchange reactions) that cannot be secreted by the model</p></li>
<li><p>Uptake_not_possible:     Vector of metabolite (exchange reactions) that cannot be uptaken by the model</p></li>
<li><p>FBA_all_secreted:        FBA results of test metabolite secretion</p></li>
<li><p>FBA_all_secreted_names:  Name of exchange test metabolite secretion</p></li>
<li><p>FBA_all_uptake:          FBA results of test metabolite uptake</p></li>
<li><p>FBA_all_uptake_names:    Name of exchange test metabolite uptake</p></li>
<li><p>uptake:                  Vector of exchange reactions that are associated with uptake in the cell line (no additional exchanges, since these reactions will not be closed)</p></li>
<li><p>uptake_value:            Matrix of flux values, constitute the lower (column 2) and upper (column 3) limits for the model uptake</p></li>
<li><p>secretion:               Vector of exchange reactions that are associated with secretion in the cell line (no additional exchanges, since these reactions will not be closed)</p></li>
<li><p>secr_value:              Matrix of flux values, constitute the lower (column 2) and upper (column 3) limits for the model secretion</p></li>
</ul>
</div></blockquote>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.pruneModel">
<span class="sig-name descname"><span class="pre">pruneModel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">minGrowth</span></em>, <em class="sig-param"><span class="pre">biomassRxn</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.pruneModel" title="Permalink to this definition">¶</a></dt>
<dd><p>This function prunes a model to its most compact subnetwork given some model
constraints and a minimal growth rate by identifying and removing all blocked reactions.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[modelUpdated, modelPruned, Ex_Rxns] = pruneModel(model, minGrowth, biomassRxn)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:         model structure
minGrowth:     minimal Growth rate to be set on biomass reaction
biomassRxn:    biomass reaction name (default: ‘biomass_reaction2’)</p>
</dd>
<dt>OUTPUTS:</dt><dd><dl class="simple">
<dt>modelUpdated:  same as input model but constraints on blocked reactions</dt><dd><p>are set to be 0</p>
</dd>
<dt>modelPruned:   pruned model, where all blocked reactions are removed</dt><dd><p>(attention this seems to cause issues with GPRs)</p>
</dd>
</dl>
<p>Ex_Rxns:       List of exchange reactions in pruned model</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.setConstraintsOnBiomassReaction">
<span class="sig-name descname"><span class="pre">setConstraintsOnBiomassReaction</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">of</span></em>, <em class="sig-param"><span class="pre">dT</span></em>, <em class="sig-param"><span class="pre">tolerance</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.setConstraintsOnBiomassReaction" title="Permalink to this definition">¶</a></dt>
<dd><p>This function sets constrains biomass objective function in the model accoring to the condition specific doubling time.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[modelBM] = setConstraintsOnBiomassReaction(model, of, dT, tolerance)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:           Metabolic model (e.g., Recon)
of:              Objective funtion, e.g., <cite>biomass_reaction</cite>
dT:              Doubling time
tolerance:       Upper and lower limit of the growth rate are adjusted according to this tolerance value, e.g., 20 (%).</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>modelBM:         Model constrained with condition-specific growth rates</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.setMediumConstraints">
<span class="sig-name descname"><span class="pre">setMediumConstraints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">set_inf</span></em>, <em class="sig-param"><span class="pre">current_inf</span></em>, <em class="sig-param"><span class="pre">medium_composition</span></em>, <em class="sig-param"><span class="pre">met_Conc_mM</span></em>, <em class="sig-param"><span class="pre">cellConc</span></em>, <em class="sig-param"><span class="pre">t</span></em>, <em class="sig-param"><span class="pre">cellWeight</span></em>, <em class="sig-param"><span class="pre">mediumCompounds</span></em>, <em class="sig-param"><span class="pre">mediumCompounds_lb</span></em>, <em class="sig-param"><span class="pre">customizedConstraints</span></em>, <em class="sig-param"><span class="pre">customizedConstraints_ub</span></em>, <em class="sig-param"><span class="pre">customizedConstraints_lb</span></em>, <em class="sig-param"><span class="pre">close_exchanges</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.setMediumConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates and sets constraints according to fresh medium composition in mM. Is based on the function <cite>Conc2Rate</cite>. Returns a model with constraints.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[modelMedium, basisMedium] = setMediumConstraints(model, set_inf, current_inf, medium_composition, met_Conc_mM, cellConc, t, cellWeight, mediumCompounds, mediumCompounds_lb, customizedConstraints, customizedConstraints_ub, customizedConstraints_lb, close_exchanges)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:                     Metabolic model (Recon)
current_inf:               Models can have differently defined infinite constraints, e.g., 500
set_inf:                   New value for infinite constraints, e.g., 1000
medium_composition:        Vector of exchange reactions of metabolites in the cell medium, e.g., RPMI medium
met_Conc_mM:               Vector of the same length of ‘Exchanges’, providing the concentration (in mM) of each metabolite in the respective row
cellConc:                  Cell concentration (cells per 1 ml)
t:                         Time in hours
cellWeight:                Cellular dry weight
mediumCompounds:           Composition of basic medium, which are usually not defined in the composition of the medium and need to be defined in addition.
mediumCompounds_lb:        Lower bound applied for all mediumCompounds (the same for all)</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>customizedConstraints:     If additional constraints should be set apart from the basic medium and medium composition, e.g. <cite>EX_o2(e)</cite>
customizedConstraints_lb:  Vecor of lower bounds to be set (mmol/gDW/hr), must be of same length as customizedConstraints
customizedConstraints_ub:  Vecor of upper bounds to be set (mmol/gDW/hr), must be of same length as customizedConstraints
close_exchanges:           If exchange reactions except those specified should be closed (Default = 1) 1= close exchanges, 0=no</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>modelMedium:               Model with set constraints, all exchanges not specified in  mediumCompounds, ions or customizedConstraints are set to secretion only
basisMedium:               Vector of adjusted constraints, for reference such that these constraints are not overwritten at a later stage</p>
</dd>
</dl>
<p>Please note that if metabolites of <cite>medium_composition</cite> and <cite>mediumCompounds</cite> overlap, the constraints of the <cite>Medium_composition</cite> will
be set to 0 in the output model. The function depends on the functions <cite>conc2Rate</cite>, <cite>changeRxnBounds</cite>.</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.setQualitativeConstraints">
<span class="sig-name descname"><span class="pre">setQualitativeConstraints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">cond_uptake</span></em>, <em class="sig-param"><span class="pre">cond_uptake_LODs</span></em>, <em class="sig-param"><span class="pre">cond_secretion</span></em>, <em class="sig-param"><span class="pre">cond_secretion_LODs</span></em>, <em class="sig-param"><span class="pre">cellConc</span></em>, <em class="sig-param"><span class="pre">t</span></em>, <em class="sig-param"><span class="pre">cellWeight</span></em>, <em class="sig-param"><span class="pre">ambiguous_metabolites</span></em>, <em class="sig-param"><span class="pre">basisMedium</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.setQualitativeConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>This function sets qualitative constraints (by enforcing minimal uptake or secretion based on individual detection limits), e.g., based on the uptake and secretion
profile of metabolites measured through mass-spectrometry. Uptake is only possible if the lower bound has been set to a value &gt;0 using <cite>setMediumConstraints</cite>
The minimal allowable uptake and secretion flux is defined by enforcing uptake at or above the limit of detection (mass-spectrometry). If these values are not available, a very small value, e.g., 1.0E-06 can be used. Note that this
value has to be below the concentrations defined in the medium. Otherwise the model will be infeasible. Alternatively to the LODs a small value can
be used (e.g., 0.00001). Note that the value has to be above the threshold later called zero.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[modelLOD] = setQualitativeConstraints(model, cond_uptake, cond_uptake_LODs, cond_secretion, cond_secretion_LODs, cellConc, t, cellWeight, ambiguous_metabolites, basisMedium)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:                   Metabolic model (Recon), with set constraints (output of <cite>setMediumConstraints</cite>)
cond_uptake:             Vector of exchanges of metabolites consumed by the cells in the experiment
cond_uptake_LODs:        Vector of detection limits (LOD in Mm) for the compounds and in the experiment
cond_secretion:          Vector of metabolite exchanges consumed by the cells in the experiment
cond_secretion_LODs:     Vector of detection limits (LOD in Mm) for the compounds and in the experiment
cellConc:                Cell concentration (cells per 1 ml)
t:                       Time in hours
cellWeight:              gDW per cell
ambiguous_metabolites:   Since all exchanges, except the ones specified in <cite>uptake</cite> and <cite>secretion</cite> are closed in <cite>modelLOD</cite>, this input</p>
<blockquote>
<div><p>variable allows to specify metabolite exchanges that should remain open.
Thus, if these exchanges are open in the starting model they can be uptaken
or secreted by the <cite>modelLOD</cite>, e.g., <cite>ambiguous_metabolites</cite>
This can for example be the case if it is suspected that an uptake or secretion might have taken place in concentrations below the
detection limit.</p>
</div></blockquote>
<p>basisMedium:             Vector defining the metabolite exchanges of the basic medium, i.e., <cite>ions</cite>, <cite>mediumCompounds</cite></p>
</dd>
<dt>OUTPUT:</dt><dd><p>modelLOD:                Model that is constrained qualitatively to the condition-specific uptake and secretion profile</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.setQuantConstraints">
<span class="sig-name descname"><span class="pre">setQuantConstraints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">tol</span></em>, <em class="sig-param"><span class="pre">minGrowth</span></em>, <em class="sig-param"><span class="pre">obj</span></em>, <em class="sig-param"><span class="pre">no_secretion</span></em>, <em class="sig-param"><span class="pre">no_uptake</span></em>, <em class="sig-param"><span class="pre">medium</span></em>, <em class="sig-param"><span class="pre">addExtraExch</span></em>, <em class="sig-param"><span class="pre">addExtraExch_value</span></em>, <em class="sig-param"><span class="pre">path</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.setQuantConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes a model and quantitative extracellular metabolomic data and returns a model in which the data is integrated as constraints described
in “MetaboTools: A Comprehensive Toolbox for Analysis of Genome-Scale Metabolic Models”, 2016, Front. Physiol., and the supplemental data “Tutorial II: Workflow for the integration
of quantitative extracellular metabolomic data into the network context.” (Data Sheet 3.pdf)</p>
<p>USAGE:</p>
<blockquote>
<div><p>[ResultsAllCellLines, OverViewResults] = setQuantConstraints(model, samples, tol, minGrowth, obj, no_secretion, no_uptake, medium, addExtraExch, addExtraExch_value, path, epsilon)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:                Global metabolic model (Recon)
samples:              Vector specifying the samples used (there must be an output file of function …. for each sample)
tol:                  Cutoff value for small numbers (e.g., -1e-8). All number smaller than tol will be treated as zero
minGrowth:            Will be the lower bound of the objective function (e.g., 0.008). Forces the output model(s) to be able to produce a minimal objective value
obj:                  Objective function, e.g. <cite>biomass_reaction2</cite>
no_secretion:         Define metabolites that should not be secreted (e.g., {<cite>EX_o2(e)</cite>})
no_uptake:            Define metabolites that should not be consumed (e.g., {<cite>EX_o2s(e)</cite>, <cite>EX_h2o2(e)</cite>})
medium:               Define if certain exchanges should be excluded from minimization of exchanges (e.g., {}, if no medium except the exometabolomic data has been defined)
addExtraExch:         After adding secretions, models are still not growing, this variable allows one to recover exchanges with a defined small value
addExtraExch_value:   e.g. 1 as arbitrary small flux value / the resulting ub = 1, lb = -1.
path:                 Location of the .mat files for samples.</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>ResultsAllCellLines:  Structure that contains pruned and unpruned model, Vector of the Exchange_reactions, the exchange reactions added by <cite>minExCard</cite>, <cite>minFLux</cite> and <cite>maxFlux</cite> of the added reactions, maximal objective value, and the results of the gene deletion
OverViewResults:      Overview of model statistics, e.g., number of reactions, metabolites, genes, number of essential genes, min and max objective values for easy comparison between sets of models</p>
</dd>
</dl>
<p>Depends on <cite>changeRxnBounds</cite>, <cite>fluxVariability</cite>, <cite>optimizeCbModel</cite>, <cite>generateCompactExchModel</cite> as well its dependent functions <cite>pruneModel</cite>, <cite>findMinCardModel</cite>, <cite>findOptExchRxns</cite></p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.setSemiQuantConstraints">
<span class="sig-name descname"><span class="pre">setSemiQuantConstraints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">modelA</span></em>, <em class="sig-param"><span class="pre">modelB</span></em>, <em class="sig-param"><span class="pre">cond1_upt_higher</span></em>, <em class="sig-param"><span class="pre">cond2_upt_higher</span></em>, <em class="sig-param"><span class="pre">cond2_secr_higher</span></em>, <em class="sig-param"><span class="pre">cond1_secr_higher</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.setSemiQuantConstraints" title="Permalink to this definition">¶</a></dt>
<dd><p>This function establishes constraints for two models (<cite>modelA</cite> and <cite>modelB</cite>) based on the relative
difference of signal intensities obtained from two samples. The prerequisite for the adjustment is that constraints have been applied on which the relative differences can be applied:</p>
<ul class="simple">
<li><ol class="upperalpha simple">
<li><p>The uptake rates have been established based on the medium composition and concentrations (i.e., <cite>setMediumConstraints</cite>), and</p></li>
</ol>
</li>
<li><ol class="upperalpha simple" start="2">
<li><p>qualitative constraints have been applied based on know detection limits or small values (i.e., <cite>setQualitativeConstraints</cite>).</p></li>
</ol>
</li>
</ul>
<p>USAGE:</p>
<blockquote>
<div><p>[modelA_QUANT, modelB_QUANT] = setSemiQuantConstraints(modelA, modelB, cond1_upt_higher, cond2_upt_higher, cond2_secr_higher, cond1_secr_higher)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>modelA:                      model constrained according to condition 1
modelB:                      model constrained according to condition 2
cond1_upt_higher:            Exchange reactions and relative differences with higher uptake in condition 1 compared to condition 2
cond2_upt_higher:            Exchange reactions and relative differences with higher uptake in condition 2 compared to condition 1
cond2_secr_higher:           Exchange reactions and relative differences with higher secretion in condition 2 compared to condition 1
cond1_secr_higher:           Exchange reactions and relative differences with higher secretion in condition 1 compared to condition 2</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>modelA_QUANT:                model constrained according to condition 1
modelB_QUANT:                model constrained according to condition 2</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.statisticsAddedExchanges">
<span class="sig-name descname"><span class="pre">statisticsAddedExchanges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">ResultsAllCellLines</span></em>, <em class="sig-param"><span class="pre">samples</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.statisticsAddedExchanges" title="Permalink to this definition">¶</a></dt>
<dd><p>This function performs statistics on the added exchange reactions using
the structure <cite>ResultsAllCellLines</cite> generated by the function
<cite>setQuantConstraints</cite> and across all samples.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[Ex_added_all_unique] = statisticsAddedExchanges(ResultsAllCellLines, samples)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><dl class="simple">
<dt>ResultsAllCellLines:         Last output <cite>ResultsAllCellLines</cite> that contains</dt><dd><p>results of all samples (last in <cite>sampleResult.mat</cite>) e.g. in
‘Y:StudiesData_integrationTutorial_paperToolboxUACC_257_2Result.mat’</p>
</dd>
</dl>
<p>samples:</p>
</dd>
<dt>OUTPUTS:</dt><dd><dl class="simple">
<dt>Ex_added_all_unique:         a table of exchange reactions which the models contain in addition to the</dt><dd><p>mapped(metabolomics-based)-exchanges, ordered by the
frequency of appearance.</p>
<ul class="simple">
<li><p>Column 1 = exchange reactions,</p></li>
<li><p>Column 2 = frequency of use,</p></li>
<li><p>Column 3 = secretion,</p></li>
<li><p>Column 4 = uptake.</p></li>
</ul>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.metabotools.summarizeSamplingResults">
<span class="sig-name descname"><span class="pre">summarizeSamplingResults</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">modelA</span></em>, <em class="sig-param"><span class="pre">modelB</span></em>, <em class="sig-param"><span class="pre">path</span></em>, <em class="sig-param"><span class="pre">nFiles</span></em>, <em class="sig-param"><span class="pre">pointsPerFile</span></em>, <em class="sig-param"><span class="pre">starting_model</span></em>, <em class="sig-param"><span class="pre">dataGenes</span></em>, <em class="sig-param"><span class="pre">show_rxns</span></em>, <em class="sig-param"><span class="pre">fonts</span></em>, <em class="sig-param"><span class="pre">hist_per_page</span></em>, <em class="sig-param"><span class="pre">bin</span></em>, <em class="sig-param"><span class="pre">fileNameA</span></em>, <em class="sig-param"><span class="pre">fileNameB</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.dataIntegration.metabotools.summarizeSamplingResults" title="Permalink to this definition">¶</a></dt>
<dd><p>The function summarizes the sampling results of the two models <cite>modelA</cite> and
<cite>modelB</cite>. Subsequently, it returnes medians from the two models’ sampling,
FVA results for the respective reaction, simple test results (<cite>stats</cite>).
Finally, histograms are produced, one for each reactions containing the
distributions of the two models. The set of analyzed reactions can be
limited (<cite>show_rxns</cite>). Reactions associated with genes of special interest,
e.g. differentially expressed genes, can be marked to facilitate the
analysis, and simplify the identification of interesting histograms.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[stats, statsR] = summarizeSamplingResults(modelA, modelB, path, nFiles, pointsPerFile, starting_model, dataGenes, show_rxns, fonts, hist_per_page, bin, fileNameA, fileNameB)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>modelA:                     Sampled modelA (condition 1)
modelB:                     Sampled modelB (condition 2)
path:                       Path to sampling output files
nFiles:                     Number of files saved, e.g., 20;
pointsPerFilePoints:        Points saved per file, e.g., 5000;
starting_model:             Original metabolic model (Recon)
dataGenes:                  Gene set, whose associated reactions should be emphasized by color, e.g., alternatively spliced or differentially expressed genes</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>show_rxns:                  Vector of reactions that should be displayed, e.g., certain pathways or non-loop reactions
fonts:                      Font size (default = 9)
hist_per_page:              Defines the number of histogramms that are printed per page, e.g., 4, 9, or 25 (default = 9)
bin:                        Binning for histogramm (default = 30)
fileNameA:                  Name of files the sampling points are stored in
fileNameB:                  Name of files the sampling points are stored in</p>
</dd>
<dt>OUTPUTS:</dt><dd><p>stats:                      Statistics from the sampling (Columns: Median modelA, Median modelB, minFlux_A, maxFlux_A, minFlux_B, maxFlux_B )
statsR:                     Reaction vector for stats output</p>
</dd>
</dl>
<p>Depends on COBRA functions: <cite>loadsamples</cite>, <cite>findRxnsFromGenes</cite>, flux
variability analysis / <cite>fastFVA</cite></p>
</dd></dl>

</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../transcriptomics/index.html" class="btn btn-default float-right" title="Transcriptomics" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../fluxomics/c13solver/index.html" class="btn btn-default" title="c13solver" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2021, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../_static/sphinx_highlight.js"></script>

  


  
  
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>