

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>rMTA &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../index.html"/>
        <link rel="up" title="Analysis" href="../index.html"/>
        <link rel="next" title="robustnessAnalysis" href="../robustnessAnalysis/index.html"/>
        <link rel="prev" title="rFBA" href="../rFBA/index.html"/> 

  
  <script src="../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="https://king.nuigalway.ie/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../_static/../" + item.website_url + "' ><img src='../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../dynamicFBA/index.html">Dynamic FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FBA/index.html">FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../findMIIS/index.html">findMIIS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fluxEnrichmentAnalysis/index.html">fluxEnrichmentAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FVA/index.html">FVA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../geometricFBA/index.html">geometricFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../gMCS/index.html">gMCS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../KLdistance/index.html">KLdistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../mdFBA/index.html">mdFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../MOMA/index.html">MOMA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../multiSpecies/index.html">MultiSpecies</a></li>
<li class="toctree-l3"><a class="reference internal" href="../parsimoniousFBA/index.html">parsimoniousFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../relaxedFBA/index.html">relaxedFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rFBA/index.html">rFBA</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">rMTA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../robustnessAnalysis/index.html">robustnessAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../ROOM/index.html">ROOM</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rumba/index.html">rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sparseFBA/index.html">sparseFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../thermo/index.html">Thermo</a></li>
<li class="toctree-l3"><a class="reference internal" href="../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../uFBA/index.html">uFBA</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../dataIntegration/index.html">Data integration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contact.html">Contact</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../contents.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Functions</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Analysis</a></li>
      <li class="breadcrumb-item active">rMTA</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="rmta">
<span id="id1"></span><h1>rMTA<a class="headerlink" href="#rmta" title="Permalink to this heading">¶</a></h1>
<span class="target" id="module-src.analysis.rMTA"></span><dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.MTA">
<span class="sig-name descname"><span class="pre">MTA</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">rxnFBS</span></em>, <em class="sig-param"><span class="pre">Vref</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.MTA" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate Metabolic Transformation Analysis (MTA) using the the
solver CPLEX.
Code was prepared to be able to be stopped and be launched again by using
a temporary file called ‘temp_MTA.mat’.
Outputs are cell array for each alpha (one simulation by alpha). It
there is only one alpha, content of cell will be returned
The code here has been based on:</p>
<blockquote>
<div><p>Yizhak, K., Gabay, O., Cohen, H., &amp; Ruppin, E. (2013).
‘Model-based identification of drug targets that revert disrupted
metabolism and its application to ageing’. Nature communications, 4.
<a class="reference external" href="http://www.nature.com/ncomms/2013/131024/ncomms3632/full/ncomms3632.html">http://www.nature.com/ncomms/2013/131024/ncomms3632/full/ncomms3632.html</a></p>
</div></blockquote>
<p>USAGE:</p>
<blockquote>
<div><p>[TSscore,deletedGenes,Vout] = MTA(model, rxnFBS, Vref, alpha, epsilon, varargin)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:               Metabolic model structure (COBRA Toolbox format).
rxnFBS:              Array that contains the desired change: Forward,</p>
<blockquote>
<div><p>Backward and Unchanged (+1;0;-1). This is calculated
from the rules and differential expression analysis.</p>
</div></blockquote>
<p>Vref:                Reference flux of the source state.</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl class="simple">
<dt>alpha:               Numeric value or array. Parameter of the quadratic</dt><dd><p>problem (default = 0.66)</p>
</dd>
<dt>epsilon:             Numeric value or array. Minimun perturbation for each</dt><dd><p>reaction (default = 0)</p>
</dd>
<dt>rxnKO:               Binary value. Calculate knock outs at reaction level</dt><dd><p>instead of gene level. (default = false)</p>
</dd>
<dt>listKO:             Cell array containing the name of genes or</dt><dd><p>reactions to knockout. (default = all)</p>
</dd>
<dt>timelimit:           Time limit for the calculation of each knockout.</dt><dd><p>(default = inf)</p>
</dd>
<dt>SeparateTranscript:  Character used to separate different transcripts of a gene. (default = ‘’).</dt><dd><dl class="simple">
<dt>Examples:</dt><dd><ul class="simple">
<li><dl class="simple">
<dt>SeparateTranscript = ‘’</dt><dd><ul>
<li><p>gene 10005.1    ==&gt;    gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005.2</p></li>
<li><p>gene 10005.3    ==&gt;    gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>SeparateTranscript = ‘.’</dt><dd><ul>
<li><p>gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005</p></li>
<li><p>gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</dd>
<dt>numWorkers:          Integer: is the maximun number of workers</dt><dd><p>used by the solver. 0 = automatic, 1 = sequential,
&gt; 1 = parallel. (default = 0)</p>
</dd>
<dt>FORCE_CPLEX:         1 to force CPLEX solver, 0 (default) for COBRA</dt><dd><p>solver.</p>
</dd>
<dt>printLevel:          Integer. 1 if the process is wanted to be shown</dt><dd><p>on the screen, 0 otherwise. (default = 1)</p>
</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt><dd><p>TSscore:             Transformation score by each transformation
deletedGenes:        The list of genes/reactions removed in each knock-out
Vres:                Matrix of resulting fluxes</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.MTA_MIQP">
<span class="sig-name descname"><span class="pre">MTA_MIQP</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">OptimizationModel</span></em>, <em class="sig-param"><span class="pre">KOrxn</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.MTA_MIQP" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the OptimizationModel solution of a particular MTA problem and
an specific model</p>
<p>USAGE:</p>
<blockquote>
<div><p>[v_res, success, unsuccess] = MTA_MIQP (OptimizationModel, KOrxn, numWorkers, timeLimit, printLevel)</p>
</div></blockquote>
<dl>
<dt>INPUT:</dt><dd><p>OptimizationModel:    Cplex Model struct
KOrxn:                perturbation in the model (reactions)
numWorkers:           number of threads used by Cplex.
FORCE_CPLEX:          1 to force CPLEX solver, 0 (default) for COBRA</p>
<blockquote>
<div><p>solver.</p>
</div></blockquote>
<dl class="simple">
<dt>printLevel:           1 if the process is wanted to be shown on the</dt><dd><p>screen, 0 otherwise. Default: 1.</p>
</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt><dd><p>Vout:                 Solution flux of MIQP formulation for each case
solution:             Cplex solution struct</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.MTA_TS">
<span class="sig-name descname"><span class="pre">MTA_TS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">Vout</span></em>, <em class="sig-param"><span class="pre">Vref</span></em>, <em class="sig-param"><span class="pre">rxnFBS</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.MTA_TS" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the TS score of a particular solution of the MTA perturbation
algorithm.</p>
<p>USAGE:</p>
<blockquote>
<div><p>score = MTA_TS(v_res,vref,Model,success,unsuccess)</p>
</div></blockquote>
<dl>
<dt>INPUT:</dt><dd><p>Vout:             Solution flux of MIQP formulation for each case
Vref:             Reference flux of source state
rxnFBS:           Array that contains the desired change: Forward,</p>
<blockquote>
<div><p>Backward and Unchanged (+1;0;-1).</p>
</div></blockquote>
</dd>
<dt>OUTPUTS:</dt><dd><p>score:            TS score for each case</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.buildMTAproblemFromModel">
<span class="sig-name descname"><span class="pre">buildMTAproblemFromModel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">rxnFBS</span></em>, <em class="sig-param"><span class="pre">Vref</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.buildMTAproblemFromModel" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the COBRA Optimization model needed to perform the MTA</p>
<p>USAGE:</p>
<blockquote>
<div><p>OptimizationModel = buildMTAproblemFromModel(model,rxnFBS,Vref,alpha,epsilon)</p>
</div></blockquote>
<dl>
<dt>INPUT:</dt><dd><p>model:                 Metabolic model (COBRA format)
rxnFBS:                Forward, Backward and Unchanged (+1;0;-1) values</p>
<blockquote>
<div><p>corresponding to each reaction.</p>
</div></blockquote>
<p>Vref:                  Reference flux of the source state.
alpha:                 parameter of the quadratic problem (default = 0.66)
epsilon                minimun disturbance for each reaction, (default = 0)</p>
</dd>
<dt>OUTPUTS:</dt><dd><dl class="simple">
<dt>OptimizationModel:     COBRA model struct that includes the</dt><dd><p>stoichiometric contrains, the thermodinamic
constrains and the binary variables.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.calculateEPSILON">
<span class="sig-name descname"><span class="pre">calculateEPSILON</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">samples</span></em>, <em class="sig-param"><span class="pre">rxnFBS</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.calculateEPSILON" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the minimum required flux change (epsilon) for reactions to be
considered as significantly changed with p-value of 0.05
The code below is based on the method presented in:</p>
<blockquote>
<div><p>Yizhak, K., Gabay, O., Cohen, H., &amp; Ruppin, E. (2013). Model-based
identification of drug targets that revert disrupted metabolism and
its application to ageing. Nature communications, 4, 2632.</p>
</div></blockquote>
<p>USAGE:</p>
<blockquote>
<div><p>epsilon = calculateEPSILON(samples, rxnFBS, ‘unique_epsilon’, false, ‘minimum’, 1e-3)</p>
</div></blockquote>
<dl>
<dt>INPUT:</dt><dd><p>samples:           Matrix of sampled fluxes.
rxnFBS:            Array that contains the desired change: Forward,</p>
<blockquote>
<div><p>Backward and Unchanged (+1;0;-1). This is calculated
from the rules and differential expression analysis.</p>
</div></blockquote>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>varargin:  <cite>ParameterName</cite> value pairs with the following options:</p>
<blockquote>
<div><ul class="simple">
<li><p><cite>unique_epsilon</cite>: True = unique epsilon, False = each reaction has different epsilon (default=false)</p></li>
<li><p><cite>minimum</cite>: Minimun value for epsilon requiered (default=1e-3)</p></li>
</ul>
</div></blockquote>
</dd>
<dt>OUTPUT:</dt><dd><p>epsilon:           Numeric value or array with the epsilon for the different reactions</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.calculateGeneKOMatrix">
<span class="sig-name descname"><span class="pre">calculateGeneKOMatrix</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.calculateGeneKOMatrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Build a rxn-gene matrix such that the i-th column indicates what
reactions become inactive because of the i-th gene’s knock-out.</p>
<p>USAGE:</p>
<blockquote>
<div><p>geneKO = calculateGeneKOMatrix(model, SeparateTranscript, printLevel)</p>
</div></blockquote>
<dl>
<dt>INPUT:</dt><dd><p>model:             The COBRA Model structure</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl>
<dt>SeparateTranscript Character used to separate</dt><dd><p>different transcripts of a gene. (default = ‘’)
Examples:</p>
<blockquote>
<div><ul class="simple">
<li><dl class="simple">
<dt>SeparateTranscript = ‘’</dt><dd><ul>
<li><p>gene 10005.1    ==&gt;    gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005.2</p></li>
<li><p>gene 10005.3    ==&gt;    gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>SeparateTranscript = ‘.’</dt><dd><ul>
<li><p>gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005</p></li>
<li><p>gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
</dd>
<dt>printLevel:       Integer. 1 if the process is wanted to be shown</dt><dd><p>on the screen, 0 otherwise. (default = 1)</p>
</dd>
</dl>
</dd>
<dt>OUTPUT:</dt><dd><dl class="simple">
<dt>geneKO:            Struct which contains matrix with blocked</dt><dd><p>reactions for each gene in the metabolic model,
name of reactions and name of genes.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.diffexprs2rxnFBS">
<span class="sig-name descname"><span class="pre">diffexprs2rxnFBS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">diff_exprs</span></em>, <em class="sig-param"><span class="pre">Vref</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.diffexprs2rxnFBS" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns Forward - Backward - Unchanged (+1;0;-1) for each reaction.
(+1)R_f    (-1)R_b     (0)unchanged</p>
<p>USAGE:</p>
<blockquote>
<div><p>rxnFBS = diffexprs2rxnFBS(model, diff_exprs, Vref, ‘SeparateTranscript’, ‘’, ‘logFC’, 0, ‘pval’,0.05)</p>
</div></blockquote>
<dl>
<dt>INPUT:</dt><dd><p>model:             The COBRA Model structure
diff_exprs:        MATLAB Table including the information of the</p>
<blockquote>
<div><p>differentially expressed genes.
Required columns (with theses names):</p>
<blockquote>
<div><ul class="simple">
<li><p>gene ( ID of gene, same as in the meabolic model)</p></li>
<li><p>logFC (SOURCE VS TARGET)</p></li>
<li><p>pval (p-value or adjusted-p-value)</p></li>
</ul>
</div></blockquote>
</div></blockquote>
<p>Vref:              Reference flux of the model</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>varargin:          <cite>ParameterName</cite> value pairs with the following options:</p>
<blockquote>
<div><ul class="simple">
<li><p><cite>SeparateTranscript</cite>: Character used to separate different transcripts of a gene. (default: ‘’)</p></li>
<li><p><cite>logFC</cite>: minimum log2 (fold change) requiered (default = 0)</p></li>
<li><p><cite>pval</cite>: maximum p-value admited (default = 0.05)</p></li>
</ul>
</div></blockquote>
</dd>
<dt>OUTPUT:</dt><dd><dl class="simple">
<dt>rxnFBS:             array containting the information of altered</dt><dd><p>reactions: Forward - Backward - Unchanged
(+1;0;-1);</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.rMTA">
<span class="sig-name descname"><span class="pre">rMTA</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">rxnFBS</span></em>, <em class="sig-param"><span class="pre">Vref</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.rMTA" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate robust Metabolic Transformation Analysis (rMTA) using the
solver CPLEX.
Code was prepared to be able to be stopped and be launched again by using
a temporally file called ‘temp_rMTA.mat’.
Outputs are cell array for each alpha (one simulation by alpha). It
there is only one alpha, content of cell will be returned</p>
<p>USAGE:</p>
<blockquote>
<div><p>[TSscore,deletedGenes,Vout] = rMTA(model, rxnFBS, Vref, alpha, epsilon, varargin)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:              Metabolic model structure (COBRA Toolbox format).
rxnFBS:             Array that contains the desired change: Forward,</p>
<blockquote>
<div><p>Backward and Unchanged (+1;0;-1). This is calculated
from the rules and differential expression analysis.</p>
</div></blockquote>
<p>Vref:               Reference flux of the source state.</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl class="simple">
<dt>alpha:              Numeric value or array. Parameter of the quadratic</dt><dd><p>problem (default = 0.66)</p>
</dd>
<dt>epsilon:            Numeric value or array. Minimun perturbation for each</dt><dd><p>reaction (default = 0)</p>
</dd>
<dt>parameterK:         Numeric value value. Parameter to calculate rTS</dt><dd><p>score. (default = 100)</p>
</dd>
<dt>rxnKO:              Binary value. Calculate knock outs at reaction level</dt><dd><p>instead of gene level. (default = false)</p>
</dd>
<dt>listKO:             Cell array containing the name of genes or</dt><dd><p>reactions to knockout. (default = all)</p>
</dd>
</dl>
<p>timelimit:          Time limit for the calculation of each knockout. (default = inf)
SeparateTranscript: Character used to separate</p>
<blockquote>
<div><p>different transcripts of a gene. (default = ‘’)
Examples:</p>
<blockquote>
<div><ul class="simple">
<li><dl class="simple">
<dt>SeparateTranscript = ‘’</dt><dd><ul>
<li><p>gene 10005.1    ==&gt;    gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005.2</p></li>
<li><p>gene 10005.3    ==&gt;    gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>SeparateTranscript = ‘.’</dt><dd><ul>
<li><p>gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005</p></li>
<li><p>gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
</div></blockquote>
<dl class="simple">
<dt>numWorkers:         Integer: is the maximun number of workers</dt><dd><p>used by the solver. 0 = automatic, 1 = sequential,
&gt; 1 = parallel. (default = 0)</p>
</dd>
<dt>FORCE_CPLEX:        1 to force CPLEX solver, 0 (default) for COBRA</dt><dd><p>solver.</p>
</dd>
<dt>printLevel:         Integer. 1 if the process is wanted to be shown</dt><dd><p>on the screen, 0 otherwise. (default = 1)</p>
</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt><dd><p>TSscore:            Transformation score by each transformation
deletedGenes:       The list of genes/reactions removed in each knock-out
Vref:               Matrix of resulting fluxes</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.rMTA.rMTAsaveInExcel">
<span class="sig-name descname"><span class="pre">rMTAsaveInExcel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">xls_filename</span></em>, <em class="sig-param"><span class="pre">TSscore</span></em>, <em class="sig-param"><span class="pre">deletedGenes</span></em>, <em class="sig-param"><span class="pre">alpha_values</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.rMTA.rMTAsaveInExcel" title="Permalink to this definition">¶</a></dt>
<dd><p>Save in Excel all the results calculated using rMTA.</p>
<p>USAGE:</p>
<blockquote>
<div><p>rMTAsaveInExcel(output_file, TSscore, deletedGenes, alpha_values, varargin)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>xls_filename:       name of the resulting excel file
TSscore:            Transformation score by each transformation
deletedGenes:       The list of genes/reactions removed in each knock-out
alpha_values:       Numeric value or array. Parameter of the quadratic</p>
<blockquote>
<div><p>problem (default = alpha_1, alpha_2,…)</p>
</div></blockquote>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl class="simple">
<dt>gene_info:          Table with the gene names and additional</dt><dd><p>information. It needs a column called ‘gene’,
which contains the same information as the
deletedGenes variable.</p>
</dd>
<dt>differ_genes:       Table with the gene differential expression</dt><dd><p>information. It needs a column called ‘gene’,
which contains the same information as the
deletedGenes variable.</p>
</dd>
<dt>RankingGeneID:      Caracter array with the ID of the gene which score</dt><dd><p>is desired to know.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../robustnessAnalysis/index.html" class="btn btn-default float-right" title="robustnessAnalysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../rFBA/index.html" class="btn btn-default" title="rFBA" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2021, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../_static/sphinx_highlight.js"></script>

  


  
  
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>