

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Pairwise Interaction Modeling &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../../../_static/css/cobra.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="../../../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../../../index.html"/>
        <link rel="up" title="Microbiome Modeling Toolbox" href="../index.html"/>
        <link rel="next" title="mgPipe" href="../mgPipe/index.html"/>
        <link rel="prev" title="additionalAnalysis" href="../additionalAnalysis/index.html"/> 

  
  <script src="../../../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="http://gibbs.unal.edu.co/cobradoc/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../../../_static/../" + item.website_url + "' ><img src='../../../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../../../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../dynamicFBA/index.html">Dynamic FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../FBA/index.html">FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../findMIIS/index.html">findMIIS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../fluxEnrichmentAnalysis/index.html">fluxEnrichmentAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../FVA/index.html">FVA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../geometricFBA/index.html">geometricFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../gMCS/index.html">gMCS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../KLdistance/index.html">KLdistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../mdFBA/index.html">mdFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../MOMA/index.html">MOMA</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="../../index.html">MultiSpecies</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="../../SteadyCom/index.html">SteadyCom</a></li>
<li class="toctree-l4 current"><a class="reference internal" href="../index.html">Microbiome Modeling Toolbox</a><ul class="current">
<li class="toctree-l5"><a class="reference internal" href="../additionalAnalysis/index.html">additionalAnalysis</a></li>
<li class="toctree-l5 current"><a class="current reference internal" href="#">Pairwise Interaction Modeling</a></li>
<li class="toctree-l5"><a class="reference internal" href="../mgPipe/index.html">mgPipe</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../parsimoniousFBA/index.html">parsimoniousFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../relaxedFBA/index.html">relaxedFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rFBA/index.html">rFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rMTA/index.html">rMTA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../robustnessAnalysis/index.html">robustnessAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../ROOM/index.html">ROOM</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rumba/index.html">rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../sparseFBA/index.html">sparseFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../thermo/index.html">Thermo</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../uFBA/index.html">uFBA</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../dataIntegration/index.html">Data integration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contact.html">Contact</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../../contents.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../../../index.html">Functions</a> &raquo;</li>
        
          <li><a href="../../../index.html">Analysis</a> &raquo;</li>
        
          <li><a href="../../index.html">MultiSpecies</a> &raquo;</li>
        
          <li><a href="../index.html">Microbiome Modeling Toolbox</a> &raquo;</li>
        
      <li>Pairwise Interaction Modeling</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling">
<span id="pairwise-interaction-modeling"></span><span id="pairwiseinteractionmodeling"></span><h1>Pairwise Interaction Modeling<a class="headerlink" href="#module-src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling" title="Permalink to this headline">¶</a></h1>
<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.calculateInteractionsByTaxon">
<code class="descname">calculateInteractionsByTaxon</code><span class="sig-paren">(</span><em>pairwiseInteractions</em>, <em>taxonInformation</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.calculateInteractionsByTaxon" title="Permalink to this definition">¶</a></dt>
<dd><p>Part of the Microbiome Modeling Toolbox. This script analyzes computed pairwise
interactions on different taxonomical levels to yield the interactions
that each taxon (genus, family, order, class, phylum) displayed in total.
The computed pairwise interactions and the taxnomical information for each
analyzed strain are required as inputs.</p>
<p>USAGE:</p>
<blockquote>
<div>[interactionsByTaxon]=calculateinteractionsByTaxon(pairwiseInteractions,taxonInfo)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><dl class="first last docutils">
<dt>pairwiseInteractions:    a table with pairwise interactions computed</dt>
<dd>for a number of analyzed microbe models (output of the function
simulatePairwiseInteractions)</dd>
<dt>taxonInformation:        a table with taxonomical information on the</dt>
<dd>analyzed microbes. Needs to contain at least six columns:
1. The names of the analyzed model structures
in the first column
2-6. A column with the appropriate information
and the header
‘Genus’,’Family’,’Order’,’Class’,’Phylum’
respectively.</dd>
</dl>
</dd>
<dt>OUTPUT:</dt>
<dd><dl class="first last docutils">
<dt>interactionsByTaxon:     a structure with the outcomes predicted for</dt>
<dd>all taxa on each taxonomical level.</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeParetoOptimality">
<code class="descname">computeParetoOptimality</code><span class="sig-paren">(</span><em>model</em>, <em>rxn1</em>, <em>rxn2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeParetoOptimality" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs Pareto optimality analysis for two objective functions by
simultaneously optimizing two reactions (e.g., the biomass objective
functions of two joined organisms). The result is a depiction of the
tradeoff between the two competing objectives.</p>
<p>Multiobjective analysis of two reactions is performed by using the
approach described in Oberhardt, M. A., J. B. Goldberg, et al. (2010).
“Metabolic network analysis of Pseudomonas aeruginosa during chronic
cystic fibrosis lung infection.” J Bacteriol 192(20): 5534-5548. One
reaction is fixed at different intervals and the other reaction is
optimized.</p>
<p>USAGE:</p>
<blockquote>
<div>[ParetoFrontier,fluxSolutions,minFluxes,maxFluxes] = computeParetoOptimality(model,rxn1,rxn2,dinc,FVAflag)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd>model:            COBRA metabolic reconstruction
rxn1:             Reaction ID of the first reaction to be optimized
rxn2:             Reaction ID of the second reaction to be optimized</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd><dl class="first last docutils">
<dt>dinc:             An index which indicates the distance between steps at</dt>
<dd>which the flux of each reaction is fixed (default=0.001).</dd>
<dt>FVAflag:          If true, flux variability analysis is performed at each</dt>
<dd>step.</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first last docutils">
<dt>ParetoFrontier:   Lists the objective values for both reactions next to</dt>
<dd>the interval step. Column 1:the interval step. Column 2
and 3: the flux values of rxn1 in and rxn2,
respectively, at the interval step.</dd>
</dl>
</dd>
<dt>OPTIONAL OUTPUTS:</dt>
<dd><dl class="first last docutils">
<dt>fluxSolutions:    Contains the flux solutions every interval step as a</dt>
<dd>matrix of structures</dd>
<dt>minFluxes:        Reports fastFVA results for every step (minFlux) with</dt>
<dd>one column for each interval step</dd>
<dt>maxFluxes:        Reports fastFVA results for every step (maxFlux) with</dt>
<dd>one column for each interval step</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeRescuedGenes">
<code class="descname">computeRescuedGenes</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeRescuedGenes" title="Permalink to this definition">¶</a></dt>
<dd><p>Part of the Microbiome Modeling Toolbox. This function determines the
effect of the presence of another species on gene deletions in a species.
A joint model consisting of two species is entered. For each gene
deletion that has an effect in each single model, the function conputes
whether the presence of the other species can rescue the flux through the
objective function (e.g., biomass).</p>
<p>USAGE:</p>
<blockquote>
<div>[OptSolKO,OptSolWT,OptSolRatio,fluxesKO]=computeRescuedGenes(varargin)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">modelJoint:    Joint model structure consisting of two COBRA models
Rxn1:          Reaction in the first joined COBRA model for which the</p>
<blockquote>
<div>effect of gene deletions is calculated</div></blockquote>
<dl class="docutils">
<dt>Rxn2:          Reaction in the second joined COBRA model for the effect</dt>
<dd>of gene deletions is calculated</dd>
</dl>
<p>nameTag1:      Species identifier for reactions of the first model
nameTag2:      Species identifier for reactions of the second model
OriModel1:     Original COBRA model structure of first model</p>
<blockquote>
<div>(needed to determine reactions to be constrained)</div></blockquote>
<dl class="last docutils">
<dt>OriModel2:     Original COBRA model structure of second model</dt>
<dd>(needed to determine reactions to be constrained)</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first last docutils">
<dt>OptSolKO:      Matlab structure containing the computed optimal solutions</dt>
<dd>for the gene deletions that had an effect on the two
reactions</dd>
<dt>OptSolRatio:   Matlab structure containing all knockout to wildtype</dt>
<dd>optimal solution ratios for the gene deletions that had
an effect on the two reactions</dd>
<dt>OptSolWT:      Matlab structure containing wildtype flux values for both</dt>
<dd>reactions in the two models</dd>
<dt>RescuedGenes:  List of genes for which deletion is lethal or causes &gt;50%</dt>
<dd>growth reduction in single but not pairwise model</dd>
<dt>fluxesKO:      Matlab structure containing all solutions for all gene</dt>
<dd>deletions and for both reactions in the two models. Can be
used to identify mechanisms for rescued genes.</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.createMultipleSpeciesModel">
<code class="descname">createMultipleSpeciesModel</code><span class="sig-paren">(</span><em>models</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.createMultipleSpeciesModel" title="Permalink to this definition">¶</a></dt>
<dd><p>Based on the implementation from <em>Klitgord and Segre 2010, PMID 21124952</em>.
The present implementation has been used in <em>PMID 23022739</em>, <em>PMID 25841013</em>,
<em>PMID 25901891</em>, <em>PMID 27893703</em>.</p>
<p>Joins one or more COBRA models with or without another COBRA model
representing the host. The created setup when a host is entered is
depicted schematically in Figures 1 and 2 in <em>PMID 27893703</em>.</p>
<p>Creates a common space u (lumen) through which all cells can feed and exchange metabolites,
and separate extracellular spaces for all joined models.
If a host model is entered, a separate compartment <cite>b</cite> (body fluids) which has no
connection to extracellular space is created for the host. Metabolites can be transported from
the lumen to <cite>e</cite>, from <cite>e</cite> to the cytosol and from the cytosol to <cite>b</cite>,
but not from body fluids to the lumen.</p>
<p>USAGE:</p>
<blockquote>
<div>[modelJoint] = createMultipleSpeciesModel(models, ‘nameTagsModels’, nameTagsModels, ‘modelHost’, modelHost, ‘nameTagHost’, nameTagHost)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><dl class="first last docutils">
<dt>models:            cell array of COBRA models(at least one).</dt>
<dd><p class="first">Format</p>
<blockquote class="last">
<div><ul class="simple">
<li>models{1,1} = model 1</li>
<li>models{2,1} = model 2…</li>
</ul>
</div></blockquote>
</dd>
</dl>
</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd><dl class="first docutils">
<dt>nameTagsModels:    cell array of tags for reaction/metabolite abbreviation</dt>
<dd><p class="first">corresponding to each model.
Format</p>
<blockquote class="last">
<div><ul class="simple">
<li>nameTagsModels{1,1} = ‘name tag 1’</li>
<li>nameTagsModels{2,1} = ‘name tag 2’…</li>
</ul>
</div></blockquote>
</dd>
</dl>
<p>modelHost:         COBRA model for host
nameTagHost:       string of tag for reaction/metabolite abbreviation of host model
mergeGenesFlag:    If true, the gene associations in both models are</p>
<blockquote class="last">
<div>included in the joined model. If false, empty fields are created
instead (default:false). Note: merging genes is time-consuming
and may crash certain models.</div></blockquote>
</dd>
<dt>OUTPUT:</dt>
<dd>modelJoint:        model structure for joint model</dd>
</dl>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This function assumes, that exchange reactions are identified by
containing ‘EX’ in the reaction name and that no other reactions do have this property!</p>
</div>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.getMultiSpeciesModelId">
<code class="descname">getMultiSpeciesModelId</code><span class="sig-paren">(</span><em>modelJoint</em>, <em>nameTagsModels</em>, <em>nameTagHost</em>, <em>metTagRe</em>, <em>rxnTagRe</em>, <em>compCom</em>, <em>compHost</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.getMultiSpeciesModelId" title="Permalink to this definition">¶</a></dt>
<dd><p>Get names and IDs for metabolites and exchange reactions in the [u] and [b] space</p>
<p>USAGE:</p>
<blockquote>
<div>[names, ids] = getMultiSpeciesModelId(modelJoint, nameTagsModels, nameTagHost, metTagRe, rxnTagRe, compCom, compHost)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">modelJoint:       COBRA multi-organism model
nameTagsModels:   cell array of tags for species to identify the respective</p>
<blockquote class="last">
<div>reactions and metabolites from <cite>modelJoint.rxns</cite> and <cite>.mets</cite></div></blockquote>
</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd><p class="first">nameTagHost:      string of tag for the host model if exist. Input [] if no host is present
metTagRe:         a regular expression to identify the tag in nameTagsModels from <cite>modelJoint.mets</cite>.</p>
<blockquote>
<div>Use ‘%s’ for the tag in <cite>nameTagsModels</cite> for each species.
Default ‘^%s’, the beginning of the string. In this case, if the tag for species 1 is
‘SP1’, then mets with id ‘SP1glc-D[e]’ will be identified as belonging to species 1.
E.g., ‘met[compartment_SP1]’ where ‘SP1’ is the tag for species 1, input ‘[[^_]+_%s]$’.</div></blockquote>
<dl class="docutils">
<dt>rxnTagRe:         a regular expression to idenify the tag in <cite>nameTagsModels</cite> from <cite>modelJoint.rxns</cite>.</dt>
<dd>Default ‘^%s’, the beginning of the string</dd>
</dl>
<p class="last">compCom:          compartment Id for the community exchange compartment. Default ‘u’
compHost:         compartment Id for the exchange compartment specific to the host. Default ‘b’</p>
</dd>
<dt>OUTPUTS:</dt>
<dd><p class="first">names:            structure of reaction/metabolite name (.rxns/.mets) with the following fields:</p>
<blockquote>
<div><ul class="simple">
<li>spAbbr - species abbreviation (= <cite>nameTags</cite>). Host put at the end</li>
<li>EXcom - #met[u]-by-1 cell. Exchange reactions for community metabolites <cite>met[u]</cite></li>
<li>EXhost - #met[b]-by-1 cell. Exchange reactions for host-specific exchange metabolites <cite>met[b]</cite></li>
<li>EXsp - #met[u]-by-#species cell. <cite>EXsp(i,j)</cite> is the species-community exchange reactions
for the i-th met[u] and the j-th species</li>
<li>Mcom - #met[u]-by-1 cell. All community metabolites <cite>met[u]</cite></li>
<li>Mhost - #met[u]-by-1 cell. Host-specific exchange metabolites <cite>met[b]</cite></li>
<li>Msp - #met[u]-by-#species cell. <cite>Msp(i,j)</cite> is the metabolite in <cite>met[e]</cite> of the <cite>j</cite>-th species
participating in reaction <cite>EXsp(i,j)</cite></li>
<li>rxnSps - #rxns-by-1 cell. <cite>rxnSps(i)</cite> would be <cite>spAbbr(j)</cite> if the <cite>i</cite>-th reaction belongs to the <cite>j</cite>-th species</li>
<li>metSps - #mets-by-1 cell. <cite>metSps(i)</cite> would be <cite>spAbbr(j)</cite> if the i-th metabolite belongs to the <cite>j</cite>-th species</li>
</ul>
</div></blockquote>
<p class="last">ids:              structure of reaction/metabolite index with the same fields as names except without <cite>spAbbr</cite></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.joinModelsPairwiseFromList">
<code class="descname">joinModelsPairwiseFromList</code><span class="sig-paren">(</span><em>modelList</em>, <em>modelFolder</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.joinModelsPairwiseFromList" title="Permalink to this definition">¶</a></dt>
<dd><p>This function joins a list of microbial genome-scale reconstructions in
all combinations. Models are not paired with themselves and pairs are
only created once (model1+model2 = model2+model1). The reactions in each
joined reconstruction structure receive a suffix (entered as input
parameter model list).
Coupling constraints are created in which the flux through all reactions
of each individual microbe is coupled to the flux through its own biomass
objective function. By default, coupling constraints are implemented with
a coupling factor c of 400 and a threshold u of 0. The coupling
parameters can be changed by entering the input parameters c and u.
By default, the gene-protein-reaction associations of the joined models
are not merged as the gene associations are not needed for the simulation
of pairwise interactions and significantly slow down pairwise model
creation.  If merging of genes is desired, please set the input
parameter mergeGenes to true.</p>
<dl class="docutils">
<dt>USAGE:</dt>
<dd>joinModelsPairwiseFromList(modelList,  modelFolder, varargin)</dd>
<dt>INPUTS:</dt>
<dd><dl class="first docutils">
<dt>modelList:          Cell array with names of reconstruction structures to be</dt>
<dd>joined</dd>
</dl>
<p class="last">modelFolder:        Path to folder with COBRA model structures to be joined</p>
</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd><dl class="first docutils">
<dt>c:                  Coupling factor by which reactions in each joined model are</dt>
<dd>coupled to its biomass reaction (default: 400)</dd>
<dt>u:                  Threshold representing minimal flux allowed when flux</dt>
<dd>through biomass reaction in zero (default: 0)</dd>
<dt>mergeGenes:         If true, genes in the joined models are merged and included</dt>
<dd>in the joined model structure (default: false)</dd>
</dl>
<p class="last">numWorkers:         Number of workers in parallel pool if desired
pairwiseModelFolder Folder where pairwise models will be saved</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.printUptakeBoundCom">
<code class="descname">printUptakeBoundCom</code><span class="sig-paren">(</span><em>model</em>, <em>SpFlag</em>, <em>metFlag</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.printUptakeBoundCom" title="Permalink to this definition">¶</a></dt>
<dd><p>Prints the uptake bounds of the whole community and individual species for a community COBRA model</p>
<p>USAGE:</p>
<blockquote>
<div>rxnIDs = printUptakeBoundCom(model, SpFlag, metFlag)</div></blockquote>
<dl class="docutils">
<dt>INPUT:</dt>
<dd><dl class="first last docutils">
<dt>model:    the community model with field <cite>.infoCom</cite> or <cite>.indCom</cite> indicating the indicies of</dt>
<dd>community exchange reactions/metabolites. Can be obtained from <cite>getMultiSpeciesModelId.m</cite></dd>
</dl>
</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd>SpFlag:   true to show individual uptake rates though community uptake is not allowed (default false)
metFlag:  true to print with <cite>model.metNames</cite> (default false)</dd>
<dt>OUTPUTS:</dt>
<dd>printMatrix: matrix of the uptake bounds being printed
printMet:    column of metabolites whose uptake bounds are printed</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.simulatePairwiseInteractions">
<code class="descname">simulatePairwiseInteractions</code><span class="sig-paren">(</span><em>pairwiseModelFolder</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.simulatePairwiseInteractions" title="Permalink to this definition">¶</a></dt>
<dd><p>This function predicts the outcome of pairwise simulations in every
combination from a given list of pairwise models. The pairwise models
need to be created first with the function joinModelsPairwiseFromList.
This script requires the COBRA Toolbox function solveCobraLP. Due to the
coupling constraints on the pairwise models, the simulations cannot
currently be run with optimizeCbModel.
Below is a description of all possible consequences for the two joined
organisms. Please note that the outcomes depend on the two genome-scale
reconstructions joined and are highly dependent on the applied
constraints.</p>
<ul class="simple">
<li>Competition: both organisms grow slower in co-growth than separately
(same outcome for both).</li>
<li>Parasitism: one organism grows faster in co-growth than separately, while
the other grows slower in co-growth than separately. Outcome for
faster-growing organism: Parasitism_Taker, for slower-growing organism:
Parasitism_Giver</li>
<li>Amensalism: one organism’s growth is unaffected by co-growth,  while
the other grows slower in co-growth than separately. Outcome for
unaffected organism: Amensalism_Unaffected, for slower-growing organism:
Amensalism_Affected</li>
<li>Neutralism: both organisms’ growths are unaffected by co-growth (same
outcome for both)</li>
<li>Commensalism: one organism’s growth is unaffected by co-growth,  while
the other grows fatser in co-growth than separately. Outcome for
unaffected organism: Commensalism_Giver, for slower-growing organism:
Commensalism_Taker</li>
<li>Mutualism: both organisms growth faster in co-growth than separately
(same outcome for both)</li>
</ul>
<dl class="docutils">
<dt>USAGE:</dt>
<dd>[pairwiseInteractions, pairwiseSolutions] = simulatePairwiseInteractions(pairwiseModelFolder, varargin)</dd>
<dt>INPUTS:</dt>
<dd>pairwiseModelFolder:     Folder where pairwise models and pairwise model info file are located</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd><dl class="first docutils">
<dt>inputDiet:               Cell array of strings with three columns containing</dt>
<dd>exchange reaction model abbreviations, lower bounds,
and upper bounds.
If no diet is input then the input pairwise models
will be used with unchanged constraints.</dd>
<dt>saveSolutionsFlag:       If true, flux solutions are stored (may result in</dt>
<dd>large output files)</dd>
</dl>
<p>numWorkers:              Number of workers in parallel pool if desired
sigD:                    Difference in growth rate that counts as significant</p>
<blockquote class="last">
<div>change (by default: 10%)</div></blockquote>
</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first last docutils">
<dt>pairwiseInteractions:    Table with computed pairwise and single growth</dt>
<dd>rates for all entered microbe-microbe models</dd>
</dl>
</dd>
<dt>OPTIONAL OUTPUT:</dt>
<dd><dl class="first last docutils">
<dt>pairwiseSolutions:       Table with all computed flux solutions saved</dt>
<dd>(may result in large output files)</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.useDiet">
<code class="descname">useDiet</code><span class="sig-paren">(</span><em>modelIn</em>, <em>dietConstraints</em>, <em>printLevel</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.useDiet" title="Permalink to this definition">¶</a></dt>
<dd><p>Implements diet constraints in a COBRA model structure.</p>
<p>USAGE:</p>
<blockquote>
<div>[modelOut] = useDiet(modelIn, dietConstraints)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">modelIn:                original model
dietConstraints:        cell array of three columns containing</p>
<blockquote>
<div>exchanges, lower bounds, and  upper bounds respectively</div></blockquote>
<p class="last">printLevel:             Verbose level (default: printLevel = 1)</p>
</dd>
<dt>OUTPUT:</dt>
<dd>modelOut:    model with applied constraints</dd>
</dl>
</dd></dl>

</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../mgPipe/index.html" class="btn btn-default float-right" title="mgPipe" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../additionalAnalysis/index.html" class="btn btn-default" title="additionalAnalysis" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2020, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../../_static/language_data.js"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/gh/mathjax/MathJax@2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  


  
  
    <script type="text/javascript" src="../../../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>