

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>gMCS &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../index.html"/>
        <link rel="up" title="Analysis" href="../index.html"/>
        <link rel="next" title="KLdistance" href="../KLdistance/index.html"/>
        <link rel="prev" title="geometricFBA" href="../geometricFBA/index.html"/> 

  
  <script src="../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="https://king.nuigalway.ie/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../_static/../" + item.website_url + "' ><img src='../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../dynamicFBA/index.html">Dynamic FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FBA/index.html">FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../findMIIS/index.html">findMIIS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fluxEnrichmentAnalysis/index.html">fluxEnrichmentAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FVA/index.html">FVA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../geometricFBA/index.html">geometricFBA</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">gMCS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../KLdistance/index.html">KLdistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../mdFBA/index.html">mdFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../MOMA/index.html">MOMA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../multiSpecies/index.html">MultiSpecies</a></li>
<li class="toctree-l3"><a class="reference internal" href="../parsimoniousFBA/index.html">parsimoniousFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../relaxedFBA/index.html">relaxedFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rFBA/index.html">rFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rMTA/index.html">rMTA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../robustnessAnalysis/index.html">robustnessAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../ROOM/index.html">ROOM</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rumba/index.html">rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sparseFBA/index.html">sparseFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../thermo/index.html">Thermo</a></li>
<li class="toctree-l3"><a class="reference internal" href="../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../uFBA/index.html">uFBA</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../dataIntegration/index.html">Data integration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contact.html">Contact</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../contents.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Functions</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Analysis</a></li>
      <li class="breadcrumb-item active">gMCS</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="gmcs">
<span id="id1"></span><h1>gMCS<a class="headerlink" href="#gmcs" title="Permalink to this heading">¶</a></h1>
<span class="target" id="module-src.analysis.gMCS"></span><dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.GPR2models">
<span class="sig-name descname"><span class="pre">GPR2models</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">metabolic_model</span></em>, <em class="sig-param"><span class="pre">selected_rxns</span></em>, <em class="sig-param"><span class="pre">separate_transcript</span></em>, <em class="sig-param"><span class="pre">numWorkers</span></em>, <em class="sig-param"><span class="pre">printLevel</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.gMCS.GPR2models" title="Permalink to this definition">¶</a></dt>
<dd><p>Each GPR rule is converted into a network where the reaction and genes
involved are interconnected.</p>
<p>USAGE:</p>
<blockquote>
<div><p>[networks, rxnNumGenes] = GPR2models(metabolic_model, selected_rxns, separate_transcript, numWorkers, printLevel)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>metabolic_model:        Metabolic model structure (COBRA Toolbox format)
selected_rxns:          Index array which indicates selected reactions to</p>
<blockquote>
<div><p>calculate the network.</p>
</div></blockquote>
<dl class="simple">
<dt>separate_transcript:    Character used to discriminate</dt><dd><p>different isoforms of a gene. Default ‘’.</p>
</dd>
</dl>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl>
<dt>numWorkers:        Maximum number of workers</dt><dd><ul class="simple">
<li><p>0 - maximum provided by the system (automatic)</p></li>
</ul>
<p>(default). If parallel pool is active, numWorkers
is defined by the system, otherwise is 1.
* 1 - sequential
* 2+ - parallel</p>
</dd>
<dt>printLevel:        show the reactions created in models.</dt><dd><ul class="simple">
<li><p>0 - shows nothing</p></li>
<li><p>1 - shows progress by reactions (default)</p></li>
<li><p>2+ - shows everything (reaction and network generation)</p></li>
</ul>
</dd>
</dl>
</dd>
<dt>OUTPUT:</dt><dd><p>networks:          Cell array which contains all the networks.
rxnNumGenes:       Array with the number of genes present in each GPR</p>
<blockquote>
<div><p>rule for each reaction.</p>
</div></blockquote>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.buildGmatrix">
<span class="sig-name descname"><span class="pre">buildGmatrix</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model_name</span></em>, <em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">separate_isoform</span></em>, <em class="sig-param"><span class="pre">numWorkers</span></em>, <em class="sig-param"><span class="pre">printLevel</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.gMCS.buildGmatrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Build the G matrix required for the calculation of genetic Minimal Cut
Sets (gMCSs).</p>
<p>USAGE:</p>
<blockquote>
<div><p>[G, G_ind, related, n_genes_KO, G_time] = buildGmatrix(model_name, model_struct, separate_isoform, numWorkers, printLevel)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model_name:          Name of the metabolic model under study.
model_struct:        Metabolic model structure (COBRA Toolbox format).
separate_isoform:    Character used to discriminate different isoforms of a gene.</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl class="simple">
<dt>numWorkers:        Maximum number of workers</dt><dd><ul class="simple">
<li><p>0 - maximum provided by the system (automatic)</p></li>
<li><p>1 - sequential</p></li>
<li><p>2+ - parallel</p></li>
</ul>
</dd>
<dt>printLevel:        show the reactions created in models.</dt><dd><ul class="simple">
<li><p>0 - shows nothing</p></li>
<li><p>1 - shows progress by reactions (default)</p></li>
<li><p>2+ - shows everything (reaction and network generation)</p></li>
</ul>
</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt><dd><p>G:             G matrix.
G_ind:         Gene knockouts associated with each row in the G matrix.
related:       Relationships among rows of the G matrix.
n_genes_KO:    Number of genes whose knockout is added by each row of the G matrix.
G_time:        Calculation times for each of the steps.</p>
</dd>
</dl>
<p class="rubric">Example</p>
<p>[G, G_ind, related, n_genes_KO, G_time] = buildGmatrix(‘Recon2.v04’, modelR204, ‘.’)</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.calculateGeneMCS">
<span class="sig-name descname"><span class="pre">calculateGeneMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model_name</span></em>, <em class="sig-param"><span class="pre">model_struct</span></em>, <em class="sig-param"><span class="pre">n_gmcs</span></em>, <em class="sig-param"><span class="pre">max_len_gmcs</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.gMCS.calculateGeneMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate genetic Minimal Cut Sets (gMCSs) using the warm-start strategy
available in CPLEX, namely cplex.populate(), with or without selecting a
given knockout, among all the genes included in the model or a given
subset of them. Apaolaza et al., 2017 (Nature Communications).</p>
<p>USAGE:</p>
<blockquote>
<div><p>[gmcs, gmcs_time] = calculateGeneMCS(model_name, model_struct, n_gmcs, max_len_gmcs, varargin)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><dl class="simple">
<dt>model_name:      Name of the metabolic model under study (in order to</dt><dd><p>identify the G matrix).</p>
</dd>
</dl>
<p>model_struct:    Metabolic model structure (COBRA Toolbox format).
n_gmcs:          Number of gMCSs to calculate.
max_len_gmcs:    Number of genes in the largest gMCS to be calculated.</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><p>KO:                 Selected gene knockout. (default = [])
gene_set:           Cell array containing the set of genes among which</p>
<blockquote>
<div><p>the gMCSs are wanted to be calculated.
(default = [], all genes)</p>
</div></blockquote>
<dl>
<dt>target_b:           Desired activity level of the metabolic task to be</dt><dd><p>disrupted. (default = 1e-3)</p>
</dd>
<dt>nutrientGMCS:       Boolean variable.  0 to calculate GeneMCS, 1 to</dt><dd><p>calculate MCS containing genes and nutrients,
known as ngMCS. (default = false)</p>
</dd>
<dt>exchangeRxns:       Cell array containing the set of reactions to be</dt><dd><p>included as inputs of nutrients from the cell
environment / culture medium. (default = [], which
are all reactions with only one 1 metabolite
consiedered as input for the model)</p>
</dd>
<dt>onlyNutrients:      Boolean variable.  1 to calculate MCS only using</dt><dd><p>selected KO and nutrients, 0 to use everything.
If there is no KO selected, it is set to false.
(default = false)</p>
</dd>
<dt>separate_transcript:Character used to separate</dt><dd><p>different transcripts of a gene. (default = ‘’)
Examples:</p>
<blockquote>
<div><ul class="simple">
<li><dl class="simple">
<dt>separate_transcript = ‘’</dt><dd><ul>
<li><p>gene 10005.1    ==&gt;    gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005.2</p></li>
<li><p>gene 10005.3    ==&gt;    gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>separate_transcript = ‘.’</dt><dd><ul>
<li><p>gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005</p></li>
<li><p>gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
</dd>
<dt>forceLength:        1 if the constraint limiting the length of the</dt><dd><p>gMCSs is to be active (recommended for
enumerating low order gMCSs), 0 otherwise
(default = 1)</p>
</dd>
<dt>timelimit:          Time limit for the calculation of gMCSs each time</dt><dd><p>the solver is called. (default = 1e75)</p>
</dd>
<dt>numWorkers:         Integer: is the maximun number of workers used</dt><dd><p>by Cplex and GPR2models. 0 = automatic,
1 = sequential, &gt; 1 = parallel. (default = 0)</p>
</dd>
<dt>printLevel:         Integer. 1 if the process is wanted to be shown</dt><dd><p>on the screen, 0 otherwise. (default = 1)</p>
</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt><dd><p>gmcs:         Cell array containing the calculated gMCSs.
gmcs_time:    Calculation times of the different processes in</p>
<blockquote>
<div><p>the algorithm.</p>
</div></blockquote>
</dd>
</dl>
<p class="rubric">Example</p>
<p>%With optional values
[gmcs, gmcs_time] = calculateGeneMCS(‘Recon2.v04’, modelR204, 100, 10, …</p>
<blockquote>
<div><p>‘KO’ = ‘6240’, …
‘gene_set’ = {‘2987’; ‘6241’},  …
‘timelimit’ = 300,  …
‘target_b’ = 1e-4,
‘separate_transcript’ = ‘.’,  …
‘forceLength’ = 0, …
‘printLevel’ = 0)</p>
</div></blockquote>
<p>%Without optional values
[gmcs, gmcs_time] = calculateGeneMCS(‘ecoli_core_model’, model, 100, 10)</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.calculateMCS">
<span class="sig-name descname"><span class="pre">calculateMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model_struct</span></em>, <em class="sig-param"><span class="pre">n_mcs</span></em>, <em class="sig-param"><span class="pre">max_len_mcs</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.gMCS.calculateMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate Minimal Cut Sets (MCSs) using the warm-start strategy available
in CPLEX, namely cplex.populate(), with or without selecting a given
knockout, among all the reactions included in the model or a given subset
of them. Tobalina et al., 2016 (Bioinformatics); von Kamp and Klamt, 2014
(PLoS Computational Biology).</p>
<p>USAGE:</p>
<blockquote>
<div><p>[mcs, mcs_time] = populateMCS(model_struct, n_mcs, max_len_mcs, options)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model_struct:    Metabolic model structure (COBRA Toolbox format).
n_mcs:           Number of MCSs to calculate.
max_len_mcs:     Number of reactions in the largest MCS to be calculated.</p>
</dd>
</dl>
<p>OPTIONAL INPUT:
OPTIONAL INPUTS:</p>
<blockquote>
<div><p>KO:                 Selected reaction knockout. (default = [])
rxn_set:            Cell array containing the set of reactions among which</p>
<blockquote>
<div><p>the MCSs are wanted to be calculated.
taken from model.rxns
(default = [], all reactions)</p>
</div></blockquote>
<dl class="simple">
<dt>target_b:           Desired activity level of the metabolic task to be</dt><dd><p>disrupted. (default = 1e-3)</p>
</dd>
<dt>forceLength:        1 if the constraint limiting the length of the</dt><dd><p>MCSs is to be active (recommended for
enumerating low order MCSs), 0 otherwise
(default = 1)</p>
</dd>
<dt>timelimit:          Time limit for the calculation of MCSs each time</dt><dd><p>the solver is called. (default = 1e75)</p>
</dd>
<dt>numWorkers:         Integer: is the maximun number of workers used</dt><dd><p>by Cplex and GPR2models. 0 = automatic,
1 = sequential, &gt; 1 = parallel. (default = 0)</p>
</dd>
<dt>printLevel:         Integer. 1 if the process is wanted to be shown</dt><dd><p>on the screen, 0 otherwise. (default = 1)</p>
</dd>
</dl>
</div></blockquote>
<dl>
<dt>OUTPUTS:</dt><dd><p>mcs:         Cell array containing the calculated MCSs.
mcs_time:    Calculation times of the different processes in</p>
<blockquote>
<div><p>the algorithm.</p>
</div></blockquote>
</dd>
</dl>
<p class="rubric">Example</p>
<p>%With optional values
[mcs, mcs_time] = calculateMCS(modelR204, 100, 10, …</p>
<blockquote>
<div><p>‘KO’, ‘r1651’, …
‘rxn_set’, {‘r1652’; ‘r1653’; ‘r1654’}, …
‘timelimit’, 300, …
‘target_b’, 1e-4, …
‘forceLength’, 0, …
‘printLevel’, 0, …</p>
</div></blockquote>
<p>%Without optional values
[mcs, mcs_time] = calculateMCS(model, 100, 10)</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.checkGeneMCS">
<span class="sig-name descname"><span class="pre">checkGeneMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">gMCSs</span></em>, <em class="sig-param"><span class="pre">isoform_separator</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.gMCS.checkGeneMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Check if the gMCS is Minimal and Cut Set
gMCS = gene Minimal Cut Set
Minimal = the knock-off of all the genes of the gMCS is neccesary to
eliminate the objective function
Cut Set = The elimination of all the genes ensures the elimination of the
objective reaction</p>
<p>USAGE:</p>
<blockquote>
<div><p>[IsCutSet, IsMinimal, geneNotMinimal] = checkGeneMCS(model, gMCSs, isoform_separator)</p>
</div></blockquote>
<dl>
<dt>INPUTS:</dt><dd><p>model:                 Metabolic model in COBRA format.
gMCSs:                 cell array which stores all the gMCSs previously</p>
<blockquote>
<div><p>calculated. Each position contains a gMCS.</p>
</div></blockquote>
<dl class="simple">
<dt>isoform_separator:     genes in the model have different isoformes,</dt><dd><p>which are identificated by a separator from the
gene name. (Default = ‘’).</p>
</dd>
</dl>
</dd>
<dt>OUTPUT:</dt><dd><dl class="simple">
<dt>IsCutSet:              array which indicates the biomass producion if</dt><dd><p>all genes of the gMCSs are deleted</p>
</dd>
<dt>IsMinimal:             array which indicates the maximum of biomass if</dt><dd><p>any gene is deleted</p>
</dd>
<dt>geneNotMinimal:        cell array which stores all the genes which</dt><dd><p>make the gMCS not minimal (this errors are
produced by the restriction of coputation
time).</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The isoform separator is neccesary to consider some genes as the same
gene. Example: genes 555.1, 555.2 and 555.3 are considered as gene 555</p>
</div>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.prepareModelNutrientGeneMCS">
<span class="sig-name descname"><span class="pre">prepareModelNutrientGeneMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">exchangeRxns</span></em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.gMCS.prepareModelNutrientGeneMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Add artificial genes for the exchange reactions and prepare model for the
gMCS implementation which includes nutrients from the culture medium.</p>
<p>USAGE:</p>
<blockquote>
<div><p>model = prepareModelNutrientGeneMCS(model, exchangeRxns)</p>
</div></blockquote>
<dl class="simple">
<dt>INPUTS:</dt><dd><p>model:             Metabolic model structure (COBRA Toolbox format).</p>
</dd>
<dt>OPTIONAL INPUTS:</dt><dd><dl class="simple">
<dt>exchangeRxns:      Exchange reactions to be included (default = all</dt><dd><p>reactions which start by ‘<a href="#id2"><span class="problematic" id="id3">EX_</span></a>’, ‘<a href="#id4"><span class="problematic" id="id5">DM_</span></a>’ or ‘<a href="#id6"><span class="problematic" id="id7">sink_</span></a>’
and only have one metabolite involved.</p>
</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt><dd><dl class="simple">
<dt>model:             Metabolic model structure with genes for selected</dt><dd><p>exchanges (COBRA Toolbox format).</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Example</p>
<p>model = prepareModelNutrientGeneMCS(model, exchangeRxns);</p>
</dd></dl>

</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../KLdistance/index.html" class="btn btn-default float-right" title="KLdistance" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../geometricFBA/index.html" class="btn btn-default" title="geometricFBA" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2021, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../_static/sphinx_highlight.js"></script>

  


  
  
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>