

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>mgPipe &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../../../_static/css/cobra.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="../../../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../../../index.html"/>
        <link rel="up" title="Microbiome Modeling Toolbox" href="../index.html"/>
        <link rel="next" title="parsimoniousFBA" href="../../../parsimoniousFBA/index.html"/>
        <link rel="prev" title="Pairwise Interaction Modeling" href="../pairwiseInteractionModeling/index.html"/> 

  
  <script src="../../../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="http://gibbs.unal.edu.co/cobradoc/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../../../_static/../" + item.website_url + "' ><img src='../../../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../../../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../dynamicFBA/index.html">Dynamic FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../FBA/index.html">FBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../findMIIS/index.html">findMIIS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../fluxEnrichmentAnalysis/index.html">fluxEnrichmentAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../FVA/index.html">FVA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../geometricFBA/index.html">geometricFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../gMCS/index.html">gMCS</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../KLdistance/index.html">KLdistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../mdFBA/index.html">mdFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../MOMA/index.html">MOMA</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="../../index.html">MultiSpecies</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="../../SteadyCom/index.html">SteadyCom</a></li>
<li class="toctree-l4 current"><a class="reference internal" href="../index.html">Microbiome Modeling Toolbox</a><ul class="current">
<li class="toctree-l5"><a class="reference internal" href="../additionalAnalysis/index.html">additionalAnalysis</a></li>
<li class="toctree-l5"><a class="reference internal" href="../pairwiseInteractionModeling/index.html">Pairwise Interaction Modeling</a></li>
<li class="toctree-l5 current"><a class="current reference internal" href="#">mgPipe</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../parsimoniousFBA/index.html">parsimoniousFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../relaxedFBA/index.html">relaxedFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rFBA/index.html">rFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rMTA/index.html">rMTA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../robustnessAnalysis/index.html">robustnessAnalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../ROOM/index.html">ROOM</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rumba/index.html">rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../sparseFBA/index.html">sparseFBA</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../thermo/index.html">Thermo</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../uFBA/index.html">uFBA</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../dataIntegration/index.html">Data integration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contact.html">Contact</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../../contents.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../../../index.html">Functions</a> &raquo;</li>
        
          <li><a href="../../../index.html">Analysis</a> &raquo;</li>
        
          <li><a href="../../index.html">MultiSpecies</a> &raquo;</li>
        
          <li><a href="../index.html">Microbiome Modeling Toolbox</a> &raquo;</li>
        
      <li>mgPipe</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="mgpipe">
<span id="id1"></span><h1>mgPipe<a class="headerlink" href="#mgpipe" title="Permalink to this headline">¶</a></h1>
<span class="target" id="module-src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe"></span><dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.adaptVMHDietToAGORA">
<code class="descname">adaptVMHDietToAGORA</code><span class="sig-paren">(</span><em>VMHDiet</em>, <em>setupUsed</em>, <em>AGORAPath</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.adaptVMHDietToAGORA" title="Permalink to this definition">¶</a></dt>
<dd><p>Part of the Microbiome Modeling Toolbox.
This function adapts a diet generated by the Diet Designer on
<a class="reference external" href="https://www.vmh.life">https://www.vmh.life</a> such that microbiome community models created from the
AGORA resource can generate biomass. All metabolites required by at least
one AGORA model are added. Note that the adapted diet that is the output
of this function is specific to the AGORA resource. It is not guaranteed
that other constraint-based models can produce biomass on this diet.
Units are given in mmol/day/person.</p>
<p>USAGE:</p>
<blockquote>
<div>[adaptedDietConstraints, growthOK] = adaptVMHDietToAGORA(VMHDiet, setupUsed, AGORAPath)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><dl class="first last docutils">
<dt>VMHDietConstraints:     Name of text file with VMH exchange reaction IDs</dt>
<dd>and values on lower bounds generated by Diet
Designer on <a class="reference external" href="https://www.vmh.life">https://www.vmh.life</a> (or manually).</dd>
<dt>setupUsed:              Model setup for which the adapted diet will be</dt>
<dd>used. Allowed inputs are AGORA (the single AGORA
models), Pairwise (the microbe-microbe models
generated by the pairwise modeling module), and
Microbiota (the microbe community models
generated by MgPipe).</dd>
</dl>
</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd><dl class="first last docutils">
<dt>AGORAPath:             Path to the AGORA model resource. If entered,</dt>
<dd>growth of all single models on the adapted diet
will be tested.</dd>
</dl>
</dd>
<dt>OUTPUT:</dt>
<dd><dl class="first last docutils">
<dt>adaptedDiet:           Cell array of exchange reaction IDs, values on</dt>
<dd>lower bounds, and values on upper bounds that
can serve as input for the function useDiet.</dd>
</dl>
</dd>
<dt>OPTIONAL OUTPUT:</dt>
<dd><dl class="first last docutils">
<dt>growthOK:              Variable indicating whether all AGORA models could</dt>
<dd>grow on the adapted diet (if 1 then yes).</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.addMicrobeCommunityBiomass">
<code class="descname">addMicrobeCommunityBiomass</code><span class="sig-paren">(</span><em>model</em>, <em>microbeNames</em>, <em>abundances</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.addMicrobeCommunityBiomass" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds a community biomass reaction to a model structure with multiple
microbes based on their relative abundances. If no abundance values are
provided, all n microbes get equal weights (1/n). Assumes a lumen
compartment [u] and a fecal secretion comparment [fe]. Creates a community
biomass metabolite ‘microbeBiomass’ that is secreted from [u] to [fe] and
exchanged from fecal compartment.</p>
<p>USAGE:</p>
<blockquote>
<div>model = addMicrobeCommunityBiomass(model, microbeNames, abundances)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><dl class="first last docutils">
<dt>model:           COBRA model structure with n joined microbes with biomass</dt>
<dd>metabolites ‘Microbe_biomass[c]’.</dd>
<dt>microbeNames:    nx1 cell array of n unique strings that represent</dt>
<dd>each microbe in the model.</dd>
</dl>
</dd>
<dt>OPTIONAL INPUT:</dt>
<dd>abundances:      nx1 vector with the relative abundance of each microbe.</dd>
<dt>OUTPUT:</dt>
<dd>model:           COBRA model structure</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.createPanModels">
<code class="descname">createPanModels</code><span class="sig-paren">(</span><em>agoraPath</em>, <em>panPath</em>, <em>taxonLevel</em>, <em>numWorkers</em>, <em>taxTable</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.createPanModels" title="Permalink to this definition">¶</a></dt>
<dd><p>This function creates pan-models for all unique taxa (e.g., species)
included in the AGORA resource. If reconstructions of multiple strains
in a given taxon are present, the reactions in these reconstructions will
be combined into a pan-reconstruction. The pan-biomass reactions will be
built from the average of all biomasses. Futile cycles that result from
the newly combined reaction content are removed by setting certain
reactions irreversible. These reactions have been determined manually.
NOTE: Futile cycle removal has only been tested at the species and genus
level. Pan-models at higher taxonomical levels (e.g., family) may
contain futile cycles and produce unrealistically high ATP flux.
The pan-models can be used an input for mgPipe if taxon abundance data is
available at a higher level than strain, e.g., species, genus.</p>
<p>USAGE:</p>
<blockquote>
<div>createPanModels(agoraPath,panPath,taxonLevel)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><dl class="first last docutils">
<dt>agoraPath     String containing the path to the AGORA reconstructions.</dt>
<dd>Must end with a file separator.</dd>
<dt>panPath       String containing the path to an empty folder that the</dt>
<dd>created pan-models will be stored in. Must end with a file separator.</dd>
<dt>taxonLevel    String with desired taxonomical level of the pan-models.</dt>
<dd>Allowed inputs are ‘Species’,’Genus’,’Family’,’Order’, ‘Class’,’Phylum’.</dd>
</dl>
</dd>
<dt>OPTIONAL INPUTS</dt>
<dd><p class="first">numWorkers    Number of workers for parallel pool (default: no pool)
taxTable      File with information on taxonomy of reconstruction</p>
<blockquote class="last">
<div>resource (default: ‘AGORA_infoFile.xlsx’)</div></blockquote>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.createPersonalizedModel">
<code class="descname">createPersonalizedModel</code><span class="sig-paren">(</span><em>abundance</em>, <em>resPath</em>, <em>model</em>, <em>sampNames</em>, <em>orglist</em>, <em>couplingMatrix</em>, <em>host</em>, <em>hostBiomassRxn</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.createPersonalizedModel" title="Permalink to this definition">¶</a></dt>
<dd><p>This function creates personalized models from integration of given
organisms abundances into the previously built global setup. Coupling
constraints are also added for each organism. All the operations are
parallelized and the generated personalized models directly saved in .mat
format.</p>
<p>USAGE:</p>
<blockquote>
<div>[createdModels] = createPersonalizedModel(abundance, resPath, model, sampNames, orglist, host, hostBiomassRxn)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">abundance:          table with abundance information
resPath:            char with path of directory where results are saved
model:              “global setup” model in COBRA model structure format
sampNames:          cell array with names of individuals in the study
orglist:            cell array with names of organisms in the study
couplingMatrix:     cell array containing pre-created coupling matrices for</p>
<blockquote>
<div>each organism to be joined (created by
buildModelStorage function)</div></blockquote>
<dl class="docutils">
<dt>host:               Contains the host model if path to host model was</dt>
<dd>defined. Otherwise empty.</dd>
</dl>
<p class="last">hostBiomassRxn:     char with name of biomass reaction in host (default: empty)</p>
</dd>
<dt>OUTPUT:</dt>
<dd>createdModels:      created personalized models</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.detectOutput">
<code class="descname">detectOutput</code><span class="sig-paren">(</span><em>resPath</em>, <em>objNam</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.detectOutput" title="Permalink to this definition">¶</a></dt>
<dd><p>This function checks the existence of a specific file in the results folder.</p>
<p>USAGE:</p>
<blockquote>
<div>mapP = detectOutput(resPath, objNam)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd>resPath:    char with path of directory where results are saved
objNam:     char with name of object to find in the results folder</dd>
<dt>OUTPUTS:</dt>
<dd>mapP:       double indicating if object was found in the result folder</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.extractFullRes">
<code class="descname">extractFullRes</code><span class="sig-paren">(</span><em>resPath</em>, <em>ID</em>, <em>dietType</em>, <em>sampName</em>, <em>fvaCt</em>, <em>nsCt</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.extractFullRes" title="Permalink to this definition">¶</a></dt>
<dd><p>This function is called from the MgPipe pipeline. Its purpose is to retrieve
and export, in a comprehensive way, all the results (fluxes) computed during
the simulations for a specified diet. Since FVA is computed on diet and fecal
exchanges, every metabolite will have four different values for each individual,
values corresponding min and max of uptake and secretion.</p>
<p>USAGE:</p>
<blockquote>
<div>[finRes]= extractFullRes(resPath, ID, dietType, sampName, fvaCt, nsCt)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">resPath:            char with path of directory where results are saved
ID:                 cell array with list of all unique Exchanges to diet/</p>
<blockquote>
<div>fecal compartment</div></blockquote>
<dl class="docutils">
<dt>dietType:           char indicating under which diet to extract results:</dt>
<dd>rDiet (rich diet), sDiet(previously specified diet)
set by default, and pDiet(personalized)if available</dd>
</dl>
<p>sampName:           nx1 cell array cell array with names of individuals in the study
fvaCt:              cell array containing FVA values for maximal uptake
nsCt:               cell array containing FVA values for minimal uptake</p>
<blockquote class="last">
<div>and secretion for setup lumen / diet exchanges</div></blockquote>
</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first last docutils">
<dt>finRes:             cell array with min and max value of uptake and</dt>
<dd>secretion for each metabolite</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.fastSetupCreator">
<code class="descname">fastSetupCreator</code><span class="sig-paren">(</span><em>exch</em>, <em>modelStoragePath</em>, <em>microbeNames</em>, <em>host</em>, <em>objre</em>, <em>buildSetupAll</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.fastSetupCreator" title="Permalink to this definition">¶</a></dt>
<dd><p>creates a microbiota model (min 1 microbe) that can be coupled with a host
model. Microbes and host are connected with a lumen compartment [u], host
can secrete metabolites into body fluids [b]. Diet is simulated as uptake
through the compartment [d], transporters are unidirectional from [d] to
[u]. Secretion goes through the fecal compartment [fe], transporters are
unidirectional from [u] to [fe].
Reaction types
Diet exchange: ‘EX_met[d]’: ‘met[d] &lt;=&gt;’
Diet transporter: ‘DUt_met’: ‘met[d] -&gt; met[u]’
Fecal transporter: ‘UFEt_met’: ‘met[u] -&gt; met[fe]’
Fecal exchanges: ‘EX_met[fe]’: ‘met[fe] &lt;=&gt;’
Microbe uptake/secretion: ‘Microbe_IEX_met[c]tr’: ‘Microbe_met[c] &lt;=&gt; met[u]’
Host uptake/secretion lumen: ‘Host_IEX_met[c]tr’: ‘Host_met[c] &lt;=&gt; met[u]’
Host exchange body fluids: ‘Host_EX_met(e)b’: ‘Host_met[b] &lt;=&gt;’</p>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">modPath:             char with path of directory where models are stored
microbeNames:        nx1 cell array of n unique strings that represent</p>
<blockquote>
<div>each microbe model. Reactions and metabolites of
each microbe will get the corresponding
microbeNames (e.g., ‘Ecoli’) prefix. Reactions
will be named ‘Ecoli_RxnAbbr’ and metabolites
‘Ecoli_MetAbbr[c]’).</div></blockquote>
<dl class="docutils">
<dt>host:                Host COBRA model structure, can be left empty if</dt>
<dd>there is no host model</dd>
</dl>
<p>objre:               char with reaction name of objective function of microbeNames
buildSetupAll:       boolean indicating the strategy that should be used to</p>
<blockquote class="last">
<div>build personalized models: if true, build a global setup model
containing all organisms in at least model (default), false: create
models one by one (recommended for more than ~500 organisms total)</div></blockquote>
</dd>
<dt>OUTPUT:</dt>
<dd>model:               COBRA model structure with all models combined</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.getIndividualSizeName">
<code class="descname">getIndividualSizeName</code><span class="sig-paren">(</span><em>abunFilePath</em>, <em>modPath</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.getIndividualSizeName" title="Permalink to this definition">¶</a></dt>
<dd><p>This function automatically detects organisms, names and number of individuals present
in the study.</p>
<p>USAGE:</p>
<blockquote>
<div>[sampNames, organisms, exMets] = getIndividualSizeName(abunFilePath,modPath)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd>abunFilePath:      char with path and name of file from which to retrieve information
modPath:           char with path of directory where models are stored</dd>
<dt>OUTPUTS:</dt>
<dd><p class="first">sampNames:         nx1 cell array cell array with names of individuals in the study
organisms:         nx1 cell array cell array with names of organisms in the study
exMets:            cell array with all unique extracellular metabolites</p>
<blockquote class="last">
<div>contained in the models</div></blockquote>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.getMappingInfo">
<code class="descname">getMappingInfo</code><span class="sig-paren">(</span><em>modPath</em>, <em>organisms</em>, <em>abunFilePath</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.getMappingInfo" title="Permalink to this definition">¶</a></dt>
<dd><p>This function automatically extracts information from strain abundances in
different individuals and combines this information into different tables.</p>
<p>USAGE:</p>
<blockquote>
<div>[reac, exMets, micRea, binOrg, patOrg, reacPat, reacNumb, reacSet, reacTab, reacAbun, reacNumber] = getMappingInfo(modPath, organisms, abunFilePath, patNumb)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd>organisms:         nx1 cell array cell array with names of organisms in the study
modPath:           char with path of directory where models are stored
abunFilePath:      char with path and name of file from which to retrieve abundance information
patNumb:           number of individuals in the study</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first docutils">
<dt>reac:              cell array with all the unique set of reactions</dt>
<dd>contained in the models</dd>
<dt>exMets:            cell array with all unique extracellular metabolites</dt>
<dd>contained in the models</dd>
<dt>micRea:            binary matrix assessing presence of set of unique</dt>
<dd>reactions for each of the microbes</dd>
<dt>binOrg:            binary matrix assessing presence of specific strains in</dt>
<dd>different individuals</dd>
<dt>reacPat:           matrix with number of reactions per individual</dt>
<dd>(organism resolved)</dd>
</dl>
<p>reacSet:           matrix with names of reactions of each individual
reacTab:           char with names of individuals in the study
reacAbun:          binary matrix with presence/absence of reaction per</p>
<blockquote>
<div>individual: to compare different individuals</div></blockquote>
<p class="last">reacNumber:        number of unique reactions of each individual</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.guidedSim">
<code class="descname">guidedSim</code><span class="sig-paren">(</span><em>model</em>, <em>rl</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.guidedSim" title="Permalink to this definition">¶</a></dt>
<dd><p>This function is part of the MgPipe pipeline and runs FVAs on a series of
selected reactions with different possible FVA functions. Solver is
automatically set to ‘cplex’, objective function is maximized, and
optPercentage set to 99.99.</p>
<p>USAGE:</p>
<blockquote>
<div>[minFlux, maxFlux] = guidedSim(model, fvaType, rl)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><dl class="first docutils">
<dt>model:         COBRA model structure with n joined microbes with biomass</dt>
<dd>metabolites ‘Microbe_biomass[c]’.</dd>
</dl>
<p class="last">rl:            nx1 vector with the reactions of interest.
solver:        char with slver name to use.</p>
</dd>
<dt>OUTPUTS:</dt>
<dd>minFlux:      Minimum flux for each reaction
maxFlux:      Maximum flux for each reaction</dd>
</dl>
<p>..Author:  Federico Baldini,  2017-2018</p>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.initMgPipe">
<code class="descname">initMgPipe</code><span class="sig-paren">(</span><em>modPath</em>, <em>abunFilePath</em>, <em>computeProfiles</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.initMgPipe" title="Permalink to this definition">¶</a></dt>
<dd><p>This function initializes the mgPipe pipeline and sets the optional input
variables if not defined.</p>
<dl class="docutils">
<dt>USAGE</dt>
<dd>[init, netSecretionFluxes, netUptakeFluxes, Y, modelStats, summary, statistics] = initMgPipe(modPath, abunFilePath, computeProfiles, varargin)</dd>
<dt>INPUTS:</dt>
<dd><p class="first">modPath:                char with path of directory where models are stored
abunFilePath:           char with path and name of file from which to retrieve abundance information
computeProfiles:        boolean defining whether flux variability analysis to</p>
<blockquote class="last">
<div>compute the metabolic profiles should be performed.</div></blockquote>
</dd>
<dt>OPTIONAL INPUTS:</dt>
<dd><p class="first">resPath:                char with path of directory where results are saved
dietFilePath:           char with path of directory where the diet is saved.</p>
<blockquote>
<div>Can also be a character array with a separate diet for
each individual, in that case, size(dietFilePath,1)
needs to equal the length of samples.</div></blockquote>
<p>infoFilePath:           char with path to stratification criteria if available
hostPath:               char with path to host model, e.g., Recon3D (default: empty)
hostBiomassRxn:         char with name of biomass reaction in host (default: empty)
hostBiomassRxnFlux:     double with the desired upper bound on flux through the host</p>
<blockquote>
<div>biomass reaction (default: 1)</div></blockquote>
<p>objre:                  char with reaction name of objective function of organisms
buildSetupAll:               boolean indicating the strategy that should be used to</p>
<blockquote>
<div>build personalized models: if true, build a global setup model
containing all organisms in at least model (default), false: create
models one by one (recommended for more than ~500 organisms total)</div></blockquote>
<dl class="docutils">
<dt>saveConstrModels:       boolean indicating if models with imposed</dt>
<dd>constraints are saved externally</dd>
</dl>
<p>numWorkers:             integer indicating the number of cores to use for parallelization
rDiet:                  boolean indicating if to enable also rich diet simulations (default: ‘false’)
pDiet:                  boolean indicating if to enable also personalized diet simulations (default: ‘false’)
includeHumanMets:       boolean indicating if human-derived metabolites</p>
<blockquote>
<div>present in the gut should be provided to the models (default: true)</div></blockquote>
<p>lowerBMBound:           lower bound on community biomass (default=0.4)
repeatSim:              boolean defining if simulations should be repeated and previous results</p>
<blockquote>
<div>overwritten (default=false)</div></blockquote>
<dl class="last docutils">
<dt>adaptMedium:            boolean indicating if the medium should be</dt>
<dd>adapted through the adaptVMHDietToAGORA
function or used as is (default=true)</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt>
<dd><p class="first">init:                   status of initialization
netSecretionFluxes:     Net secretion fluxes by microbiome community models
netUptakeFluxes:        Net uptake fluxes by microbiome community models
Y:                      Classical multidimensional scaling
modelStats:             Reaction and metabolite numbers for each model
summary:                Table with average, median, minimal, and maximal</p>
<blockquote>
<div>reactions and metabolites</div></blockquote>
<dl class="last docutils">
<dt>statistics:             If info file with stratification is provided, will</dt>
<dd>determine if there is a significant difference.</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.loadUncModels">
<code class="descname">loadUncModels</code><span class="sig-paren">(</span><em>modPath</em>, <em>organisms</em>, <em>objre</em>, <em>printLevel</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.loadUncModels" title="Permalink to this definition">¶</a></dt>
<dd><p>This function loads and unconstrains metabolic models from a specific folder</p>
<p>USAGE:</p>
<blockquote>
<div>models = loadUncModels(modPath, organisms, objre)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd>organisms:           nx1 cell array cell array with names of organisms in the study
modPath:             char with path of directory where models are stored
objre:               char with reaction name of objective function of organisms
printLevel:          Verbose level (default: printLevel = 1)</dd>
<dt>OUTPUT:</dt>
<dd>models:              nx1 cell array cell array with models of organisms in the study</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.makeDummyModel">
<code class="descname">makeDummyModel</code><span class="sig-paren">(</span><em>numMets</em>, <em>numRxns</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.makeDummyModel" title="Permalink to this definition">¶</a></dt>
<dd><p>Makes an empty model with numMets rows for metabolites and numRxns columns
for reactions. Includes all fields that are necessary to join models.</p>
<p>USAGE:</p>
<blockquote>
<div>dummy = makeDummyModel(numMets, numRxns)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd>numMets:       Number of metabolites
numRxns:       Number of reactions</dd>
<dt>OUTPUT:</dt>
<dd>dummy:         Empty COBRA model structure</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.mgPipe">
<code class="descname">mgPipe</code><span class="sig-paren">(</span><em>modPath</em>, <em>abunFilePath</em>, <em>computeProfiles</em>, <em>resPath</em>, <em>dietFilePath</em>, <em>infoFilePath</em>, <em>hostPath</em>, <em>hostBiomassRxn</em>, <em>hostBiomassRxnFlux</em>, <em>objre</em>, <em>buildSetupAll</em>, <em>saveConstrModels</em>, <em>figForm</em>, <em>numWorkers</em>, <em>rDiet</em>, <em>pDiet</em>, <em>includeHumanMets</em>, <em>lowerBMBound</em>, <em>repeatSim</em>, <em>adaptMedium</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.mgPipe" title="Permalink to this definition">¶</a></dt>
<dd><p>mgPipe is a MATLAB based pipeline to integrate microbial abundances
(coming from metagenomic data) with constraint based modeling, creating
individuals’ personalized models.
The pipeline is divided in 3 parts:
[PART 1] Analysis of individuals’ specific microbes abundances are computed.
[PART 2]: 1 Constructing a global metabolic model (setup) containing all the
microbes listed in the study. 2 Building individuals’ specific models
integrating abundance data retrieved from metagenomics. For each organism,
reactions are coupled to the objective function.
[PART 3] Simulations under different diet regimes.
mgPipe was created (and tested) for AGORA 1.0 please first download AGORA
version 1.0 from <a class="reference external" href="https://www.vmh.life/#downloadview">https://www.vmh.life/#downloadview</a> and place the mat files
into a folder.</p>
<dl class="docutils">
<dt>USAGE:</dt>
<dd>[netSecretionFluxes, netUptakeFluxes, Y, modelStats,summary, statistics] = mgPipe(modPath, abunFilePath, computeProfiles, resPath, dietFilePath, infoFilePath, hostPath, hostBiomassRxn, hostBiomassRxnFlux, objre, buildSetupAll, saveConstrModels, figForm, numWorkers, rDiet, pDiet, includeHumanMets, lowerBMBound, repeatSim, adaptMedium)</dd>
<dt>INPUTS:</dt>
<dd><p class="first">modPath:                char with path of directory where models are stored
abunFilePath:           char with path and name of file from which to retrieve abundance information
computeProfiles:        boolean defining whether flux variability analysis to</p>
<blockquote>
<div>compute the metabolic profiles should be performed.</div></blockquote>
<p>resPath:                char with path of directory where results are saved
dietFilePath:           char with path of directory where the diet is saved
infoFilePath:           char with path to stratification criteria if available
hostPath:               char with path to host model, e.g., Recon3D (default: empty)
hostBiomassRxn:         char with name of biomass reaction in host (default: empty)
hostBiomassRxnFlux:     double with the desired flux through the host</p>
<blockquote>
<div>biomass reaction (default: zero)</div></blockquote>
<p>objre:                  char with reaction name of objective function
buildSetupAll:               boolean indicating the strategy that should be used to</p>
<blockquote>
<div>build personalized models: if true, build a global setup model
containing all organisms in at least model (default), false: create
models one by one (recommended for more than ~500 organisms total)</div></blockquote>
<dl class="docutils">
<dt>saveConstrModels:       boolean indicating if models with imposed</dt>
<dd>constraints are saved externally</dd>
</dl>
<p>figForm:                format to use for saving figures
numWorkers:             integer indicating the number of cores to use for parallelization
rDiet:                  boolean indicating if to enable also rich diet simulations (default: ‘false’)
pDiet:                  boolean indicating if to enable also personalized diet simulations (default: ‘false’)
includeHumanMets:       boolean indicating if human-derived metabolites</p>
<blockquote>
<div>present in the gut should be provided to the models (default: true)</div></blockquote>
<p>lowerBMBound:           lower bound on community biomass (default=0.4)
repeatSim:              boolean defining if simulations should be repeated and previous results</p>
<blockquote>
<div>overwritten (default=false)</div></blockquote>
<dl class="last docutils">
<dt>adaptMedium:            boolean indicating if the medium should be</dt>
<dd>adapted through the adaptVMHDietToAGORA
function or used as is (default=true)</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt>
<dd><p class="first">init:                   status of initialization
netSecretionFluxes:     Net secretion fluxes by microbiome community models
netUptakeFluxes:        Net uptake fluxes by microbiome community models
Y:                      Classical multidimensional scaling
modelStats:             Reaction and metabolite numbers for each model
summary:                Table with average, median, minimal, and maximal</p>
<blockquote>
<div>reactions and metabolites</div></blockquote>
<dl class="last docutils">
<dt>statistics:             If info file with stratification is provided, will</dt>
<dd>determine if there is a significant difference.</dd>
</dl>
</dd>
<dt>AUTHORS:</dt>
<dd><ul class="first last simple">
<li>Federico Baldini, 2017-2018</li>
<li>Almut Heinken, 07/20: converted to function</li>
<li>Almut Heinken, 01/21: added option for creation of each personalized model separately</li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.mgSimResCollect">
<code class="descname">mgSimResCollect</code><span class="sig-paren">(</span><em>resPath</em>, <em>sampNames</em>, <em>exchanges</em>, <em>rDiet</em>, <em>pDiet</em>, <em>infoFilePath</em>, <em>netProduction</em>, <em>netUptake</em>, <em>figForm</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.mgSimResCollect" title="Permalink to this definition">¶</a></dt>
<dd><p>This function is called from the MgPipe pipeline. Its purpose is to compute
NMPCs from simulations with different diet on multiple microbiota models.
Results are outputted as .csv and a PCoA on NMPCs to group microbiota
models of individuals for similar metabolic profile is also
computed and outputted.</p>
<p>USAGE:</p>
<blockquote>
<div>[fSp, Y]= mgSimResCollect(resPath, sampNames, sampNames, rDiet, pDiet, infoFilePath, netProduction, figForm)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">resPath:            char with path of directory where results are saved
sampNames:          nx1 cell array cell array with names of individuals in the study
exchanges:          cell array with list of all unique Exchanges to diet/</p>
<blockquote>
<div>fecal compartment</div></blockquote>
<p>rDiet:              number (double) indicating if to simulate a rich diet
pDiet:              number (double) indicating if a personalized diet</p>
<blockquote>
<div>is available and should be simulated</div></blockquote>
<dl class="docutils">
<dt>infoFilePath:    char indicating, if stratification criteria are available,</dt>
<dd>full path and name to related documentation(default: no)
is available</dd>
</dl>
<p class="last">netProduction:              cell array containing FVA values for maximal uptake
figForm:            char indicating the format of figures</p>
</dd>
<dt>OUTPUTS:</dt>
<dd>netSecretionFluxes: cell array with computed NMPCs
netUptakeFluxes:    cell array with computed uptake potential
Y:                  classical multexchangesimensional scaling</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.microbiotaModelSimulator">
<code class="descname">microbiotaModelSimulator</code><span class="sig-paren">(</span><em>resPath</em>, <em>exMets</em>, <em>sampNames</em>, <em>dietFilePath</em>, <em>hostPath</em>, <em>hostBiomassRxn</em>, <em>hostBiomassRxnFlux</em>, <em>numWorkers</em>, <em>rDiet</em>, <em>pDiet</em>, <em>saveConstrModels</em>, <em>computeProfiles</em>, <em>includeHumanMets</em>, <em>lowerBMBound</em>, <em>repeatSim</em>, <em>adaptMedium</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.microbiotaModelSimulator" title="Permalink to this definition">¶</a></dt>
<dd><p>This function is called from the MgPipe pipeline. Its purpose is to apply
different diets (according to the user’s input) to the microbiota models
and run simulations computing FVAs on exchanges reactions of the microbiota
models. The output is saved in multiple .mat objects. Intermediate saving
checkpoints are present.</p>
<p>USAGE:</p>
<blockquote>
<div>[exchanges, netProduction, netUptake, presol, inFesMath] = microbiotaModelSimulator(resPath, exMets, sampNames, dietFilePath, hostPath, hostBiomassRxn, hostBiomassRxnFlux, numWorkers, rDiet, pDiet, saveConstrModels, computeProfiles, includeHumanMets, lowerBMBound, repeatSim, adaptMedium)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">resPath:            char with path of directory where results are saved
exMets:             cell array with all unique extracellular metabolites</p>
<blockquote>
<div>contained in the models</div></blockquote>
<p>sampNames:          cell array with names of individuals in the study
dietFilePath:       path to and name of the text file with dietary information
hostPath:           char with path to host model, e.g., Recon3D (default: empty)
hostBiomassRxn:     char with name of biomass reaction in host (default: empty)
hostBiomassRxnFlux: double with the desired upper bound on flux through the host</p>
<blockquote>
<div>biomass reaction (default: 1)</div></blockquote>
<p>numWorkers:         integer indicating the number of cores to use for parallelization
rDiet:              boolean indicating if to simulate a rich diet
pDiet:              boolean indicating if a personalized diet</p>
<blockquote>
<div>is available and should be simulated</div></blockquote>
<dl class="docutils">
<dt>saveConstrModels:   boolean indicating if models with imposed</dt>
<dd>constraints are saved externally</dd>
<dt>computeProfiles:    boolean defining whether flux variability analysis to</dt>
<dd>compute the metabolic profiles should be performed.</dd>
<dt>includeHumanMets:   boolean indicating if human-derived metabolites</dt>
<dd>present in the gut should be provexchangesed to the models (default: true)</dd>
</dl>
<p>lowerBMBound        Minimal amount of community biomass in mmol/person/day enforced (default=0.4)
repeatSim:          boolean defining if simulations should be repeated and previous results</p>
<blockquote>
<div>overwritten (default=false)</div></blockquote>
<dl class="last docutils">
<dt>adaptMedium:        boolean indicating if the medium should be adapted through the</dt>
<dd>adaptVMHDietToAGORA function or used as is (default=true)</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first docutils">
<dt>exchanges:          cell array with list of all unique Exchanges to diet/</dt>
<dd>fecal compartment</dd>
<dt>netProduction:              cell array containing FVA values for maximal uptake</dt>
<dd>and secretion for setup lumen / diet exchanges</dd>
<dt>netUptake:               cell array containing FVA values for minimal uptake</dt>
<dd>and secretion for setup lumen / diet exchanges</dd>
<dt>presol              array containing values of microbiota models</dt>
<dd>objective function</dd>
</dl>
<p class="last">inFesMat            cell array with names of infeasible microbiota models</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.normalizeCoverage">
<code class="descname">normalizeCoverage</code><span class="sig-paren">(</span><em>abunFilePath</em>, <em>cutoff</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.normalizeCoverage" title="Permalink to this definition">¶</a></dt>
<dd><p>This functions normalizes the coverage in a given file with organism
abundances such that they sum up to 1 for each sample.</p>
<dl class="docutils">
<dt>USAGE</dt>
<dd>[normalizedCoverage,normalizedCoveragePath] = normalizeCoverage(abunFilePath,cutoff)</dd>
</dl>
<p>INPUT
abunFilePath                   Path to table with not yet normalized relative</p>
<blockquote>
<div>abundances</div></blockquote>
<p>OPTIONAL INPUT
cutoff                    Cutoff for normalized abundances that are</p>
<blockquote>
<div>considered below detection limit, respective
organisms will be removed from the samples
(default: 0.0001)</div></blockquote>
<p>OUTPUTS
normalizedCoverage        Table with normalized abundances
normalizedCoveragePath    Path to csv file with normalized abundances</p>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.parsave">
<code class="descname">parsave</code><span class="sig-paren">(</span><em>fname</em>, <em>microbiota_model</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.parsave" title="Permalink to this definition">¶</a></dt>
<dd><p>Saves a model from a parfor loop - might not work in R2105b</p>
<p>USAGE:</p>
<blockquote>
<div>parsave(fname, microbiota_model)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd>fname:               name of file
microbiota_model:    name of variable</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.plotMappingInfo">
<code class="descname">plotMappingInfo</code><span class="sig-paren">(</span><em>resPath</em>, <em>patOrg</em>, <em>reacPat</em>, <em>reacTab</em>, <em>reacNumber</em>, <em>infoFilePath</em>, <em>figForm</em>, <em>sampNames</em>, <em>organisms</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.plotMappingInfo" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes and automatically plots information coming from
the mapping data as metabolic diversity and classical multidimensional
scaling of individuals’ reactions repertoire. If the last 2 arguments are
specified MDS plots will be annotated with samples and organisms names</p>
<p>USAGE:</p>
<blockquote>
<div>Y = plotMappingInfo(resPath, patOrg, reacPat, reacTab, reacNumber, infoFilePath, figForm, sampNames, organisms)</div></blockquote>
<dl class="docutils">
<dt>INPUTS:</dt>
<dd><p class="first">resPath:            char with path of directory where results are saved
reac:               nx1 cell array with all the unique set of reactions</p>
<blockquote>
<div>contained in the models</div></blockquote>
<dl class="docutils">
<dt>micRea:             binary matrix assessing presence of set of unique</dt>
<dd>reactions for each of the microbes</dd>
</dl>
<p>reacSet:            matrix with names of reactions of each individual
reacTab:            binary matrix with presence/absence of reaction per</p>
<blockquote>
<div>individual.</div></blockquote>
<p>reacAbun:           matrix with abundance of reaction per individual
reacNumber:         number of unique reactions of each individual
infoFilePath:       char indicating, if stratification criteria are available,</p>
<blockquote>
<div>full path and name to related documentation(default: no)
is available</div></blockquote>
<p class="last">figForm:            format to use for saving figures
sampNames:           nx1 cell array cell array with names of individuals in the study
organisms:          nx1 cell array cell array with names of organisms in the study</p>
</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first last docutils">
<dt>Y:                  classical multidimensional scaling of individuals’</dt>
<dd>reactions repertoire</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.rebuildModel">
<code class="descname">rebuildModel</code><span class="sig-paren">(</span><em>model</em>, <em>database</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.rebuildModel" title="Permalink to this definition">¶</a></dt>
<dd><p>Rebuilds a genome-scale reconstruction with Virtual Metabolic Human (VMH)
metabolic and reaction nomenclature while ensuring quality control through
rBioNet.</p>
<p>USAGE
[rebuiltModel] = rebuildModel(model,database)</p>
<dl class="docutils">
<dt>INPUT</dt>
<dd><p class="first">model         COBRA model structure
database      Structure containing rBioNet reaction and metabolite</p>
<blockquote class="last">
<div>database</div></blockquote>
</dd>
<dt>OUTPUT</dt>
<dd>rebuiltModel  Quality-controlled COBRA model structure</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.retrieveModelStats">
<code class="descname">retrieveModelStats</code><span class="sig-paren">(</span><em>modelPath</em>, <em>modelList</em>, <em>infoFilePath</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.retrieveModelStats" title="Permalink to this definition">¶</a></dt>
<dd><p>This function retrieves statistics on the number of reactions and
metabolites across microbiome models. If a file with stratification
information on individuals is provided, it will also determine if
reaction and metabolites numbers are significantly different between
groups.</p>
<p>USAGE:</p>
<blockquote>
<div>[modelStats,summary,statistics]=retrieveModelStats(modelPath, modelList, infoFilePath)</div></blockquote>
<p>INPUTS
modelPath:        Path to models for which statistics should be retrieved
modelList:        Cell array with names of models for which statistics</p>
<blockquote>
<div>should be retrieved</div></blockquote>
<p>OPTIONAL INPUT:
infoFilePath:     char with path to stratification criteria if available</p>
<p>OUTPUT
modelStats:       Reaction and metabolite numbers for each model
summary:          Table with average, median, minimal, and maximal</p>
<blockquote>
<div>reactions and metabolites</div></blockquote>
<p>OPTIONAL OUTPUT:
statistics:       If info file with stratification is provided, will</p>
<blockquote>
<div>determine if there is a significant difference.</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.translateMetagenome2AGORA">
<code class="descname">translateMetagenome2AGORA</code><span class="sig-paren">(</span><em>MetagenomeAbundancePath</em>, <em>sequencingDepth</em><span class="sig-paren">)</span><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.mgPipe.translateMetagenome2AGORA" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates organism identifiers in a published metagenomic or 16S rRNA
data file with organism abundances (retrieved e.g., from  MetaPhlAn) to
AGORA pan-model IDs. This will not catch every case since the format of
input files with abundance data greatly varies. Feel free to modify this
function and submit a pull request to enable more input files to be
translated to AGORA. Moreover, slight spelling variations in taxa across
input files may  result to taxa not being mapped. Check the unmappedRows
output to identify these cases and modify the function accordingly.
Pan-models that can be used to create microbiome models in mgPipe can be
created with the function createPanModels.</p>
<p>USAGE:</p>
<blockquote>
<div>[translatedAbundances,normalizedAbundances,unmappedRows]=translateMetagenome2AGORA(MetagenomeAbundancePath,sequencingDepth)</div></blockquote>
<dl class="docutils">
<dt>INPUT:</dt>
<dd><dl class="first last docutils">
<dt>MetagenomeAbundancePath   String containing the path to csv file with</dt>
<dd>organism abundance data retrieved from
16S rRNA or metagenomic samples (example:
‘SRP065497_taxonomy_abundances_v3.0.tsv’).</dd>
</dl>
</dd>
<dt>OPTIONAL INPUT:</dt>
<dd><dl class="first last docutils">
<dt>sequencingDepth           Sequencing depth on the taxonomical level</dt>
<dd>in the input data (e.g., genus, species).
Allowed inputs are ‘Species’,’Genus’,
‘Family’,’Order’, ‘Class’, ‘Phylum’.
Default: ‘Species’.</dd>
</dl>
</dd>
<dt>OUTPUTS:</dt>
<dd><dl class="first last docutils">
<dt>translatedAbundances      Abundances with organism names from the</dt>
<dd>input file translated to AGORA pan-model IDs</dd>
<dt>normalizedAbundances      Translated abundances normalized so they sum</dt>
<dd>up to 1 for each sample</dd>
<dt>unmappedRows              Taxa on the selected taxonomical level that</dt>
<dd>could not be mapped to AGORA pan-models</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../../../parsimoniousFBA/index.html" class="btn btn-default float-right" title="parsimoniousFBA" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../pairwiseInteractionModeling/index.html" class="btn btn-default" title="Pairwise Interaction Modeling" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2020, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../../_static/language_data.js"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/gh/mathjax/MathJax@2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  


  
  
    <script type="text/javascript" src="../../../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>